---
title: "Expedition-eDNA-Ecol-analysis"
author: "Liz Suter"
date: "started 2025-03-27"
output: html_document
---

# Load packages

```{r}
# rm(list = ls())
library(tidyverse)
library(phyloseq)
library(vegan)
library(RColorBrewer)
library(microbiome)
library(metagMisc)
library(fantaxtic)
library(ggrepel)
library(paletteer)
library(pals)
library(compositions)
library(hms)
library(ggvegan)
library(sf)
library(ggimage)
library(cowplot)
library(patchwork)
library(spOccupancy)
library(corrplot)
library(geosphere)
library(plotly)

sessionInfo()
```

# Load data from QC and Filtering Workbook
```{r}
load("figures-expedition/environment_upto_export.RData")
```


# Visualizations

## Bar plots of top species from whole dataset
### Prepare data
#### Expedition/ Mifish
```{r}
# Remove control samples
ps2024_exp_mifish_nocontrols <- subset_samples(ps2024_exp_mifish, is.na(controls))

# Convert to dataframe
exp_mifish_species_abundance <- data.frame(otu_table(ps2024_exp_mifish_nocontrols)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_mifish_nocontrols)) %>%
  mutate(ASV = rownames(.))

exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Group and sum
exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  group_by(Species.CommonName) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()

# Calculate relative abundances across the whole dataset
exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Retain only most abundant sp
exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 12) # Show top 12 species


# Add columns indicating which library and sample type
exp_mifish_species_abundance$library <- "MiFish"
exp_mifish_species_abundance$sample_type <- "USV"


exp_mifish_species_abundance
```


#### Expedition/ Elas
```{r}
# Remove control samples
ps2024_exp_elas_nocontrols <- subset_samples(ps2024_exp_elas, is.na(controls))

# Convert to dataframe
exp_elas_species_abundance <- data.frame(otu_table(ps2024_exp_elas_nocontrols)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_elas_nocontrols)) %>%
  mutate(ASV = rownames(.))

exp_elas_species_abundance <- exp_elas_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Group and sum
exp_elas_species_abundance <- exp_elas_species_abundance %>%
  group_by(Species.CommonName) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()

# Calculate relative abundances across the whole dataset
exp_elas_species_abundance <- exp_elas_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Plot only most abundant
exp_elas_species_abundance <- exp_elas_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 2) # Show top 2 species


# Add columns indicating which library and sample type
exp_elas_species_abundance$library <- "Elas02"
exp_elas_species_abundance$sample_type <- "USV"

exp_elas_species_abundance
```

#### Expedition/ CO1

There is too much going on here to make a similar figure of top species, so group by different taxonomic levels of interest...

##### Phyla
First, make a figure for top 10 phyla
```{r}
# Remove control samples
ps2024_exp_co1_phyla <- subset_samples(ps2024_exp_co1, is.na(controls))


# Aggregate
ps2024_exp_co1_phyla <- tax_glom(ps2024_exp_co1_phyla, taxrank = "Phylum") 

# Summarize total abundances for each species across the whole dataset
exp_co1_phyla_abundance <- data.frame(otu_table(ps2024_exp_co1_phyla)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_phyla))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))

exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Remove columns no longer relevant
exp_co1_phyla_abundance <- exp_co1_phyla_abundance[, colSums(!is.na(exp_co1_phyla_abundance)) > 0]

# Add columns indicating which library and sample type
exp_co1_phyla_abundance$library <- "CO1"
exp_co1_phyla_abundance$sample_type <- "USV"

# Add back in colors and common names of phyla
commonnames <- read_csv(file = "../eDNA-databases/commonnames.csv") 

# Add back in colors and common name
exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>% 
  left_join(commonnames, by = c("Phylum" = "Taxon"))

exp_co1_phyla_abundance

```

##### Phyla of Animals only
All animals
```{r}
ps2024_exp_co1_animals <- subset_taxa(ps2024_exp_co1_nocontrols, Phylum %in% c("Annelida","Arthropoda","Cnidaria","Echinodermata","Mollusca","Ctenophora","Chordata","Porifera","Onychophora","Nemertea","Bryozoa","Platyhelminthes","Hemichordata","Nematoda","Gastrotricha","Placozoa","Xenacoelomorpha","Entoprocta","Tardigrada"))


# Aggregate
ps2024_exp_co1_animals_phyla <- tax_glom(ps2024_exp_co1_animals, taxrank = "Phylum") 

# Summarize total abundances for each species across the whole dataset
exp_co1_animals_phyla <- data.frame(otu_table(ps2024_exp_co1_animals_phyla)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
exp_co1_animals_phyla <- exp_co1_animals_phyla %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_animals_phyla))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))

exp_co1_animals_phyla <- exp_co1_animals_phyla %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Remove columns no longer relevant
exp_co1_animals_phyla <- exp_co1_animals_phyla[, colSums(!is.na(exp_co1_animals_phyla)) > 0]

# Add columns indicating which library and sample type
exp_co1_animals_phyla$library <- "CO1"
exp_co1_animals_phyla$sample_type <- "USV"

# note- this table doesn't have colors yet

exp_co1_animals_phyla

```

##### Animals of interest
```{r}
ps2024_exp_co1_animals_ofinterest <- subset_taxa(ps2024_exp_co1_nocontrols, Phylum %in% c("Arthropoda","Mollusca","Chordata","Echinodermata"))


# Aggregate
ps2024_exp_co1_animals_ofinterest_class <- tax_glom(ps2024_exp_co1_animals_ofinterest, taxrank = "Class") 

# Summarize total abundances for each species across the whole dataset
exp_co1_animals_ofinterest_class <- data.frame(otu_table(ps2024_exp_co1_animals_ofinterest_class)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
exp_co1_animals_ofinterest_class <- exp_co1_animals_ofinterest_class %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_animals_ofinterest_class))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))

exp_co1_animals_ofinterest_class <- exp_co1_animals_ofinterest_class %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Remove columns no longer relevant
exp_co1_animals_ofinterest_class <- exp_co1_animals_ofinterest_class[, colSums(!is.na(exp_co1_animals_ofinterest_class)) > 0]

# Add columns indicating which library and sample type
exp_co1_animals_ofinterest_class$library <- "CO1"
exp_co1_animals_ofinterest_class$sample_type <- "USV"

# Add back in colors and common names of phyla
commonnames <- read_csv(file = "../eDNA-databases/commonnames.csv") 

# Add back in colors and common name
exp_co1_animals_ofinterest_class <- exp_co1_animals_ofinterest_class %>% 
  left_join(commonnames, by = c("Class" = "Taxon"))

exp_co1_animals_ofinterest_class

```




##### Teleosts and Elasmobranchs
Then separate out all Actinopteri and elasmos and aggregate at species level
```{r}
# Remove control samples
ps2024_exp_co1_nocontrols <- subset_samples(ps2024_exp_co1, is.na(controls))

# Filter to just Actinopteri
ps2024_exp_co1_actinopteri_elasmos <- subset_taxa(ps2024_exp_co1_nocontrols, Class %in% c("Actinopteri","Chondrichthyes"))

# Convert to dataframe
exp_co1_actinopteri_elasmos_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_actinopteri_elasmos)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_actinopteri_elasmos)) %>%
  mutate(ASV = rownames(.))

exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Group and sum
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  group_by(Species.CommonName) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()

# Remove unclassified Actinopteri
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  filter(!Species.CommonName == "Actinopteri")

# Calculate relative abundances across the whole dataset
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Plot only most abundant
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 14) # Show top 14 species


# Add columns indicating which library and sample type
exp_co1_actinopteri_elasmos_sp_abundance$library <- "CO1"
exp_co1_actinopteri_elasmos_sp_abundance$sample_type <- "USV"



exp_co1_actinopteri_elasmos_sp_abundance
```




##### Other Chordates
```{r}
# Filter 
ps2024_exp_co1_chordates <- subset_taxa(ps2024_exp_co1_nocontrols,
  Phylum == "Chordata" & Class != "Actinopteri" & Class != "Chondrichthyes")

# Convert to dataframe
exp_co1_chordates_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_chordates)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_chordates)) %>%
  mutate(ASV = rownames(.))

exp_co1_chordates_sp_abundance <- exp_co1_chordates_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))


# Group and Sum
exp_co1_chordates_sp_abundance <- exp_co1_chordates_sp_abundance %>%
  group_by(Species) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()


# Calculate relative abundances across the whole dataset
exp_co1_chordates_sp_abundance <- exp_co1_chordates_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add columns indicating which library and sample type
exp_co1_chordates_sp_abundance$library <- "CO1"
exp_co1_chordates_sp_abundance$sample_type <- "USV"

exp_co1_chordates_sp_abundance

```
Mostly tunicates


##### Mollusks
Check Molluscs
```{r}
# Filter 
ps2024_exp_co1_molluscs <- subset_taxa(ps2024_exp_co1_nocontrols,
  Phylum == "Mollusca")

# Convert to dataframe
exp_co1_molluscs_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_molluscs)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_molluscs)) %>%
  mutate(ASV = rownames(.))

exp_co1_molluscs_sp_abundance <- exp_co1_molluscs_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))


# Group and Sum
exp_co1_molluscs_sp_abundance <- exp_co1_molluscs_sp_abundance %>%
  group_by(Species) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()


# Calculate relative abundances across the whole dataset
exp_co1_molluscs_sp_abundance <- exp_co1_molluscs_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add columns indicating which library and sample type
exp_co1_molluscs_sp_abundance$library <- "CO1"
exp_co1_molluscs_sp_abundance$sample_type <- "USV"

exp_co1_molluscs_sp_abundance

```

##### Arthropods

```{r}
# Filter 
ps2024_exp_co1_arthropods <- subset_taxa(ps2024_exp_co1_nocontrols,
  Phylum == "Arthropoda")

# Convert to dataframe
exp_co1_arthropods_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_arthropods)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_arthropods)) %>%
  mutate(ASV = rownames(.))

exp_co1_arthropods_sp_abundance <- exp_co1_arthropods_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))


# Group and Sum
exp_co1_arthropods_sp_abundance <- exp_co1_arthropods_sp_abundance %>%
  group_by(Species) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()


# Calculate relative abundances across the whole dataset
exp_co1_arthropods_sp_abundance <- exp_co1_arthropods_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add columns indicating which library and sample type
exp_co1_arthropods_sp_abundance$library <- "CO1"
exp_co1_arthropods_sp_abundance$sample_type <- "USV"

exp_co1_arthropods_sp_abundance

```



#### Trawl/ Mifish
```{r}
# Remove controls
ps2024_trawl_mifish_species <- subset_taxa(subset_samples(ps2024_trawl_mifish, is.na(controls)))

# Aggregate by Common Name
ps2024_trawl_mifish_species <- tax_glom(ps2024_trawl_mifish_species, taxrank = "Species.CommonName") 

# Summarize total abundances for each species across the whole dataset
trawl_mifish_species_abundance <- data.frame(otu_table(ps2024_trawl_mifish_species)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_trawl_mifish_species))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))


trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Plot only most abundant
trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 12) # Show top 12 species

# Remove ASV column
trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>% select(-ASV)

# Add columns indicating which library and sample type
trawl_mifish_species_abundance$library <- "MiFish"
trawl_mifish_species_abundance$sample_type <- "Manual Sample"

trawl_mifish_species_abundance
```

#### Trawl/ Elas
```{r}
# Remove control samples
ps2024_trawl_elas_nocontrols <- subset_samples(ps2024_trawl_elas, is.na(controls))

# Convert to dataframe
trawl_elas_species_abundance <- data.frame(otu_table(ps2024_trawl_elas_nocontrols)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_trawl_elas_nocontrols)) %>%
  mutate(ASV = rownames(.))

trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Group and sum
trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  group_by(Species.CommonName) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()

# Calculate relative abundances across the whole dataset
trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Plot only most abundant
trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 2) # Show top 2 species


# Add columns indicating which library and sample type
trawl_elas_species_abundance$library <- "Elas02"
trawl_elas_species_abundance$sample_type <- "Manual Sample"

trawl_elas_species_abundance
```




#### Summary of top Species
Sum what % of total the top species comprise
```{r}
sum(exp_mifish_species_abundance$RelativeAbundance)
sum(trawl_mifish_species_abundance$RelativeAbundance)

sum(exp_elas_species_abundance$RelativeAbundance)
sum(trawl_elas_species_abundance$RelativeAbundance)

sum(exp_co1_phyla_abundance$RelativeAbundance)
sum(exp_co1_actinopteri_elasmos_sp_abundance$RelativeAbundance)
sum(exp_co1_chordates_sp_abundance$RelativeAbundance)
sum(exp_co1_molluscs_sp_abundance$RelativeAbundance)
sum(exp_co1_arthropods_sp_abundance$RelativeAbundance)
```

### Visualize
Append the 5 datasets of actinopteri and elasmos for comparison
```{r}
commonnames <- read_csv(file = "../eDNA-databases/commonnames.csv")

most_abund_all_datasets <- exp_mifish_species_abundance %>%
  bind_rows(trawl_mifish_species_abundance) %>%
  bind_rows(exp_elas_species_abundance) %>%
  bind_rows(trawl_elas_species_abundance) %>% 
  bind_rows(exp_co1_actinopteri_elasmos_sp_abundance) %>% 
  select (-Color, -CommonName) # this formatting got messed up in some of the rows

commonnames_mod <- commonnames %>% select(-Taxon) %>% distinct()

# Add back in colors and common name
most_abund_all_datasets <- most_abund_all_datasets %>% 
  left_join(commonnames_mod, by = c("Species.CommonName" = "Taxon.CommonName"))


most_abund_all_datasets
```



Fix colors so they remain constant in following plots
Some [predefined color names](https://sape.inf.usi.ch/quick-reference/ggplot2/colour) for reference
I started a colors column in commonnames database to keep assignment consistent across notebooks, but they got lost from the df when aggregating. Join back in and format for plotting
```{r}

myColors_sciname <- setNames(commonnames$Color, commonnames$Taxon)
myColors_comname <- setNames(commonnames$Color, commonnames$CommonName)
myColors_taxcomname <- setNames(commonnames$Color, commonnames$Taxon.CommonName)

head(myColors_sciname)
head(myColors_comname)
head(myColors_taxcomname)
```

#### Top Species- Teleosts and Elasmos
For report: MiFish and Elas02 only
```{r}
# Plot a stacked barplot- 
top_species_abundance_plot <- ggplot(most_abund_all_datasets %>% filter(!library == "CO1"), aes(x = sample_type, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::percent_format()) + 
  scale_fill_manual(values = myColors_comname) +
  facet_grid(~library) +
  labs(x = NULL, y = "Relative Abundance (%)", title = "Relative Abundance of Top Species: Expedition vs. Manual Samples") +
  theme_minimal() +
  guides(fill = guide_legend(ncol = 2)) +  # Adjust # of legend columns
  theme(legend.title = element_blank())
  
  

top_species_abundance_plot

ggsave(plot = top_species_abundance_plot, filename = "figures-expedition/top_species_abundance_plot.jpg")

```



For presentation: MiFish and Elas02 alongside CO1
```{r}
# Plot a stacked barplot- 
top_species_abundance_plot_CO1 <- ggplot(most_abund_all_datasets, aes(x = sample_type, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::percent_format()) + 
  scale_fill_manual(values = myColors_comname) +
  facet_grid(~library, scales = "free_x", space = "free") +
  labs(x = NULL, y = "Relative Abundance (%)", title = "Relative Abundance of Top Species: Expedition vs. Manual Samples") +
  theme_minimal() +
  guides(fill = guide_legend(ncol = 2)) +  # Adjust # of legend columns
  theme(legend.title = element_blank())
  
  

top_species_abundance_plot_CO1

ggsave(plot = top_species_abundance_plot_CO1, filename = "figures-expedition/top_species_abundance_plot_with_CO1.jpg")

```

#### Top Phyla- CO1
```{r}
# Retain only most abundant sp
exp_co1_phyla_abundance_top <- exp_co1_phyla_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 15) # Show top 15 phyla


# Plot a stacked barplot- 
top_phyla_abundance_plot_CO1 <- ggplot(exp_co1_phyla_abundance_top, aes(x = sample_type, y = RelativeAbundance, fill = Taxon.CommonName)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::percent_format()) + 
  scale_fill_manual(values = myColors_taxcomname) +
  facet_grid(~library, scales = "free_x", space = "free") +
  labs(x = NULL, y = "Relative Abundance (%)", title = "Relative Abundance of Top Phyla: CO1") +
  theme_minimal() +
  guides(fill = guide_legend(ncol = 2)) +  # Adjust # of legend columns
  theme(legend.title = element_blank())
  
  

top_phyla_abundance_plot_CO1

ggsave(plot = top_phyla_abundance_plot_CO1, filename = "figures-expedition/top_phyla_abundance_plot_CO1.jpg")

```

#### Classes of Animals of Interest
```{r}
# Plot a stacked barplot- 
animals_plot_CO1 <- ggplot(exp_co1_animals_ofinterest_class, aes(x = sample_type, y = RelativeAbundance, fill = Taxon.CommonName)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::percent_format()) + 
  scale_fill_manual(values = myColors_taxcomname) +
  facet_grid(~library, scales = "free_x", space = "free") +
  labs(x = NULL, y = "Relative Abundance (%)", title = "CO1: Arthropods, Molluscs, Chordates, and Echinoderms") +
  theme_minimal() +
  guides(fill = guide_legend(ncol = 2)) +  # Adjust # of legend columns
  theme(legend.title = element_blank())

animals_plot_CO1

ggsave(plot = animals_plot_CO1, filename = "figures-expedition/animals_plot_CO1.jpg")

```



## Examine controls 

### Prepare data for positive controls
Extract positive controls and convert to dataframe
NOTE- there are not Elasmo pos controls in this dataset
```{r}
ps2024_exp_mifish_controls <- subset_samples(ps2024_exp_mifish, controls == "positive")
ps2024_exp_co1_controls <- subset_samples(ps2024_exp_co1, controls == "positive")
ps2024_trawl_mifish_controls <- subset_samples(ps2024_trawl_mifish, controls == "positive")

ps2024_exp_mifish_controls_long_df <- ps2024_exp_mifish_controls %>%
  psmelt() %>%
  as_tibble()

ps2024_exp_co1_controls_long_df <- ps2024_exp_co1_controls %>%
  psmelt() %>%
  as_tibble()

ps2024_trawl_mifish_controls_long_df <- ps2024_trawl_mifish_controls %>%
  psmelt() %>%
  as_tibble()

ps2024_exp_mifish_controls_long_df
ps2024_exp_co1_controls_long_df
ps2024_trawl_mifish_controls_long_df
```


#### Exp/MiFish-

Plot a stacked barplot- 
```{r}
# Sum by common name, calculate total abundance and relative abundance
ps2024_exp_mifish_controls_long_df_sum <- ps2024_exp_mifish_controls_long_df %>%
  group_by(CommonName, Name.Deploy_Cartr_Library) %>%
  summarise(TotalAbundance = sum(Abundance, na.rm = TRUE), .groups = "drop_last") %>%
  group_by(Name.Deploy_Cartr_Library) %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(TotalAbundance > 0) %>%
  arrange(desc(TotalAbundance))

# Pos Controls by read abundance
ps2024_exp_mifish_posctl_barplot <- ggplot(ps2024_exp_mifish_controls_long_df_sum, 
       aes(x = Name.Deploy_Cartr_Library, y = TotalAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_mifish_posctl_barplot

# Pos Controls by relative abundance
ps2024_exp_mifish_posctl_ra_barplot <- ggplot(ps2024_exp_mifish_controls_long_df_sum, 
       aes(x = Name.Deploy_Cartr_Library, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Relative Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_mifish_posctl_ra_barplot


ggsave(plot = ps2024_exp_mifish_posctl_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_barplot.jpg", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_ra_barplot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = ps2024_exp_mifish_posctl_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_barplot.eps", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_ra_barplot.eps", width = 4, height = 4, units = "in")

```

#### Exp/CO1-
(Note- there are no pos ctls for elasmobranchs in 2024)
Plot a stacked barplot- 
```{r}
# Sum by common name, calculate total abundance and relative abundance
ps2024_exp_co1_controls_long_df_sum <- ps2024_exp_co1_controls_long_df %>%
  group_by(CommonName, Name_Deploy_Cartr) %>%
  summarise(TotalAbundance = sum(Abundance, na.rm = TRUE), .groups = "drop_last") %>%
  filter(!is.na(CommonName)) %>%  # Remove NAs BEFORE calculating relative abundance
  group_by(Name_Deploy_Cartr) %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(TotalAbundance > 0) %>%
  arrange(desc(TotalAbundance))


# Pos Controls by read abundance
ps2024_exp_co1_posctl_barplot <- ggplot(ps2024_exp_co1_controls_long_df_sum, 
       aes(x = Name_Deploy_Cartr, y = TotalAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_co1_posctl_barplot

# Pos Controls by relative abundance
ps2024_exp_co1_posctl_ra_barplot <- ggplot(ps2024_exp_co1_controls_long_df_sum, 
       aes(x = Name_Deploy_Cartr, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Relative Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_co1_posctl_ra_barplot


ggsave(plot = ps2024_exp_co1_posctl_barplot, filename = "figures-expedition/2024_exp_co1_posctl_barplot.jpg", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_co1_posctl_ra_barplot, filename = "figures-expedition/2024_exp_co1_posctl_ra_barplot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = ps2024_exp_co1_posctl_barplot, filename = "figures-expedition/2024_exp_co1_posctl_barplot.eps", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_co1_posctl_ra_barplot, filename = "figures-expedition/2024_exp_co1_posctl_ra_barplot.eps", width = 4, height = 4, units = "in")

```


#### Trawl/MiFish-
(again, no Elasmo pos controls from hand samples)

Plot a stacked barplot- 
```{r}
# Sum by common name, calculate total abundance and relative abundance
ps2024_trawl_mifish_controls_long_df_sum <- ps2024_trawl_mifish_controls_long_df %>%
  group_by(CommonName, replicates) %>%
  summarise(TotalAbundance = sum(Abundance, na.rm = TRUE), .groups = "drop_last") %>%
  group_by(replicates) %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(TotalAbundance > 0) %>%
  arrange(desc(TotalAbundance))

# Pos Controls by read abundance
ps2024_trawl_mifish_posctl_barplot <- ggplot(ps2024_trawl_mifish_controls_long_df_sum, 
       aes(x = replicates, y = TotalAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 trawl, Pos. Controls: Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_trawl_mifish_posctl_barplot

# Pos Controls by relative abundance
ps2024_trawl_mifish_posctl_ra_barplot <- ggplot(ps2024_trawl_mifish_controls_long_df_sum, 
       aes(x = replicates, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 trawl, Pos. Controls: Relative Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_trawl_mifish_posctl_ra_barplot


ggsave(plot = ps2024_trawl_mifish_posctl_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_barplot.jpg", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_trawl_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_ra_barplot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = ps2024_trawl_mifish_posctl_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_barplot.eps", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_trawl_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_ra_barplot.eps", width = 4, height = 4, units = "in")

```


### Remove Pos and Neg  Controls from all phyloseq objects:
(Note, they were already removed from Elas objects just based on quality filtering above)

ASV-based, reads:
ps2024_exp_mifish

ASV-based, relative abundance:
ps2024_exp_mifish_ra

Taxonomy (species)- based, reads:
ps2024_exp_mifish_glommed


Taxonomy (species)- based, relative abundance:
ps2024_exp_mifish_glommed_ra


```{r}
ps2024_exp_mifish <- subset_samples(ps2024_exp_mifish, is.na(controls))
ps2024_exp_mifish_ra <- subset_samples(ps2024_exp_mifish_ra, is.na(controls))
ps2024_exp_mifish_glommed <- subset_samples(ps2024_exp_mifish_glommed, is.na(controls))
ps2024_exp_mifish_glommed_ra <- subset_samples(ps2024_exp_mifish_glommed_ra, is.na(controls))

ps2024_exp_elas <- subset_samples(ps2024_exp_elas, is.na(controls))
ps2024_exp_elas_ra <- subset_samples(ps2024_exp_elas_ra, is.na(controls))
ps2024_exp_elas_glommed <- subset_samples(ps2024_exp_elas_glommed, is.na(controls))
ps2024_exp_elas_glommed_ra <- subset_samples(ps2024_exp_elas_glommed_ra, is.na(controls))

ps2024_exp_co1 <- subset_samples(ps2024_exp_co1, is.na(controls))
ps2024_exp_co1_ra <- subset_samples(ps2024_exp_co1_ra, is.na(controls))
ps2024_exp_co1_glommed <- subset_samples(ps2024_exp_co1_glommed, is.na(controls))
ps2024_exp_co1_glommed_ra <- subset_samples(ps2024_exp_co1_glommed_ra, is.na(controls))

ps2024_exp_co1_actinopteri_elasmos <- subset_samples(ps2024_exp_co1_actinopteri_elasmos, is.na(controls))
ps2024_exp_co1_arthropods <- subset_samples(ps2024_exp_co1_arthropods, is.na(controls))
ps2024_exp_co1_chordates <- subset_samples(ps2024_exp_co1_chordates, is.na(controls))
ps2024_exp_co1_molluscs <- subset_samples(ps2024_exp_co1_molluscs, is.na(controls))

ps2024_trawl_mifish <- subset_samples(ps2024_trawl_mifish, is.na(controls))
ps2024_trawl_mifish_ra <- subset_samples(ps2024_trawl_mifish_ra, is.na(controls))
ps2024_trawl_mifish_glommed <- subset_samples(ps2024_trawl_mifish_glommed, is.na(controls))
ps2024_trawl_mifish_glommed_ra <- subset_samples(ps2024_trawl_mifish_glommed_ra, is.na(controls))

ps2024_trawl_elas <- subset_samples(ps2024_trawl_elas, is.na(controls))
ps2024_trawl_elas_ra <- subset_samples(ps2024_trawl_elas_ra, is.na(controls))
ps2024_trawl_elas_glommed <- subset_samples(ps2024_trawl_elas_glommed, is.na(controls))
ps2024_trawl_elas_glommed_ra <- subset_samples(ps2024_trawl_elas_glommed_ra, is.na(controls))

```

#### Save

```{r}
save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_posctlanalysis.RData")
```

or load back in
```{r}
load(file = "figures-expedition/exp_ecol_analysis_environment_upto_posctlanalysis.RData")
```



## Bubble plots

Based on my previous [scripts](https://github.com/lizsuter/Cariaco_Euk)

Work with tax-based dataframes (rather than ASV-based):

Taxonomy (species)- based, reads:
ps2024_exp_mifish_glommed

Taxonomy (species)- based, relative abundance:
ps2024_exp_mifish_glommed_ra

### Prepare data
#### Expedition/ MiFish
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_mifish_glommed_df <- psmelt(ps2024_exp_mifish_glommed)
ps2024_exp_mifish_glommed_ra_df <- psmelt(ps2024_exp_mifish_glommed_ra)

# replace zeroes in the table with NA
ps2024_exp_mifish_glommed_df[ps2024_exp_mifish_glommed_df == 0] <- NA
ps2024_exp_mifish_glommed_ra_df[ps2024_exp_mifish_glommed_ra_df == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_mifish_glommed_df <-  filter(ps2024_exp_mifish_glommed_df, !is.na(Abundance))
ps2024_exp_mifish_glommed_ra_df <-  filter(ps2024_exp_mifish_glommed_ra_df, !is.na(Abundance))


# and view
ps2024_exp_mifish_glommed_df
ps2024_exp_mifish_glommed_ra_df
```

#### Expedition/ Elas02-
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_elas_glommed_df <- psmelt(ps2024_exp_elas_glommed)
ps2024_exp_elas_glommed_ra_df <- psmelt(ps2024_exp_elas_glommed_ra)

# replace zeroes in the table with NA
ps2024_exp_elas_glommed_df[ps2024_exp_elas_glommed_df == 0] <- NA
ps2024_exp_elas_glommed_ra_df[ps2024_exp_elas_glommed_ra_df == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_elas_glommed_df <-  filter(ps2024_exp_elas_glommed_df, !is.na(Abundance))
ps2024_exp_elas_glommed_ra_df <-  filter(ps2024_exp_elas_glommed_ra_df, !is.na(Abundance))

# and view
ps2024_exp_elas_glommed_df
ps2024_exp_elas_glommed_ra_df
```

#### Expedition/ CO1- 
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_co1_glommed_df <- psmelt(ps2024_exp_co1_glommed)

# replace zeroes in the table with NA
ps2024_exp_co1_glommed_df[ps2024_exp_co1_glommed_df == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_co1_glommed_df <-  filter(ps2024_exp_co1_glommed_df, !is.na(Abundance))


# and view
ps2024_exp_co1_glommed_df
```

CO1 Species
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_co1_glommed_df_species <- psmelt(ps2024_exp_co1_glommed)

# replace zeroes in the table with NA
ps2024_exp_co1_glommed_df_species[ps2024_exp_co1_glommed_df_species == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_co1_glommed_df_species <-  filter(ps2024_exp_co1_glommed_df_species, !is.na(Abundance))


# and view
ps2024_exp_co1_glommed_df_species
```


CO1 Actinopteri and Elasmobranchs only 
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_co1_glommed_df_actinopteri_elasmos <- psmelt(ps2024_exp_co1_actinopteri_elasmos)

# replace zeroes in the table with NA
ps2024_exp_co1_glommed_df_actinopteri_elasmos[ps2024_exp_co1_glommed_df_actinopteri_elasmos == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_co1_glommed_df_actinopteri_elasmos <-  filter(ps2024_exp_co1_glommed_df_actinopteri_elasmos, !is.na(Abundance))


# and view
ps2024_exp_co1_glommed_df_actinopteri_elasmos
```

CO1 Chordates only 
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_co1_glommed_df_chordates <- psmelt(ps2024_exp_co1_chordates)

# replace zeroes in the table with NA
ps2024_exp_co1_glommed_df_chordates[ps2024_exp_co1_glommed_df_chordates == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_co1_glommed_df_chordates <-  filter(ps2024_exp_co1_glommed_df_chordates, !is.na(Abundance))


# and view
ps2024_exp_co1_glommed_df_chordates
```

CO1 Molluscs only 
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_co1_glommed_df_molluscs <- psmelt(ps2024_exp_co1_chordates)

# replace zeroes in the table with NA
ps2024_exp_co1_glommed_df_chordates[ps2024_exp_co1_glommed_df_chordates == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_co1_glommed_df_chordates <-  filter(ps2024_exp_co1_glommed_df_chordates, !is.na(Abundance))


# and view
ps2024_exp_co1_glommed_df_chordates

```


### Plot by sites

#### MiFish-

```{r}

# First sort the site names in a sensible way
levels(unique(c(ps2024_exp_mifish_glommed_df$sites)))
sitelevels <- as.character(sort(as.numeric(levels(unique(c(ps2024_exp_mifish_glommed_df$sites))))))
sitelevels

# Order E-W in sensible way by making it a factor
ps2024_exp_mifish_glommed_df$Bayside_f= factor(ps2024_exp_mifish_glommed_df$Bayside, levels=c('W','E'))

```



##### By bayside
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_bayside <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Bayside_f, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_bayside


bubbleplot_comname_2024_expedition_mifish_Bayside <- ggplot(ps2024_exp_mifish_glommed_df_bayside, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_Bayside


  
  
# save
ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Bayside.eps", width = 9.5, height = 10, units = "in")

ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Bayside.jpg", width = 9.5, height = 10, units = "in")
```

##### By Day/Night
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_daynight <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Night__Day, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_daynight


bubbleplot_comname_2024_expedition_mifish_DayNight <- ggplot(ps2024_exp_mifish_glommed_df_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Night__Day, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_DayNight


ggsave(plot = bubbleplot_comname_2024_expedition_mifish_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_DayNight.eps", width = 12, height = 7, units = "in")
```

##### By Habitat
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_habitat <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Habitat, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_habitat


bubbleplot_comname_2024_expedition_mifish_Habitat <- ggplot(ps2024_exp_mifish_glommed_df_habitat, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Habitat, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_Habitat

ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Habitat, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Habitat.eps", width = 12, height = 7, units = "in")
```
##### By Bayside and Day-Night
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_bayside_daynight <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Bayside_f, Night__Day, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_bayside_daynight


bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight <- ggplot(ps2024_exp_mifish_glommed_df_bayside_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(Night__Day~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight




ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight.eps", width = 10, height = 9, units = "in")

```





#### Elas02-

```{r}

# First sort the site names in a sensible way
levels(unique(c(ps2024_exp_elas_glommed_df$sites)))
sitelevels <- as.character(sort(as.numeric(levels(unique(c(ps2024_exp_elas_glommed_df$sites))))))
sitelevels

# Order E-W in sensible way by making it a factor
ps2024_exp_elas_glommed_df$Bayside_f= factor(ps2024_exp_elas_glommed_df$Bayside, levels=c('W','E'))

```



##### By bayside
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_bayside <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Bayside_f, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_bayside



bubbleplot_comname_2024_expedition_elas_Bayside <- ggplot(ps2024_exp_elas_glommed_df_bayside, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_Bayside

# save
ggsave(plot = bubbleplot_comname_2024_expedition_elas_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Bayside.eps", width = 8, height = 2.5, units = "in")

ggsave(plot = bubbleplot_comname_2024_expedition_elas_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Bayside.jpg", width = 8, height = 2.5, units = "in")
```

##### By DayNight
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_daynight <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Night__Day, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_daynight


bubbleplot_comname_2024_expedition_elas_DayNight <- ggplot(ps2024_exp_elas_glommed_df_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Night__Day, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_DayNight


ggsave(plot = bubbleplot_comname_2024_expedition_elas_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_DayNight.eps", width = 10, height = 2.5, units = "in")
```

##### By Habitat
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_habitat <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Habitat, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_habitat


bubbleplot_comname_2024_expedition_elas_Habitat <- ggplot(ps2024_exp_elas_glommed_df_habitat, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Habitat, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_Habitat

ggsave(plot = bubbleplot_comname_2024_expedition_elas_Habitat, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Habitat.eps", width = 10, height = 2.5, units = "in")
```
##### By Bayside and Day-Night
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_bayside_daynight <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Bayside_f, Night__Day, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_bayside_daynight


bubbleplot_comname_2024_expedition_elas_Bayside_DayNight <- ggplot(ps2024_exp_elas_glommed_df_bayside_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(Night__Day~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_Bayside_DayNight




ggsave(plot = bubbleplot_comname_2024_expedition_elas_Bayside_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Bayside_DayNight.eps", width = 10, height = 4, units = "in")

```

#### CO1

```{r}
# First sort the site names in a sensible way
levels(unique(c(ps2024_exp_co1_glommed_df$sites)))
sitelevels <- as.character(sort(as.numeric(levels(unique(c(ps2024_exp_co1_glommed_df$sites))))))
sitelevels

# Order E-W in sensible way by making it a factor
ps2024_exp_co1_glommed_df$Bayside_f= factor(ps2024_exp_co1_glommed_df$Bayside, levels=c('W','E'))

```


##### By bayside
```{r}
# First average replicates at same sites
ps2024_exp_co1_glommed_df_phyla_bayside <- ps2024_exp_co1_glommed_df %>%
    group_by(sites, Bayside_f, Phylum) %>%
    dplyr::summarize(Abundance = mean(Abundance)) %>% 
  left_join(commonnames, by = c("Phylum" = "Taxon"))


bubbleplot_comname_2024_expedition_co1_Bayside <- ggplot(ps2024_exp_co1_glommed_df_phyla_bayside, aes(x = factor(sites, levels = sitelevels), y = fct_rev(Taxon.CommonName))) + 
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = Taxon.CommonName), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_taxcomname)+
   theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 11), axis.text.y = element_text(size = 11), legend.position = "bottom") +
  guides(fill="none") +
  facet_grid(~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_co1_Bayside

# save
ggsave(plot = bubbleplot_comname_2024_expedition_co1_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_co1_Bayside.eps", width = 9, height = 9.5, units = "in")

ggsave(plot = bubbleplot_comname_2024_expedition_co1_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_co1_Bayside.jpg", width = 9, height = 9.5, units = "in")
```



## Krona plots
Re-create Krona plots using cleaned up datasets--

Following [this](https://github.com/markschl/embed_krona/blob/main/example.Rmd)
Compatible with phyloseq objects
Must run in the console:
```{r}
source('https://raw.githubusercontent.com/markschl/embed_krona/master/embed_krona.R')
plot_krona(ps2024_exp_mifish_glommed)
plot_krona(ps2024_exp_elas_glommed)
plot_krona(ps2024_exp_co1_glommed)
 
```

Manually changed the names in folder to 
Generates the [MiFish krona plot](krona_files/krona_1.html) and [Elas02 krona plot](krona_files/krona_2.html) as an html file in current working directory. 


#### Save

```{r}
save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_kronaplots.RData")
```

or load back in
```{r}
load(file = "figures-expedition/exp_ecol_analysis_environment_upto_kronaplots.RData")
```

## Summary stats
### CO1
#### Total species, genera, etc.
```{r}
paste("Unique Species (includes unknown sp with assigned genus, family, etc)")
length(unique(ps2024_exp_co1_glommed_df$Species))

paste("Unique Species (no, unknowns)")
ps2024_exp_co1_glommed_df %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique Genera (includes unknown sp with assigned family, order, etc)")
length(unique(ps2024_exp_co1_glommed_df$Genus))

paste("Unique Genera (no, unknowns)")
ps2024_exp_co1_glommed_df %>%
  filter(!str_detect(Genus, "Unknown")) %>%
  select(Genus) %>%
  unique() %>%
  dim()

paste("Unique Families (includes unknown sp with assigned order, class, etc)")
length(unique(ps2024_exp_co1_glommed_df$Family))

paste("Unique Families (no, unknowns)")
ps2024_exp_co1_glommed_df %>%
  filter(!str_detect(Family, "Unknown")) %>%
  select(Family) %>%
  unique() %>%
  dim()

paste("Unique Orders (includes unknowns)")
length(unique(ps2024_exp_co1_glommed_df$Order))

paste("Unique Orders (no, unknowns)")
ps2024_exp_co1_glommed_df %>%
  filter(!str_detect(Order, "Unknown")) %>%
  select(Order) %>%
  unique() %>%
  dim()

paste("Unique Classes (includes unknowns)")
length(unique(ps2024_exp_co1_glommed_df$Family))

paste("Unique Classes (no, unknowns)")
ps2024_exp_co1_glommed_df %>%
  filter(!str_detect(Class, "Unknown")) %>%
  select(Class) %>%
  unique() %>%
  dim()


paste("Unique Phyla (includes unknowns)")
length(unique(ps2024_exp_co1_glommed_df$Phylum))

paste("Unique Phyla (no, unknowns)")
ps2024_exp_co1_glommed_df %>%
  filter(!str_detect(Phylum, "Unknown")) %>%
  select(Phylum) %>%
  unique() %>%
  dim()

```

#### Inverts
Invertebrate Animal Phyla:

Arthropoda: Includes insects, spiders, crustaceans, and others. They are characterized by exoskeletons and segmented bodies.
Cnidaria: Known for jellyfish, corals, sea anemones, etc., distinguished by their cnidocytes.
Annelida: Segment worms like earthworms and leeches fall under this phylum.
Ctenophora: Also known as comb jellies, they are marine and gelatinous, similar to cnidarians but distinct.
Echinodermata: Sea stars, sea urchins, and their relatives. They are known for their radially symmetrical body plans (in the adult form) and a unique water vascular system.
Rotifera: Rotifers are microscopic, mostly aquatic organisms.
Nemertea: Ribbon worms, which are notable for their proboscis.
Mollusca: This diverse phylum includes snails, slugs, clams, octopuses, and squids, characterized by a muscular foot and mantle.
Porifera: Sponges, simple animals with porous bodies through which water flows.
Platyhelminthes: Flatworms, including both free-living and parasitic varieties.
Bryozoa: Moss animals, small colonial animals mostly found in marine environments.
Onychophora: Velvet worms, small worm-like creatures with numerous stubby legs.
Nematoda: Roundworms, which are ubiquitous in both aquatic and terrestrial environments.
Gastrotricha: Microscopic, worm-like animals mainly found in freshwater and marine environments.
Tardigrada: Water bears or moss piglets, known for their incredible resilience.
Xenacoelomorpha: A recently recognized phylum of simple marine invertebrates.

```{r}
invertebrate_list <- c(
  "Arthropoda",
  "Cnidaria", 
  "Annelida", 
  "Ctenophora", 
  "Echinodermata", 
  "Rotifera", 
  "Nemertea", 
  "Mollusca", 
  "Porifera", 
  "Platyhelminthes", 
  "Bryozoa", 
  "Onychophora", 
  "Nematoda", 
  "Gastrotricha", 
  "Tardigrada", 
  "Xenacoelomorpha", 
  "Entoprocta",
  "Placozoa")
```


```{r}
paste("Unique invertebrates, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Phylum %in% invertebrate_list) %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique invertebrates, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Phylum %in% invertebrate_list) %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()
```
#### Chordates
```{r}
paste("Unique vertebrates, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Phylum == "Chordata") %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique vertebrates, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Phylum == "Chordata") %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()
```
Who are the chordates
```{r}
ps2024_exp_co1_glommed_df %>%
  filter(Phylum == "Chordata") %>%
  select(Class) %>%
  unique()
```


Sea squirts, tunicates, fish, birds, larvaceans, elasmobranchs, amphibians, reptiles. 
Count the fish, birds, elasmos, amphibians, reptiles....
```{r}
paste("Unique teleost fish, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Actinopteri") %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique teleost fish, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Actinopteri") %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique elasmobranchs, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Chondrichthyes") %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique elasmobranchs, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Chondrichthyes") %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique birds, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Aves") %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique birds, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Aves") %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()


paste("Unique amphibians, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Amphibia") %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique amphibians, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Amphibia") %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique reptiles, including unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Lepidosauria") %>%
  select(Species) %>%
  unique() %>%
  dim()

paste("Unique reptiles, no unknowns")
ps2024_exp_co1_glommed_df %>%
  filter(Class == "Lepidosauria") %>%
  filter(!str_detect(Species, "Unknown")) %>%
  select(Species) %>%
  unique() %>%
  dim()


```



## Maps
Import data and draw map (see CTD notebook for more details)
```{r}
# lets get a box to make a smaller sized objects and facilitate the examples:
geo.box <- c(xmin=-72.61, xmax=-72.4, ymin=40.79, ymax=40.9)

sf_use_s2(FALSE)

# Shoreline data for us detail
gshhg.l1 <- sf::read_sf("raw_data/gshhg-shp-2.3.7/GSHHS_shp/f/GSHHS_f_L1.shp") %>% st_crop(geo.box)
plot(gshhg.l1["id"])

# make ggplot object
shinnbay_map <- ggplot(data = gshhg.l1) +
  geom_sf(aes(geometry = geometry)) +
    scale_x_continuous(breaks = seq(-72.6, -72.4, 0.1)) +
    scale_y_continuous(breaks = seq(40.80, 40.9, 0.05))

shinnbay_map

```

### MiFish
#### Prepare data
Group MiFish species abundance by station
```{r}
# Average replicates at same sites
ps2024_exp_mifish_glommed_df_bystation <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Species.CommonName) %>%
    dplyr::summarize(Abundance = mean(Abundance)) %>% 
  left_join(commonnames_mod, by = c("Species.CommonName" = "Taxon.CommonName"))

# Calculate relative abundances across the whole dataset
ps2024_exp_mifish_glommed_df_bystation <- ps2024_exp_mifish_glommed_df_bystation %>%
  group_by(sites) %>%
  mutate(RelativeAbundance = Abundance / sum(Abundance))


# Calculate relative abundances across the whole dataset
exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))


# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
ps2024_exp_mifish_glommed_df_bystation <- ps2024_exp_mifish_glommed_df_bystation %>%
  left_join(stationdata, by = c("sites" = "sites"))


ps2024_exp_mifish_glommed_df_bystation

```


#### Plot
Plot only top species, same as those from `trawl_mifish_species_abundance`
```{r}
# First create a dummy plot to extract the legend
legend_plot <- ggplot(ps2024_exp_mifish_glommed_df_bystation %>% filter(Species.CommonName %in% exp_mifish_species_abundance$Species.CommonName),
                      aes(x = sites, y = RelativeAbundance, fill = Species.CommonName)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = myColors_taxcomname) +
  theme_minimal() +
  theme(legend.position = "right",  #right is the only one that works later with get_legend
        legend.title = element_blank()) + 
  guides(fill = guide_legend(ncol = 2))


# Create a list of mini barplots for each site
mini_barplots_mifish <- ps2024_exp_mifish_glommed_df_bystation %>%
  group_by(sites) %>%
  filter(Species.CommonName %in% exp_mifish_species_abundance$Species.CommonName) %>%
  group_split() %>%
  setNames(unique(ps2024_exp_mifish_glommed_df_bystation$sites)) %>%
  lapply(function(df_site) {
    ggplot(df_site, aes(x = sites, y = Abundance, fill = Species.CommonName)) +
      geom_bar(stat = "identity") +
      theme_void() +
      theme(legend.position = "none") +
      scale_fill_manual(values = myColors_taxcomname)})

# Prepare a summary table of one row per site with lat/long and the mini-plot
site_coords <- ps2024_exp_mifish_glommed_df_bystation %>%
  group_by(sites, lat, long) %>%
  summarise(.groups = "drop") %>%
  mutate(plot = mini_barplots_mifish[as.character(sites)])

# Grab legend from top species plot
fish_legend <- cowplot::get_legend(legend_plot)
cowplot::ggdraw(fish_legend)

# Add to base map
map_with_subplots <- shinnbay_map +
  ggimage::geom_subview(
    data = site_coords,
    aes(x = long, y = lat, subview = plot),
    width = 0.0085, height = 0.0085  # adjust size as needed
  )

# 4. Combine map and legend
mifish_map <- cowplot::plot_grid(
  map_with_subplots,
  fish_legend,
  ncol = 1,
  rel_heights = c(.9, 0.3)  # adjust the space for legend
)

mifish_map

  
ggsave("figures-expedition/mifish_expedition_topspecies_map.eps", plot = mifish_map, width = 8, height = 6, units = "in")
ggsave("figures-expedition/mifish_expedition_topspecies_map.jpg", plot = mifish_map, width = 8, height = 6, units = "in")

```



### Aureococcus
#### Prepare data
```{r}
# Calculate relative abundance of Aureococcus anophagefferens per Site
aureo_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            AureococcusCounts = sum(Species == "Aureococcus anophagefferens")) %>%
  mutate(RelativeAbundance = AureococcusCounts / TotalCounts) %>%
  select(-TotalCounts, -AureococcusCounts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot


# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
aureo_relative_abundance_df <- aureo_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


aureo_relative_abundance_df

```

#### Plot Aureococcus map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_aureococcus <- shinnbay_map +
  geom_point(data = aureo_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "brown") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, " * italic("Aureococcus anophagefferens") * " CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_aureococcus



ggsave("figures-expedition/aureococcus_map.eps", plot = shinn_bay_aureococcus, width = 8, height = 6, units = "in")
ggsave("figures-expedition/aureococcus_map.jpg", plot = shinn_bay_aureococcus, width = 8, height = 6, units = "in")

```


### Bivalves
#### Prepare data
```{r}
# Calculate relative abundance per Site
bivalves_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Class == "Bivalvia")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot


# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
bivalves_relative_abundance_df <- bivalves_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


bivalves_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_bivalves <- shinnbay_map +
  geom_point(data = bivalves_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "grey2") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, " * italic("Bivalve") * " CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_bivalves



ggsave("figures-expedition/bivalves_map.eps", plot = shinn_bay_bivalves, width = 8, height = 6, units = "in")
ggsave("figures-expedition/bivalves_map.jpg", plot = shinn_bay_bivalves, width = 8, height = 6, units = "in")

```


### Eastern oyster
#### Prepare data
```{r}
# Calculate relative abundance  per Site
cvirginica_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            CvirginicaCounts = sum(Species == "Crassostrea virginica")) %>%
  mutate(RelativeAbundance = CvirginicaCounts / TotalCounts) %>%
  select(-TotalCounts, -CvirginicaCounts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot


# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
cvirginica_relative_abundance_df <- cvirginica_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


cvirginica_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_cvirginica <- shinnbay_map +
  geom_point(data = cvirginica_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "grey2") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, " * italic("C. virginica") * " CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_cvirginica



ggsave("figures-expedition/cvirginica_map.eps", plot = shinn_bay_cvirginica, width = 8, height = 6, units = "in")
ggsave("figures-expedition/cvirginica_map.jpg", plot = shinn_bay_cvirginica, width = 8, height = 6, units = "in")

```





### Hard clam
#### Prepare data
```{r}
# Calculate relative abundance per Site
hardclam_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Mercenaria mercenaria")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot


# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
hardclam_relative_abundance_df <- hardclam_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


hardclam_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_hardclam <- shinnbay_map +
  geom_point(data = hardclam_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "grey") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, " * italic("M. mercenaria") * ", CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_hardclam



ggsave("figures-expedition/hardclam_map.eps", plot = shinn_bay_hardclam, width = 8, height = 6, units = "in")
ggsave("figures-expedition/hardclam_map.jpg", plot = shinn_bay_hardclam, width = 8, height = 6, units = "in")

```


### Other bivalves
#### Prepare data
```{r}
# Calculate relative abundance per Site
otherbivalves_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  filter(Class == "Bivalvia", 
         !Species %in% c("Mercenaria mercenaria", "Crassostrea virginica")) %>%
  group_by(sites) %>%
  summarise(TotalCounts = n()) %>%
  left_join(ps2024_exp_co1_glommed_df %>%
              group_by(sites) %>%
              summarise(TotalSpeciesCount = n()), by = "sites") %>%
  mutate(RelativeAbundance = TotalCounts / TotalSpeciesCount) %>%
  select(sites, RelativeAbundance) %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance) * 100) %>%
  filter(AverageRelativeAbundancePercent > 0)


# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
otherbivalves_relative_abundance_df <- otherbivalves_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


otherbivalves_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_otherbivalves <- shinnbay_map +
  geom_point(data = otherbivalves_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "grey3") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, other " * italic("Bivalvia") * ", CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_otherbivalves



# what are they? 
otherbivalveslist <- ps2024_exp_co1_glommed_df %>%
  filter(Class == "Bivalvia", 
         !Species %in% c("Mercenaria mercenaria", "Crassostrea virginica")) %>%
  distinct(Species) %>%
  filter(!str_starts(Species, "Unknown")) %>%
  arrange(Species) %>%
  bind_rows(tibble(Species = "Unknown bivalves")) 
otherbivalveslist



ggsave("figures-expedition/otherbivalves_map.eps", plot = shinn_bay_otherbivalves, width = 8, height = 6, units = "in")
ggsave("figures-expedition/otherbivalves_map.jpg", plot = shinn_bay_otherbivalves, width = 8, height = 6, units = "in")

# also save list of other bivalve names
write.csv(otherbivalveslist, "figures-expedition/otherbivalves_names.csv", row.names = FALSE)



```


### Eelgrass wasting diseases (EWD)

See [review](https://pmc.ncbi.nlm.nih.gov/articles/PMC6445351/#:~:text=Human%2Dinduced%20global%20change%20is,under%20future%20global%20change%20scenarios.) and specifically table 1.

- No hits to Labyrinthula zosterae, the known agent of eelgrass wasting disease, nor to anything in the class Labyrinthulae.
- No hits to Phytomyxea, the class containing pathogenic Plasmodiophora species.
- There are many hits to Pythiales, an order of water molds in the Oomycetes that contains potential pathogens (including Halophytophthora. Note there are no hits to Phytophthora species or Halophytophthora zostera specifically so these are POTENTIALLY pathogens).

#### Prepare data
```{r}
# Calculate relative abundance  per Site
EWD_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Order == "Pythiales")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
EWD_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
EWD_relative_abundance_df <- EWD_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


EWD_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_EWD <- shinnbay_map +
  geom_point(data = EWD_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "green3") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, " * italic("Pythiales") * ", Potential Eelgrass Wasting Disease, CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_EWD



ggsave("figures-expedition/eelgrassdisease_map.eps", plot = shinn_bay_EWD, width = 8, height = 6, units = "in")
ggsave("figures-expedition/eelgrassdisease_map.jpg", plot = shinn_bay_EWD, width = 8, height = 6, units = "in")

```



### Invasives: Gracilaria vermiculophylla
From [source](https://www.deq.nc.gov/energy-mineral-and-land-resources/land-resources/publications/technical-paper-series/tech-6-invasive-species-gracilaria-vermiculophylla/download):

Gracilaria vermiculophylla is a highly branched rhodophyte, or red algae, indigenous to the coast of the Pacific Northwest. It easily recruits to hard substrates such as oyster and other shell material and fragmented specimens recover well and travel easily. This exotic is thought to have arrived to the U.S. East Coast near the Chesapeake Bay via Asian oysters imported into Virginia, or through ballast water transport. G. vermiculophylla is now considered an established invasive species along both coasts of North America and parts of central Europe.

#### Prepare data
```{r}
# Calculate relative abundance  per Site
Gvermiculophylla_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Gracilaria vermiculophylla")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Gvermiculophylla_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Gvermiculophylla_relative_abundance_df <- Gvermiculophylla_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Gvermiculophylla_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_Gvermiculophylla <- shinnbay_map +
  geom_point(data = Gvermiculophylla_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "red4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Invasive " * italic("Gracilaria vermiculophylla") * ", CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Gvermiculophylla



ggsave("figures-expedition/Gvermiculophylla_map.eps", plot = shinn_bay_Gvermiculophylla, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Gvermiculophylla_map.jpg", plot = shinn_bay_Gvermiculophylla, width = 8, height = 6, units = "in")

```



### Invasives: Botryllus schlosseri

From [source](https://www.jstor.org/stable/pdf/24866162.pdf):
According to Van Name (1945), Botryllus schlosseri was probably introduced to the east coast of North America prior to the 1830s, presumably through European shipping traffic.

From [source](https://invasions.si.edu/nemesis/species_summary/159373#:~:text=In%20the%20lower%20Chesapeake%20Bay,where%20it%20can%20grow%20rapidly.): The origin of the Golden Star Tunicate, Botryllus schlosseri, is uncertain. It is globally widespread and found on the temperate coasts of Europe, Asia, both sides of North America, Chile, Argentina, South Africa, Australia and oceanic islands such as Bermuda, the Azores and New Zealand. It is common in fouling communities and has likely spread globally through shipping, oyster culture and other aquaculture transfers. In the lower Chesapeake Bay, B. schlosseri is a fouling pest impacting oyster aquaculture, but not natural oyster beds. In some areas of its introduced range, there is concern that B. schlosseri competes with native species for space, especially on artificial substrates where it can grow rapidly.


#### Prepare data
```{r}
# Calculate relative abundance  per Site
Bschlosseri_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Botryllus schlosseri")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Bschlosseri_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Bschlosseri_relative_abundance_df <- Bschlosseri_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Bschlosseri_relative_abundance_df

```

#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_Bschlosseri <- shinnbay_map +
  geom_point(data = Bschlosseri_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "gold2") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(0.25, 12)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Invasive " * italic("Botryllus schlosseri") * ", CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Bschlosseri



ggsave("figures-expedition/Bschlosseri_map.eps", plot = shinn_bay_Bschlosseri, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Bschlosseri_map.jpg", plot = shinn_bay_Bschlosseri, width = 8, height = 6, units = "in")

```


### Invasives: Didemnum vexillum

From [source](https://invasions.si.edu/nemesis/species_summary/-334):
Didemnum vexillum is a rapidly spreading colonial tunicate that overgrows rocks, shellfish, and other organisms (e.g. sponges, hydroids, tunicates, algae). It probably originated from the Northwest Pacific, possibly Japan, but has been reported in several parts of the world including New Zealand, North America, and Europe. Its identification in these introduced areas had been debated, but recent genetic and morphological studies confirmed that many of these global populations are D. vexillum. Its first known occurrence on the East Coast of the US was in Damariscotta River estuary, Maine in 1982. It is now common from Shinnecock Inlet, New York to Eastport, Maine. 


#### Prepare data
```{r}
# Calculate relative abundance  per Site
Dvexillum_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Didemnum vexillum")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Dvexillum_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Dvexillum_relative_abundance_df <- Dvexillum_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Dvexillum_relative_abundance_df

```


#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_Dvexillum <- shinnbay_map +
  geom_point(data = Dvexillum_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "gold4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Invasive " * italic("Didemnum vexillum") * ", CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Dvexillum



ggsave("figures-expedition/Dvexillum_map.eps", plot = shinn_bay_Dvexillum, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Dvexillum_map.jpg", plot = shinn_bay_Dvexillum, width = 8, height = 6, units = "in")

```



### Invasives: Carcinus maenas (Green crab)

From [source](https://dec.ny.gov/nature/animals-fish-plants/green-crab): Predation and Competition: In New York, green crabs consume native bivalves (e.g., blue mussels) and other invertebrates and compete with native crabs for food and habitat. However, the relative ecological impact of green crabs in New York has declined due to the arrival of another non-native species, the shore crab (Hemigraspus sanguineus). Since the arrival of the shore crab in the 1990s, shore crabs have partially displaced green crabs in rocky intertidal habitats.

#### Prepare data
```{r}
# Calculate relative abundance  per Site
Cmaenas_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Carcinus maenas")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Cmaenas_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Cmaenas_relative_abundance_df <- Cmaenas_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Cmaenas_relative_abundance_df

```




#### Plot map
```{r}
# Assuming shinn_bay is a ggplot object, add points for relative abundance
shinn_bay_Cmaenas <- shinnbay_map +
  geom_point(data = Cmaenas_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "green4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Invasive " * italic("Carcinus maenas ") * "(Green crab), CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Cmaenas



ggsave("figures-expedition/Cmaenas_map.eps", plot = shinn_bay_Cmaenas, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Cmaenas_map.jpg", plot = shinn_bay_Cmaenas, width = 8, height = 6, units = "in")

```


### Other detected invasives (skip plotting for now):

- Styela canopus: detected but "no economic or ecological impacts have been documented for this species." [Source](https://invasions.si.edu/nemesis/species_summary/-101)
- Mytilus trossulus: Pacific blue mussel, native range is northern Pacific, California to Alaska. It is not native however found in many places and forms hybrids with native species, M. edulis, so this could also be a misidentification(?). Just being cautious



### Endangered: Isurus oxyrinchus (Shortfin mako shark)

From [source](https://www.iucnredlist.org/species/39341/2903170): 

#### Prepare data
```{r}
# Calculate relative abundance  per Site
Ioxyrinchus_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Isurus oxyrinchus")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Ioxyrinchus_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Ioxyrinchus_relative_abundance_df <- Ioxyrinchus_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Ioxyrinchus_relative_abundance_df

```

also detected by Elas02, pull that out as well

```{r}
# Calculate relative abundance  per Site
Ioxyrinchus_relative_abundance_df_elas <- ps2024_exp_elas_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Isurus oxyrinchus")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Ioxyrinchus_relative_abundance_df_elas

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Ioxyrinchus_relative_abundance_df_elas <- Ioxyrinchus_relative_abundance_df_elas %>%
  left_join(stationdata, by = c("sites" = "sites"))


Ioxyrinchus_relative_abundance_df_elas

```


#### Plot map
Both CO1 and Elas02 for comparison
```{r}
shinn_bay_Ioxyrinchus <- shinnbay_map +
  geom_point(data = Ioxyrinchus_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "blue4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Endangered " * italic("Isurus oxyrinchus ") * "(Shortfin mako shark), CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Ioxyrinchus


shinn_bay_Ioxyrinchus_elas <- shinnbay_map +
  geom_point(data = Ioxyrinchus_relative_abundance_df_elas, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "blue4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Endangered " * italic("Isurus oxyrinchus ") * "(Shortfin mako shark), Elas02 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Ioxyrinchus_elas


ggsave("figures-expedition/Ioxyrinchus_map_CO1.eps", plot = shinn_bay_Ioxyrinchus, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Ioxyrinchus_map_CO1.jpg", plot = shinn_bay_Ioxyrinchus, width = 8, height = 6, units = "in")


ggsave("figures-expedition/Ioxyrinchus_map_Elas02.eps", plot = shinn_bay_Ioxyrinchus_elas, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Ioxyrinchus_map_Elas02.jpg", plot = shinn_bay_Ioxyrinchus_elas, width = 8, height = 6, units = "in")

```


### Vulnerable: Limulus polyphemus (American Horseshoe Crab)

From [source](https://www.iucnredlist.org/species/11987/80159830): 

#### Prepare data
```{r}
# Calculate relative abundance  per Site
Lpolyphemus_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Limulus polyphemus")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Lpolyphemus_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Lpolyphemus_relative_abundance_df <- Lpolyphemus_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Lpolyphemus_relative_abundance_df

```



#### Plot map

```{r}
shinn_bay_Lpolyphemus <- shinnbay_map +
  geom_point(data = Lpolyphemus_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "brown4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Vulnerable " * italic("Limulus polyphemus ") * "(American Horseshoe Crab), CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Lpolyphemus

ggsave("figures-expedition/Lpolyphemus_map_CO1.eps", plot = shinn_bay_Lpolyphemus, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Lpolyphemus_map_CO1.jpg", plot = shinn_bay_Lpolyphemus, width = 8, height = 6, units = "in")

```



### Endangered: Anguilla rostrata (American Eel)

Endangered on IUCN [source](https://www.iucnredlist.org/species/191108/129638652).

Special concern according to NY DEC. [Source](https://dec.ny.gov/nature/animals-fish-plants/biodiversity-species-conservation/endangered-species/list)

#### Prepare data
```{r}
# Calculate relative abundance  per Site
Arostrata_relative_abundance_df <- ps2024_exp_co1_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Anguilla rostrata")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Arostrata_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Arostrata_relative_abundance_df <- Arostrata_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Arostrata_relative_abundance_df

```

also detected by MiFish, pull that out as well

```{r}
# Calculate relative abundance  per Site
Arostrata_relative_abundance_df_mifish <- ps2024_exp_mifish_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Anguilla rostrata")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Arostrata_relative_abundance_df_mifish

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Arostrata_relative_abundance_df_mifish <- Arostrata_relative_abundance_df_mifish %>%
  left_join(stationdata, by = c("sites" = "sites"))


Arostrata_relative_abundance_df_mifish

```

#### Plot map
Both CO1 and MiFish for comparison
```{r}
shinn_bay_Arostrata <- shinnbay_map +
  geom_point(data = Arostrata_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "darkgoldenrod4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Endangered " * italic("Anguilla rostrata Anguilla rostrata ") * "(American Eel), CO1 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Arostrata


shinn_bay_Arostrata_mifish <- shinnbay_map +
  geom_point(data = Arostrata_relative_abundance_df_mifish, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "darkgoldenrod4") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Endangered " * italic("Anguilla rostrata Anguilla rostrata ") * "(American Eel), MiFish Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Arostrata_mifish


ggsave("figures-expedition/Arostrata_map_CO1.eps", plot = shinn_bay_Arostrata, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Arostrata_map_CO1.jpg", plot = shinn_bay_Arostrata, width = 8, height = 6, units = "in")


ggsave("figures-expedition/Arostrata_map_mifish.eps", plot = shinn_bay_Arostrata_mifish, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Arostrata_map_mifish.jpg", plot = shinn_bay_Arostrata_mifish, width = 8, height = 6, units = "in")

```


### Endangered: Megaptera novaeangliae (Humpback whale)

[Endangered in NY state](https://dec.ny.gov/nature/animals-fish-plants/biodiversity-species-conservation/endangered-species/list)

But least concern in IUCN. [Source](https://www.iucnredlist.org/species/13006/50362794)

#### Prepare data
Only detected in MiFish
```{r}
# Calculate relative abundance  per Site
Mnovaeangliae_relative_abundance_df <- ps2024_exp_mifish_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(Species == "Megaptera novaeangliae")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Mnovaeangliae_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Mnovaeangliae_relative_abundance_df <- Mnovaeangliae_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Mnovaeangliae_relative_abundance_df

```

#### Plot map

```{r}
shinn_bay_Mnovaeangliae <- shinnbay_map +
  geom_point(data = Mnovaeangliae_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "darkslategrey") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Endangered " * italic("Megaptera novaeangliae ") * "(Humpback whale), MiFish Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Mnovaeangliae

ggsave("figures-expedition/Mnovaeangliae_map_mifish.eps", plot = shinn_bay_Mnovaeangliae, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Mnovaeangliaes_map_mifish.jpg", plot = shinn_bay_Mnovaeangliae, width = 8, height = 6, units = "in")

```


### Vulnerable: (Roughtail stingray)
Vulnerable overall according to IUCN: https://www.iucnredlist.org/ja/species/104065040/3122808
#### Prepare data
Only detected in MiFish
```{r}
# Calculate relative abundance  per Site
Bcentroura_relative_abundance_df <- ps2024_exp_elas_glommed_df %>%
  group_by(sites) %>%
  summarise(TotalCounts = n(), 
            Counts = sum(CommonName == "Roughtail stingray")) %>%
  mutate(RelativeAbundance = Counts / TotalCounts) %>%
  select(-TotalCounts, -Counts)  %>%
  group_by(sites) %>%
  summarise(AverageRelativeAbundancePercent = mean(RelativeAbundance)*100) %>%
  filter(AverageRelativeAbundancePercent > 0) # remove zeroes so they do no appear as bubbles in plot
Bcentroura_relative_abundance_df

# add in station metadata
stationdata <- read_csv("../eDNA-databases/stationsmetadata.csv")
stationdata$sites <- as.factor(stationdata$sites) 

# combine
Bcentroura_relative_abundance_df <- Bcentroura_relative_abundance_df %>%
  left_join(stationdata, by = c("sites" = "sites"))


Bcentroura_relative_abundance_df

```

#### Plot map

```{r}
shinn_bay_Bcentroura <- shinnbay_map +
  geom_point(data = Bcentroura_relative_abundance_df, aes(x = long, y = lat, size = AverageRelativeAbundancePercent), alpha = 0.6, color = "#4D5D53") +
  scale_size_continuous(name = "Average Relative\nAbundance (%)", range = c(2, 9)) +
  theme_minimal() +
labs(title = expression("Relative Abundance, Vulnerable " * italic("Bathytoshia centroura ") * "(Roughtail Stingray), Elas02 Reads"),
       x = "Longitude", y = "Latitude")

shinn_bay_Bcentroura

ggsave("figures-expedition/Bcentroura_map_mifish.eps", plot = shinn_bay_Bcentroura, width = 8, height = 6, units = "in")
ggsave("figures-expedition/Bcentroura_map_mifish.jpg", plot = shinn_bay_Bcentroura, width = 8, height = 6, units = "in")

```


### Other detected species to watch wrt conservation:

- Syngnathus fuscus(Northern pipefish): Not endangered/threatened/special concern but identified as "Priority Species of Greatest Conservation Need". [DEC source](https://dec.ny.gov/nature/animals-fish-plants/biodiversity-species-conservation/endangered-species/list)
- Similarly, eastern oyster, hard clam, tautog, American Shad, Winter Flounder, Roughtail Stingray are all on this NY Conservation list as well
- Didn't plot Xiphias gladius (Swordfish), Lopholatilus chamaeleonticeps (Tilefish), Thunnus obesus (Bigeye tuna) bc while threatened/ endangered, based on where these were [sts 1 and 19], we suspect they are runoff from restaurants/marinas
- Cross check list of species with the [NYS DEC site on invasives](https://dec.ny.gov/nature/invasive-species/aquatic/marine)
- Pull out those that were identified to the species level and see if any are bioindicators (i.e. [Polygordius jouinae is a polychaete that might indicate well-sorted sandy bottoms](https://link.springer.com/article/10.1007/s00227-008-0936-9))



### Save/ Re-load
```{r}
save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_maps.RData")
```

or load back in
```{r}
load(file = "figures-expedition/exp_ecol_analysis_environment_upto_maps.RData")
```


# Replication
How do replicates compare?

## Paired barplots

### Mifish
Look at species level
```{r}
sample_data(ps2024_exp_mifish)$replicates
# 90 samples, with 73 unique levels. So there are 17 samples which were collected in duplicate

duplicate_IDs <- sample_data(ps2024_exp_mifish)$replicates[duplicated(sample_data(ps2024_exp_mifish)$replicates)]
duplicate_IDs

# make new phyloseq object with only replicates
ps2024_exp_reps_mifish <- subset_samples(ps2024_exp_mifish_glommed, replicates %in% duplicate_IDs)
ps2024_exp_reps_mifish_ra <- subset_samples(ps2024_exp_mifish_glommed_ra, replicates %in% duplicate_IDs)


ps2024_exp_barplot_replicates <- plot_bar(ps2024_exp_reps_mifish, x = "Name.Deploy_Cartr_Library", fill = "CommonName")+
  scale_fill_manual(values = myColors_comname)
ps2024_exp_barplot_replicates

ps2024_exp_ra_barplot_replicates_ra <- plot_bar(ps2024_exp_reps_mifish_ra, x = "Name.Deploy_Cartr_Library", fill = "CommonName")+
  scale_fill_manual(values = myColors_comname)
ps2024_exp_ra_barplot_replicates_ra

# view replicates pair-by-pair
ps2024_exp_ra_barplot_replicates_ra <- ps2024_exp_ra_barplot_replicates_ra +
    facet_grid(~replicates, scales = "free", space = "free", drop= TRUE)+
    scale_fill_manual(values = myColors_comname)+
    theme_minimal() +
    theme(legend.position = "none", 
          axis.title.x = element_blank(), 
          axis.text.x = element_text(size = 0), 
          strip.text = element_text(size = 12))
ps2024_exp_ra_barplot_replicates_ra


ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside.eps", width = 12, height = 7, units = "in")

ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside.jpg", width = 12, height = 7, units = "in")
```

- Some replicates look very close to each other (~10 of the 17) while others do not. I manually checked the Sample Inventory. One issue may be filtration volume- these are not always consistent bc of the issues with overpressuring
- I also notice that the more diverse the sample is, the less similar the replicates are. eg. 5_19 and 8_11a


### Elas02
Look at species level
```{r}
sample_data(ps2024_exp_elas)$replicates

duplicate_IDs <- sample_data(ps2024_exp_elas)$replicates[duplicated(sample_data(ps2024_exp_elas)$replicates)]
duplicate_IDs

# make new phyloseq object with only replicates
ps2024_exp_reps_elas <- subset_samples(ps2024_exp_elas_glommed, replicates %in% duplicate_IDs)
ps2024_exp_reps_elas_ra <- subset_samples(ps2024_exp_elas_glommed_ra, replicates %in% duplicate_IDs)


ps2024_exp_barplot_replicates <- plot_bar(ps2024_exp_reps_elas, x = "Name.Deploy_Cartr", fill = "CommonName")+
  scale_fill_manual(values = myColors_comname)
ps2024_exp_barplot_replicates

ps2024_exp_ra_barplot_replicates_ra <- plot_bar(ps2024_exp_reps_elas_ra, x = "Name.Deploy_Cartr", fill = "CommonName")+
  scale_fill_manual(values = myColors_comname)
ps2024_exp_ra_barplot_replicates_ra

# view replicates pair-by-pair
ps2024_exp_ra_barplot_replicates_ra <- ps2024_exp_ra_barplot_replicates_ra +
    facet_grid(~replicates, scales = "free", space = "free", drop= TRUE)+
    scale_fill_manual(values = myColors_comname)+
    theme_minimal() +
    theme(legend.position = "none", 
          axis.title.x = element_blank(), 
          axis.text.x = element_text(size = 0), 
          strip.text = element_text(size = 12))
ps2024_exp_ra_barplot_replicates_ra


ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside_elas02.eps", width = 8, height = 7, units = "in")

ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside_elas02.jpg", width = 8, height = 7, units = "in")
```
Replication is pretty big but there is not a lot of variability over all


### CO1
Look at Phlyum level
```{r}
sample_data(ps2024_exp_co1)$replicates

duplicate_IDs <- sample_data(ps2024_exp_co1)$replicates[duplicated(sample_data(ps2024_exp_co1)$replicates)]
duplicate_IDs

# make new phyloseq object with only replicates
ps2024_exp_reps_co1 <- subset_samples(ps2024_exp_co1_glommed, replicates %in% duplicate_IDs)
ps2024_exp_reps_co1_ra <- subset_samples(ps2024_exp_co1_glommed_ra, replicates %in% duplicate_IDs)


ps2024_exp_barplot_replicates <- plot_bar(ps2024_exp_reps_co1, x = "Name_Deploy_Cartr", fill = "Phylum")
ps2024_exp_barplot_replicates

ps2024_exp_ra_barplot_replicates_ra <- plot_bar(ps2024_exp_reps_co1_ra, x = "Name_Deploy_Cartr", fill = "Phylum")
ps2024_exp_ra_barplot_replicates_ra

# view replicates pair-by-pair
ps2024_exp_ra_barplot_replicates_ra <- ps2024_exp_ra_barplot_replicates_ra +
    facet_grid(~replicates, scales = "free", space = "free", drop= TRUE) +
    theme_minimal() +
    theme(axis.title.x = element_blank(), 
          axis.text.x = element_text(size = 0), 
          strip.text = element_text(size = 12))
ps2024_exp_ra_barplot_replicates_ra


ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside_co1.eps", width = 18, height = 6, units = "in")

ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside_co1.jpg", width = 18, height = 6, units = "in")
```
Replication looks pretty decent in most cases for the phylum level.


## Aitchison dissimilarity among repilcates


For [compositional data](https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2017.02224/full), use Aitchison distance on CLR-transformed data


From [vegan](https://rdrr.io/cran/vegan/man/vegdist.html) documentation, "aitchison" method in `vegdist` automatically does the clr transformation before calculating the distance matrix. Phyloseq has [some options](https://joey711.github.io/phyloseq/distance.html) to do run vegdist with a phyloseq object using a `distance` function, however aitchison is not listed:
```{r}
dist_methods <- unlist(distanceMethodList)
print(dist_methods)
```

On the other hand, from the vegan documentation, Aitchison distance is just the Euclidean distance on clr-transformed data, so I can do a transformation first and then the Euclidean distance calculation.



I also spent a long time figuring out why clr transformation using 2 different packages was giving different distance matrices. Finally figured it out using practice dataset, dietswap, below. Next apply to real data:
```{r}
###

data(dietswap)

otu_df <- as.data.frame(otu_table(dietswap))

otu_df[otu_df == 0] <- 1e-8 # Small value replacement for zeros

otu_df


## FINALLY FIGURED THAT THESE 2 THINGS ARE EQUIVLENT:
data.frame(microbiome::transform(t(otu_df), transform = "clr") ) 
data.frame(compositions::clr(otu_df))

```

### Mifish
Dissimilarity between replicates
```{r}
# replace zeroes with very very small numbers
otu_df <- as.data.frame(otu_table(ps2024_exp_reps_mifish))
otu_df[otu_df == 0] <- 1e-8 

# check that
data.frame(microbiome::transform(t(otu_df), transform = 'clr'))
# and
data.frame(compositions::clr(otu_df))
# are equivalent


ps2024_exp_reps_clr_mifish <- microbiome::transform(t(otu_df), transform = 'clr')
  
ps2024_exp_reps_dist_mifish <- vegdist(ps2024_exp_reps_clr_mifish, method= "euclidean")

ps2024_exp_reps_dist_mat_mifish <- as.matrix(ps2024_exp_reps_dist_mifish)

ps2024_exp_reps_dist_df_mifish <- as.data.frame(ps2024_exp_reps_dist_mat_mifish)
ps2024_exp_reps_dist_df_mifish

write.csv(ps2024_exp_reps_dist_df_mifish,"figures-expedition/mifish_2024_exp_aitchisondistance_replicates.csv", row.names = TRUE)
```
Analyzed the above in Excel in the file, [mifish_2024_exp_aitchisondistance_replicates.xlsx](figures-expedition/mifish_2024_exp_aitchisondistance_replicates.xlsx)
- Highlighted dissimilarity index between replicates
- Calculated average dissimilarity between replicates = 47.13; SD = 24.94; SE = 6.05


Get dissimilarity indices and their average across whole dataset, for comparison:
```{r}
# replace zeroes with very very small numbers
otu_df <- as.data.frame(otu_table(ps2024_exp_mifish))
otu_df[otu_df == 0] <- 1e-8 

# check that
data.frame(microbiome::transform(t(otu_df), transform = 'clr'))
# and
data.frame(compositions::clr(otu_df))
# are equivalent


ps2024_exp_clr_mifish <- microbiome::transform(t(otu_df), transform = 'clr')
  
ps2024_exp_dist_mifish <- vegdist(ps2024_exp_clr_mifish, method= "euclidean")

ps2024_exp_dist_mat_mifish <- as.matrix(ps2024_exp_dist_mifish)

ps2024_exp_dist_df_mifish <- as.data.frame(ps2024_exp_dist_mat_mifish)
ps2024_exp_dist_df_mifish

# mean of all distance indices?
mean(ps2024_exp_dist_mifish)
sd(ps2024_exp_dist_mifish)
sd(ps2024_exp_dist_mifish)/sqrt(dim(ps2024_exp_dist_mifish)[1]*dim(ps2024_exp_dist_mifish)[2])

```
On average, across the whole dataset, the dissimilarity is 101.48 (SD 17.19; SE 0.19) while. Replicates are much less dissimilar than every sample compared to every other sample.



### Elas02
Dissimilarity between replicates
```{r}
# replace zeroes with very very small numbers
otu_df <- as.data.frame(otu_table(ps2024_exp_reps_elas))
otu_df[otu_df == 0] <- 1e-8 

# check that
data.frame(microbiome::transform(t(otu_df), transform = 'clr'))
# and
data.frame(compositions::clr(otu_df))
# are equivalent


ps2024_exp_reps_clr_elas <- microbiome::transform(t(otu_df), transform = 'clr')
  
ps2024_exp_reps_dist_elas <- vegdist(ps2024_exp_reps_clr_elas, method= "euclidean")

ps2024_exp_reps_dist_mat_elas <- as.matrix(ps2024_exp_reps_dist_elas)

ps2024_exp_reps_dist_df_elas <- as.data.frame(ps2024_exp_reps_dist_mat_elas)
ps2024_exp_reps_dist_df_elas

write.csv(ps2024_exp_reps_dist_df_elas,"figures-expedition/elas_2024_exp_aitchisondistance_replicates.csv", row.names = TRUE)
```

Analyzed the above in Excel in the file, [elas_2024_exp_aitchisondistance_replicates.xlsx](figures-expedition/elas_2024_exp_aitchisondistance_replicates.xlsx)
- Highlighted dissimilarity index between replicates
- Calculated average dissimilarity between replicates = 5.76; SD = 9.26; SE = 2.48


Get dissimilarity indices and their average across whole dataset, for comparison:
```{r}
# replace zeroes with very very small numbers
otu_df <- as.data.frame(otu_table(ps2024_exp_elas))
otu_df[otu_df == 0] <- 1e-8 

# check that
data.frame(microbiome::transform(t(otu_df), transform = 'clr'))
# and
data.frame(compositions::clr(otu_df))
# are equivalent


ps2024_exp_clr_elas <- microbiome::transform(t(otu_df), transform = 'clr')
  
ps2024_exp_dist_elas <- vegdist(ps2024_exp_clr_elas, method= "euclidean")

ps2024_exp_dist_mat_elas <- as.matrix(ps2024_exp_dist_elas)

ps2024_exp_dist_df_elas <- as.data.frame(ps2024_exp_dist_mat_elas)
ps2024_exp_dist_df_elas

# mean of all distance indices?
mean(ps2024_exp_dist_elas)
sd(ps2024_exp_dist_elas)
sd(ps2024_exp_dist_elas)/sqrt(dim(ps2024_exp_dist_elas)[1]*dim(ps2024_exp_dist_elas)[2])

```

On average, across the whole dataset, the dissimilarity is 168.04 (SD 25.87; SE 0.42) while. Replicates are much less dissimilar than every sample compared to every other sample.


### CO1
Dissimilarity between replicates
```{r}
# replace zeroes with very very small numbers
otu_df <- as.data.frame(otu_table(ps2024_exp_reps_co1))
otu_df[otu_df == 0] <- 1e-8 

# check that
data.frame(microbiome::transform(t(otu_df), transform = 'clr'))
# and
data.frame(compositions::clr(otu_df))
# are equivalent


ps2024_exp_reps_clr_co1 <- microbiome::transform(t(otu_df), transform = 'clr')
  
ps2024_exp_reps_dist_co1 <- vegdist(ps2024_exp_reps_clr_co1, method= "euclidean")

ps2024_exp_reps_dist_mat_co1 <- as.matrix(ps2024_exp_reps_dist_co1)

ps2024_exp_reps_dist_df_co1 <- as.data.frame(ps2024_exp_reps_dist_mat_co1)
ps2024_exp_reps_dist_df_co1

write.csv(ps2024_exp_reps_dist_df_co1,"figures-expedition/co1_2024_exp_aitchisondistance_replicates.csv", row.names = TRUE)
```


Analyzed the above in Excel in the file, [co1_2024_exp_aitchisondistance_replicates.xlsx](figures-expedition/co1_2024_exp_aitchisondistance_replicates.xlsx)
- Highlighted dissimilarity index between replicates
- Calculated average dissimilarity between replicates = 218.05; SD = 38.98; SE = 9.19


Get dissimilarity indices and their average across whole dataset, for comparison:
```{r}
# replace zeroes with very very small numbers
otu_df <- as.data.frame(otu_table(ps2024_exp_co1))
otu_df[otu_df == 0] <- 1e-8 

# check that
data.frame(microbiome::transform(t(otu_df), transform = 'clr'))
# and
data.frame(compositions::clr(otu_df))
# are equivalent


ps2024_exp_clr_co1 <- microbiome::transform(t(otu_df), transform = 'clr')
  
ps2024_exp_dist_co1 <- vegdist(ps2024_exp_clr_co1, method= "euclidean")

ps2024_exp_dist_mat_co1 <- as.matrix(ps2024_exp_dist_co1)

ps2024_exp_dist_df_co1 <- as.data.frame(ps2024_exp_dist_mat_co1)
ps2024_exp_dist_df_co1

# mean of all distance indices?
mean(ps2024_exp_dist_co1)
sd(ps2024_exp_dist_co1)
sd(ps2024_exp_dist_co1)/sqrt(dim(ps2024_exp_dist_co1)[1]*dim(ps2024_exp_dist_co1)[2])

```

On average, across the whole dataset, the dissimilarity is 1134.02 (SD 234.37; SE 2.49) while. Replicates are much less dissimilar than every sample compared to every other sample.







#### Save

```{r}
save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_Aitchison_reps.RData")
```


```{r}
load(file = "figures-expedition/exp_ecol_analysis_environment_upto_Aitchison_reps.RData")
```

# Occupancy

Present Presence/ Absence as site occupancy, eg. across all samples from XX habitat type or east/west etc etc, how many times did species X appear. Check out [Lawson Handley et al. 2019](https://onlinelibrary.wiley.com/doi/full/10.1002/edn3.5)

### MiFIsh

```{r}
# How many total samples taken in each habitat type?
Total_samples_habitat <- as.data.frame(samples_2024_exp_mifish) %>%
  group_by(Habitat) %>%
  count(name = "Habitat_sample_total") %>%
  na.omit()

# in east vs west?
Total_samples_EW <- as.data.frame(samples_2024_exp_mifish) %>%
  group_by(Bayside) %>%
  count(name = "Bayside_sample_total") %>%
  filter(!Bayside=='Control')

# in day vs night?
Total_samples_DayNight <- as.data.frame(samples_2024_exp_mifish) %>%
  group_by(Night__Day) %>%
  count(name = "Night_day_sample_total") %>%
  na.omit()

Total_samples_habitat
Total_samples_EW
Total_samples_DayNight
```

```{r}
# join sample totals to dataframe so there is a denomenator for the calculation
ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df, Total_samples_habitat, by = "Habitat")
ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Total_samples_EW, by = "Bayside")
ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Total_samples_DayNight, by = "Night__Day")


ps2024_exp_mifish_glommed_df_2

```

Calculate site occupancy for each species
```{r}
# Habitat
Species_occurences_habitat <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat) %>%
  count(name = "Sp_occ_in_habitat")

ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Species_occurences_habitat, by = c("Habitat"="Habitat","Species.CommonName"="Species.CommonName"))

#Bayside
Species_occurences_bayside <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside) %>%
  count(name = "Sp_occ_in_bayside")

ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Species_occurences_bayside, by = c("Bayside"="Bayside","Species.CommonName"="Species.CommonName"))

#Daynight
Species_occurences_daynight <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day) %>%
  count(name = "Sp_occ_in_DayNight")

ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Species_occurences_daynight, by = c("Night__Day"="Night__Day","Species.CommonName"="Species.CommonName"))


# Calculate occurrence over sample total
Species_occurences_habitat <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat, Habitat_sample_total, Sp_occ_in_habitat) %>%
  mutate(Species_occurences_in_habitat = Sp_occ_in_habitat/Habitat_sample_total) %>%
  select(Species.CommonName, CommonName, Habitat, Species_occurences_in_habitat) %>%
  distinct()
Species_occurences_habitat

Species_occurences_bayside <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside, Bayside_sample_total, Sp_occ_in_bayside) %>%
  mutate(Species_occurences_in_bayside = Sp_occ_in_bayside/Bayside_sample_total) %>%
  select(Species_occurences_in_bayside, CommonName, Bayside, Species_occurences_in_bayside) %>%
  distinct()
Species_occurences_bayside

Species_occurences_daynight <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day, Night_day_sample_total, Sp_occ_in_DayNight) %>%
  mutate(Species_occurences_in_daynight = Sp_occ_in_DayNight/Night_day_sample_total) %>%
  select(Species.CommonName, CommonName, Night__Day, Species_occurences_in_daynight) %>%
  distinct()
Species_occurences_daynight


```

Plot
```{r}
speciesoccurrence_habitat <- ggplot(data = Species_occurences_habitat, aes(x = CommonName, y = Species_occurences_in_habitat, fill = Habitat))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
speciesoccurrence_habitat

speciesoccurrence_daynight <- ggplot(data = Species_occurences_daynight, aes(x = CommonName, y = Species_occurences_in_daynight, fill = Night__Day))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
speciesoccurrence_daynight

speciesoccurrence_bayside <- ggplot(data = Species_occurences_bayside, aes(x = CommonName, y = Species_occurences_in_bayside, fill = Bayside))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("coral", "cornflowerblue"))
speciesoccurrence_bayside


ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat.eps", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight.eps", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside.eps", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat.jpg", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight.jpg", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside.jpg", width = 12, height = 5, units = "in")

```

### Elas02

```{r}
# How many total samples taken in each habitat type?
Total_samples_habitat <- as.data.frame(samples_2024_exp_elas) %>%
  group_by(Habitat) %>%
  count(name = "Habitat_sample_total") %>%
  na.omit()

# in east vs west?
Total_samples_EW <- as.data.frame(samples_2024_exp_elas) %>%
  group_by(Bayside) %>%
  count(name = "Bayside_sample_total") %>%
  filter(!Bayside=='Control')

# in day vs night?
Total_samples_DayNight <- as.data.frame(samples_2024_exp_elas) %>%
  group_by(Night__Day) %>%
  count(name = "Night_day_sample_total") %>%
  na.omit()

Total_samples_habitat
Total_samples_EW
Total_samples_DayNight
```



```{r}
# join sample totals to dataframe so there is a denomenator for the calculation
ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df, Total_samples_habitat, by = "Habitat")
ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Total_samples_EW, by = "Bayside")
ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Total_samples_DayNight, by = "Night__Day")


# there are no elasmos detected at oyster reef sites so there is an empty row. Delete
ps2024_exp_elas_glommed_df_2 <- ps2024_exp_elas_glommed_df_2 %>% filter(!is.na(OTU))


ps2024_exp_elas_glommed_df_2

```


Calculate site occupancy for each species
```{r}
# Habitat
Species_occurences_habitat <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat) %>%
  count(name = "Sp_occ_in_habitat")

ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Species_occurences_habitat, by = c("Habitat"="Habitat","Species.CommonName"="Species.CommonName"))

#Bayside
Species_occurences_bayside <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside) %>%
  count(name = "Sp_occ_in_bayside")

ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Species_occurences_bayside, by = c("Bayside"="Bayside","Species.CommonName"="Species.CommonName"))

#Daynight
Species_occurences_daynight <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day) %>%
  count(name = "Sp_occ_in_DayNight")

ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Species_occurences_daynight, by = c("Night__Day"="Night__Day","Species.CommonName"="Species.CommonName"))


# Calculate occurrence over sample total
Species_occurences_habitat <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat, Habitat_sample_total, Sp_occ_in_habitat) %>%
  mutate(Species_occurences_in_habitat = Sp_occ_in_habitat/Habitat_sample_total) %>%
  select(Species.CommonName, CommonName, Habitat, Species_occurences_in_habitat) %>%
  distinct()
Species_occurences_habitat

Species_occurences_bayside <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside, Bayside_sample_total, Sp_occ_in_bayside) %>%
  mutate(Species_occurences_in_bayside = Sp_occ_in_bayside/Bayside_sample_total) %>%
  select(Species.CommonName, CommonName, Bayside, Species_occurences_in_bayside) %>%
  distinct()
Species_occurences_bayside

Species_occurences_daynight <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day, Night_day_sample_total, Sp_occ_in_DayNight) %>%
  mutate(Species_occurences_in_daynight = Sp_occ_in_DayNight/Night_day_sample_total) %>%
  select(Species.CommonName, CommonName, Night__Day, Species_occurences_in_daynight) %>%
  distinct()
Species_occurences_daynight


```



Plot
```{r}
speciesoccurrence_habitat <- ggplot(data = Species_occurences_habitat, aes(x = CommonName, y = Species_occurences_in_habitat, fill = Habitat))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
speciesoccurrence_habitat

speciesoccurrence_daynight <- ggplot(data = Species_occurences_daynight, aes(x = CommonName, y = Species_occurences_in_daynight, fill = Night__Day))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
speciesoccurrence_daynight

speciesoccurrence_bayside <- ggplot(data = Species_occurences_bayside, aes(x = CommonName, y = Species_occurences_in_bayside, fill = Bayside))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("coral", "cornflowerblue"))
speciesoccurrence_bayside


ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat_elas.eps", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight_elas.eps", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside_elas.eps", width = 5, height = 5, units = "in")




ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat_elas.jpg", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight_elas.jpg", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside_elas.jpg", width = 5, height = 5, units = "in")

```




## spOccupancy
There are more sophisticated ways of examining occupancy that depend on modeling probabilities and that will give error estimates. Several R packages exist, [spOccupancy](https://doserlab.com/files/spoccupancy-web/) seems the most well developed and widely used. See [vignette](https://doserlab.com/files/spoccupancy-web/articles/modelfitting).

Start to format data set. 
- `spOccupancy` expects presence/absence data rather than counts
- assesses both "occurrence" covariates and "detection" covariates in separate matrices. From what I can tell:
  - "occurrence" covariates are "site-level" variables. Environmental parameters (salinity, depth, etc)
  - "detection" covariates are "observation/survey-level" variables due to sampling itself (day, time, etc.)
  - there is a separate matrix with coordinates

### Elas02
#### Prepare data
Test with Elas df to start since it's smaller...
Use dfs where controls were already removed. Simplify metadata variables
```{r}
elas_2024_exp_asv_table_taxbased
elas_2024_exp_sample_data_taxbased
elas_2024_exp_tax_table_taxbased
colnames(elas_2024_exp_sample_data_taxbased)
```


Simplify sample names, remove redundant or unuseful metadata
```{r}
# Simplify sample names in count table
elas_exp_df_counttable <- elas_2024_exp_asv_table_taxbased %>%
  rename_with(~str_extract(.x, "\\d+_\\d+"))

# Simplify sample names in sample data table
elas_exp_df_sampledata <- elas_2024_exp_sample_data_taxbased
rownames(elas_exp_df_sampledata) <- str_extract(rownames(elas_exp_df_sampledata), "\\d+_\\d+")

# Remove some columns in sample data
elas_exp_df_sampledata <- elas_exp_df_sampledata %>%
  select(
    -Name_Deploy_Cartr_Library,
    -Name.Deploy_Cartr,
    -Deployment,
    -Cartridge_ID,
    -Site_type,
    -controls,
    -Date,
    -End_Time__ET_,
    -Datecode,
    -Year,
    -Month,
    -Deploy_Cartr_Library)

elas_exp_df_counttable
glimpse(elas_exp_df_sampledata)

```







Convert counts to presence/ absence 
```{r}
elas_exp_df_countmatrix <- elas_exp_df_counttable
elas_exp_df_countmatrix[] <- ifelse(elas_exp_df_countmatrix > 0, 1, 0)

# compare
elas_exp_df_counttable
elas_exp_df_countmatrix
```


Occupancy and Detection Covariates
```{r}
# Helper function to get mean for numerics, mode for factors/characters
mean_or_mode <- function(x) {
  if (is.numeric(x)) {
    return(mean(x, na.rm = TRUE))
  } else if (is.factor(x) || is.character(x)) {
    ux <- unique(na.omit(x))
    if (length(ux) == 1) return(ux)
    return(ux[which.max(tabulate(match(x, ux)))])
  } else {
    return(NA)
  }
}


# Occupancy covariates: one row per unique replicate group
# Don't include spatial variables because that is built into coords later and would lead to redundancy
occ.covs <- elas_exp_df_sampledata %>%
  group_by(replicates) %>%
  summarize(across(c(Bayside, Habitat, Night__Day, Avg_Depth_m, Avg_Temperature_C, Avg_Salinity_psu, Avg_Chlorophyll_ugL, Avg_Turbidity_NTU), mean_or_mode)) %>%
  as.data.frame()
# remove rep names and use as IDs
rownames(occ.covs) <- occ.covs$replicates
occ.covs$replicates <- NULL



# Detection covariates: list of matrices for each sampling variable. Matrix contains sites and replicates
# Extract and sort by rep
det_grouped <- elas_exp_df_sampledata[, c("replicates", "Volume_liter", "Start_Time__ET_", "Day")] %>%
  group_by(replicates) %>%
  arrange(replicates) %>%
  mutate(rep_id = row_number()) %>%
  ungroup()

# Reshape to wide format for each covariate
vol_mat <- det_grouped %>%
  select(replicates, rep_id, Volume_liter) %>%
  pivot_wider(names_from = rep_id, values_from = Volume_liter) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

day_mat <- det_grouped %>%
  select(replicates, rep_id, Day) %>%
  pivot_wider(names_from = rep_id, values_from = Day) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

# use decimal hour for time
start_time_mat <- det_grouped %>%
  select(replicates, rep_id, Start_Time__ET_) %>%
  mutate(Start_Time__ET_ = hms::as_hms(as.character(Start_Time__ET_)),
         Start_Time__ET_ = hour(Start_Time__ET_) + 
                           minute(Start_Time__ET_) / 60 + 
                           second(Start_Time__ET_) / 3600) %>%
  pivot_wider(names_from = rep_id, values_from = Start_Time__ET_) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

# Combine into a named list
det.covs <- list(
  Volume_liter = vol_mat,
  Day = day_mat,
  Start_Time = start_time_mat
)

occ.covs
det.covs
```

Drop strongly correlated covariates (leads to errors later)---
```{r}
# Get Pearson correlation matrix for numeric occupancy covariates
num_covs <- occ.covs[, sapply(occ.covs, is.numeric)]
cor_matrix <- cor(num_covs, use = "complete.obs")
round(cor_matrix, 2)
```
None are greater than 0.9. Keep all

Check categorical variables to see if any are represented in only 1-2 sites, which can cause errors
```{r}
table(occ.covs$Bayside)
table(occ.covs$Habitat)
```
Clam sanctuary only represented once...

Change both "CLAM SANCTUARY" and "EELGRASS" to "RESTORED" so they will be one category
```{r}
occ.covs$Habitat <- as.character(occ.covs$Habitat)  # convert to character if factor
occ.covs$Habitat[occ.covs$Habitat %in% c("CLAM SANCTUARY", "EELGRASS")] <- "RESTORED"
occ.covs$Habitat <- factor(occ.covs$Habitat)  # convert back to factor

occ.covs
```



Transform count matrix to a 3D array y[i,j,k], where:
i indexes species (ASVs),
j indexes sites (i.e., unique replicate groups),
k indexes replicate samples at that site.

Group samples based on replicates column. Reorganize so all replicates are aligned based on `occ.covs`
```{r}
# Unique replicate groups
replicate_groups <- rownames(occ.covs)

# Sort samples by replicate groups
elas_exp_df_sampledata <- elas_exp_df_sampledata[order(elas_exp_df_sampledata$replicates), ]

# Initialize 3D y array
n_species <- nrow(elas_exp_df_countmatrix)
n_sites <- length(unique(elas_exp_df_sampledata$replicates)) # "sites" are unique to time/day/location
max_reps <- max(table(elas_exp_df_sampledata$replicates))

y_array <- array(NA, dim = c(n_species, n_sites, max_reps))

# Populate the array
for (i in seq_along(replicate_groups)) {
  reps <- rownames(elas_exp_df_sampledata)[elas_exp_df_sampledata$replicates == replicate_groups[i]]
  for (j in seq_along(reps)) {
    y_array[, i, j] <- elas_exp_df_countmatrix[, reps[j]]
  }
}

# Retain names
dimnames(y_array) <- list(
  rownames(elas_exp_df_countmatrix),  # species (rows)
  unique(elas_exp_df_sampledata$replicates),     # sites (columns)
  as.character(seq_len(dim(y_array)[3]))       # replicate number (slices)
)

y_array
```
I triple checked that these columns match up with the same order of sample/ replicates in the covariate matrices.



and coords
```{r}
# Create coords data frame for spatial transformation
coords_df <- elas_exp_df_sampledata %>%
  group_by(replicates) %>%
  summarize(across(c(long, lat), mean_or_mode), .groups = "drop")  # .groups avoids grouping issues

# Convert to sf object
coords_sf <- st_as_sf(coords_df, coords = c("long", "lat"), crs = 4326)

# Transform to UTM Zone 18N
coords_utm <- st_transform(coords_sf, crs = 32618)

# Extract projected coordinates in meters
coords_projected <- st_coordinates(coords_utm)

# Set row names to match your occupancy covariates
rownames(coords_projected) <- rownames(occ.covs)

head(coords_projected)

```


Make into one list for spOccupancy
```{r}
# Final data list for spPGOcc
elas.exp.list <- list(
  y = y_array,
  occ.covs = occ.covs,
  det.covs = det.covs,
  coords = coords_projected
)

str(elas.exp.list)
```

#### Run non-spatial model (Multi-Species Occupancy Using Polya-Gamma Latent Variables)
Try runnning the multi species latency model (no spatial effect) first. Use default parameters (vignette says this is generally OK).
I kept messing with n.samples, n.burn, and n.thin based on Rhat and ESS values to get to these values:
```{r}
set.seed(20250615)

# Set priors
priors <- list(
  beta.comm.normal = list(mean = 0, var = 2.72),
  alpha.comm.normal = list(mean = 0, var = 2.72),
  tau.sq.beta.ig = list(a = 0.1, b = 0.1),
  tau.sq.alpha.ig = list(a = 0.1, b = 0.1)
)

# Set initial values
inits <- list(
  beta.comm = rep(0, 7),  # Adjust to number of occupancy covariates
  alpha.comm = rep(0, 4), # Adjust to number of detection covariates
  tau.sq.beta = 1,
  tau.sq.alpha = 1
)

model_out_elas <- msPGOcc(
  occ.formula = ~ Bayside + Habitat + Night__Day + 
                 scale(Avg_Depth_m) + scale(Avg_Temperature_C) + 
                 scale(Avg_Salinity_psu),
  det.formula = ~ scale(Volume_liter) + scale(Day) + scale(Start_Time),
  data = elas.exp.list,
  n.samples = 75000,
  n.burn = 25000,
  n.thin = 10,
  n.chains = 3,
  priors = priors, 
  inits = inits,
  verbose = TRUE,
  n.report = 1000
)


```

##### Save model
```{r}
# save(model_out_elas, file = "figures-expedition/model_out_elas")
# load("figures-expedition/model_out_elas")
```

##### Examine output
```{r}
sink(file.path("figures-expedition", "model_out_elas_summary.txt"))
summary(model_out_elas)
sink()
```


How to interpret?!?!

- 1. Community-level Parameters (average effect of covariates across all species)
  - All the means are on the logit scale (i.e. log-odds).
  - If a credible interval does not exclude zero, theres no strong community-level signal. For example: if Avg_Temperature_C mean = 0.84, but CI: -1.79 to 3.85 --> inconclusive.
  - Rhat values > 1.1, and low ESS (effective sample size) are a sign of poor convergence, so treat with caution
- 2. Species-specific Occupancy Effects. These show how each ASVs occupancy probability changes per unit change in the covariate.
  - Example:
    - Avg_Temperature_C-ASV_1: Mean = 1.99, SD = 3.41, CI = -2.45 to 11.14
    - --> ASV_1 may respond positively to temperature, though the CI is wide.
    - Avg_Salinity_psu-ASV_9: Mean = -1.48, SD = 2.89, CI = -8.76 to 2.29
    - --> ASV_9 is negatively associated with salinity
  - Interpret with caution if ASVs have large standard deviations (including for intercepts) and wide credible intervals, especially if CI includes 0.
    - Rhat values above 1.1 or 1.2, and low ESS (<100), suggest parameters didn't converge well.
    - Might reflect either high variability in occupancy or model instability due to limited data for some ASVs.
- 3. Detection Effects. These tell you how detection varies conditional on presence.
  - Example:
    - Volume_liter: Mean = 1.02, CI = -2.33 to 4.18
    - Day: Mean = 1.13, CI = -1.29 to 3.41
    - Detection is generally better during the day and with higher water volume, but again, CI includes 0 --> uncertain.
    - For specific ASVs (e.g., Day-ASV_7 = 4.58  2.2, CI does not include 0), day sampling clearly improves detection.
    - Conversely, Intercept-ASV_7 = -74.8  35.9 means that even if present, detection is very unlikely unless covariates push it up.
- 4. Variance Terms. These describe variation among species in their responses to covariates.
  - Large variances (especially >1000) suggest strong heterogeneity across species.
  - Example: Variance-BaysideW = 14,263, Variance-Intercept = 3601 --> huge spread in species responses.
  - If convergence is poor (Rhat > 1.1, ESS < 100), dont over-interpret.



*Interesting results* from summary (focus on parameters with ~ high posterior means [+ or -], 95% credible intervals that do not overlap zero (or barely do), and acceptable convergence diagnostics (Rhat < ~1.1 and ESS > ~200))


Community Level Occurrence Effects- 
No statistically significant effects at the community level. All 95% credible intervals include zero, indicating uncertainty in direction and strength of effect.

Community Level Detection Effects -
Also nothing is statistically significant. 
Filtration volume has a notable positive impact, but CI overlaps 0:
                    Mean     SD    2.5%     50%  97.5%   Rhat  ESS
scale(Volume_liter) 1.1490 0.8079 -0.4427 1.1305 2.8114 1.0557  2478

Species Level Detection Effects- 
Filtration volume and day of sampling show strong, significant positive effects for ASV_7 (Roughtail stingray):
                              Mean      SD     2.5%     50%   97.5%   Rhat  ESS
scale(Volume_liter)-ASV_7  1.8273  1.0864   0.3057  1.6447  4.5368 1.0836  908
scale(Day)-ASV_7           13.3178  6.0320   5.6628 12.0203 28.5780 1.1648  459



Species Level Occurrence Effects- 
No statistically significant effects 


*Caveats*
- While Rhat and ESS are acceptable with these model parameters, many effects (e.g., salinity, habitat) have wide credible intervals, indicating uncertainty.



##### Check Model diagnostics
###### 1. Convergence diagnostics
'Density` plots are a visual way of looking at mean and CI. See if the mean is far from zero and if the tails overlap zero.
Rhat values should ideally be close to 1.0 (below ~1.1);, especially for variances  which may suggest convergence issues.
Check Effective Sample Size (ESS): Low ESS (<200) can indicate poor sampling.

(for some reason doesn't plot in notebook- run in console)
NOTE- the model above was iteratively adjusted based on feedback from these diagnostics

```{r}
plot(model_out_elas, param = "beta")  # Fixed effects
plot(model_out_elas, param = "alpha") # Detection
```


What you want to see

- Good mixing: all three chains should overlap heavily, with no one chain stuck in a corner.
- Stationarity: after a short burn-in wiggle, the trace should look like random noise, not a slow trend upward or downward.
- Reasonable spread: the posterior density should be unimodal (one bump), not multi-peaked or extremely skewed (unless really expected).
- No crazy outliers or spikes that suggest the sampler is occasionally taking massive jumps.


Scanning through these, look for whether: 1) the posterior is centered away from zero. If it is, it suggests a non zero effect of the covariate on species occupancy, 2) if the 95% intervals are narrow and exclude zero, 3) that the shape is unimodal and smooth
- Overall, these are all unimodal and smooth although some have longish tails. The ones that center away from zero largely overlap with the interpretation based on mean and CI above

###### 2. Posterior predictive checks


```{r}
ppc.out1 <- ppcOcc(model_out_elas, fit.stat = 'freeman-tukey', group = 1)
summary(ppc.out1)

ppc.out2 <- ppcOcc(model_out_elas, fit.stat = 'freeman-tukey', group = 2)
summary(ppc.out2)

```

--> Tells how well the model can replicate the observed data, using a Freeman-Tukey fit statistic. This statistic is good for sparse or zero-inflated data (like eDNA) and down-weights large count differences.

- A Bayesian p-value  0.5 is ideal.
- Values < 0.05 or > 0.95 suggest lack of fit.
- Slightly low/high values (e.g., 0.070.10 or 0.900.93) suggest potential mild misfit, but aren't typically cause for major concern.

group = 1: pooled over replicates per site
This groups all observations per site (e.g., sample replicates) into one aggregate count per species per site.

Results:

- Community p-value: 0.2348  good fit.
- Species-level p-values:
  - ASV_1 & ASV_2: ~0.10  slight underfitting, but still acceptable.
  - ASV_7: 0.67  strong fit.
  - Others: all between 0.150.21  well within reasonable bounds.
Interpretation: The model captures the site-level variation in detection and occupancy fairly well. No glaring misfit here.

roup = 2: pooled over sites per replicate
This checks the model's ability to match replicate-level detection across all sites for each species. It's slightly more sensitive to detection process misfit.

Results:

- Community p-value: 0.2794  strong overall fit.
- Species-level p-values:
  - ASV_1 & ASV_2: ~0.070.08  slightly low but still acceptable.
  - ASV_714: 0.250.47  well within acceptable bounds.
Interpretation:
Model fits replicate-level detection reasonably well. There may be mild underfitting for ASV_1 and ASV_2, but nothing serious.

###### 3. WAIC for model comparison

```{r}
waicOcc(model_out_elas)

```


What each metric means:
1. WAIC (Widely Applicable Information Criterion) = 72.33
Lower WAIC is better when comparing models.
It reflects both how well the model fits the data and how complex it is (i.e., it penalizes overfitting).
Since this is a single model, the WAIC alone isnt meaningful unless comparing it to WAIC from another model.

2. elpd (Expected Log Predictive Density) = -20.91
Measures the models predictive accuracy: how well it predicts new data.
More positive values indicate better predictive performance.
This number is related to WAIC by: WAIC  -2 * elpd.

3. pD (effective number of parameters) = 15.26
Reflects the models complexity.
A higher value indicates a more flexible (and potentially overfit) model.
Given that there are only have 6 species and a modest number of covariates, a pD around 15 is reasonable.

##### Summary for manuscript

*Text Summary*
The non-spatial multi-species occupancy model for the elasmobranch primer set did not identify any statistically significant covariate effects at the community level for occupancy or detection. All 95% credible intervals for predictors such as habitat type, temperature, salinity, and sampling time overlapped zero. However, at the species level, ASV_7 (roughtail stingray) exhibited strong positive detection effects. Filtration volume significantly improved detection probability (mean = 1.83, 95% CI: 0.314.54), and day of sampling had an even more pronounced effect (mean = 13.32, 95% CI: 5.6628.58). Other ASVs showed no statistically significant species-level occurrence or detection responses. 

Model diagnostics indicated overall good convergence: most parameters had Rhat values close to 1.00 and effective sample sizes (ESS) exceeded 200, suggesting adequate mixing. Posterior predictive checks using both chi-squared and Freeman-Tukey statistics showed Bayesian p-values near 0.250.28 at the community level and between 0.070.67 at the species level, indicating no strong evidence of model misfit. WAIC for the model was 72.33, with an effective number of parameters (pD) of 15.26, suggesting a moderately complex model given the small number of species (n = 6).

*Discussion*
These results from the non-spatial model suggest that most environmental covariates do not exert a consistent influence on elasmobranch occupancy or detection across the study area when spatial structure is ignored. The significant positive effect of filtration volume and sampling day on ASV_7 detection highlights how this species may be particularly sensitive to sampling effort and temporal variationfactors that likely affect short-term fluctuations in eDNA concentration.

However, the lack of other significant species-level responses, combined with wide credible intervals for many effects, underscores the model's limitations in resolving ecological signals from noise. While convergence diagnostics were generally strong, the modest Bayesian p-values from posterior predictive checks and the moderate WAIC suggest that the model may still be missing important structure in the data.

In particular, the assumption of independence between sites may obscure meaningful spatial patterns. Given the likely spatial autocorrelation in elasmobranch distributions and the fine-scale environmental gradients present in the estuarine system, transitioning to a spatial occupancy framework is warranted. Incorporating spatial random effects or spatially structured latent factors can improve the models ability to detect true ecological effects and provide more robust insight into the drivers of elasmobranch occurrence and detection dynamics.


#### Run spatial model (Multi-Species Spatial Occupancy)
Try running the multi species spatial model, `sfMsPGOcc`

Already have the coordinates in the data object as `coords` but there is an error when same location is sampled twice, even on different days. These are not true replicates but model does want coords repeated (considers them as same site?).

Solution-- use `grid.index` which tells the model to apply the spatial effect at the site level, even if there are multiple samples per site. Preserves temporal replication while satisfying the models spatial requirements.

```{r}
# Convert coords to data frame with column names (these are already projected in meters)
elas.exp.list$coords <- as.data.frame(elas.exp.list$coords)
colnames(elas.exp.list$coords) <- c("long", "lat")

# Create a single string identifier for each coordinate pair
coord_strings <- paste(elas.exp.list$coords$long, elas.exp.list$coords$lat)

# Factor to assign unique grid index numbers for each site
grid.index <- as.integer(as.factor(coord_strings))

# Extract unique coordinates in the same order as grid.index levels
coords_unique <- elas.exp.list$coords[!duplicated(coord_strings), ]

# Make sure coords_unique is in the order of the factor levels
coords_unique <- coords_unique[order(as.integer(factor(coord_strings[!duplicated(coord_strings)]))), ]

# Assign back to data list
elas.exp.list$grid.index <- grid.index # an integer vector mapping the 48 samples to 38 site coordinates
elas.exp.list$coords <- coords_unique  # Must match order of grid.index

```

Check 

```{r}
dim(elas.exp.list$y)[1] # species
dim(elas.exp.list$y)[2] # sites
dim(elas.exp.list$y)[3] # replicates
length(elas.exp.list$grid.index) # one site per sample
nrow(elas.exp.list$coords) #unique sites
```

```{r}
set.seed(20250615)

# Set MCMC parameters
n.batch <- 7500
batch.length <- 10
n.samples <- n.batch * batch.length

# Define priors and tuning
priors.list <- list(
  beta.comm.normal = list(mean = 0, var = 2.72),
  alpha.comm.normal = list(mean = 0, var = 2.72),
  tau.sq.beta.ig = list(a = 2, b = 2),
  tau.sq.alpha.ig = list(a = 2, b = 2)
)

tuning.list <- list(phi = rep(1, 2))  # 4 latent factors

# Lambda initialization
n.factors <- 2
n.species <- dim(elas.exp.list$y)[1]
lambda.inits <- matrix(0, nrow = n.species, ncol = n.factors)
diag(lambda.inits) <- 1
lambda.inits[lower.tri(lambda.inits)] <- rnorm(sum(lower.tri(lambda.inits)))

# Initial values
inits.list <- list(
  alpha.comm = 0,
  beta.comm = 0,
  beta = 0,
  alpha = 0,
  tau.sq.beta = 1,
  tau.sq.alpha = 1,
  phi = rep(3 / mean(dist(elas.exp.list$coords)), n.factors),
  lambda = lambda.inits,
  z = apply(elas.exp.list$y, c(1, 2), max, na.rm = TRUE)
)

# Run the model
sf.elas.out <- sfMsPGOcc(
  occ.formula = ~ Bayside + Habitat + Night__Day + 
                 scale(Avg_Depth_m) + scale(Avg_Temperature_C) + 
                 scale(Avg_Salinity_psu),
  det.formula = ~ scale(Volume_liter) + scale(Day) + scale(Start_Time),
  data = list(
    y = elas.exp.list$y,
    occ.covs = elas.exp.list$occ.covs,
    det.covs = elas.exp.list$det.covs,
    coords = elas.exp.list$coords,
    grid.index = elas.exp.list$grid.index
  ),
  inits = inits.list,
  priors = priors.list,
  tuning = tuning.list,
  cov.model = "exponential",
  NNGP = TRUE,
  n.neighbors = 20,
  n.factors = n.factors,
  n.batch = n.batch,
  batch.length = batch.length,
  n.burn = round(n.samples / 3),
  n.thin = 10,
  n.chains = 3,
  n.report = 750,
  n.omp.threads = 3,
  verbose = TRUE)
```



##### Save model
```{r}
# save(sf.elas.out, file = "figures-expedition/sf.elas.out")
# load("figures-expedition/sf.elas.out")
```

##### Examine output

Summarize results
I've been iteratively adjusting parameters in model above to optimize model convergence (Rhat, ESS), and SDs for means/ variances. Variances are no longer extreme (e.g., ~12, not 100+).
```{r}
# sink(file.path("figures-expedition", "model_out_spatial_elas_summary.txt"))
# summary(sf.elas.out)
# sink()

```

I am really happy with the version:

- Model Convergence Is Excellent All Rhat values are ~1.001.01  even species-level parameters.
- Effective Sample Size (ESS) is high across everything, including for variances and latent spatial parameters (phi).
- Spatial covariates (phi-1, phi-2) show robust convergence and precision.
- SDs for means/variances are low
- Variances are not extreme 
- Variability between species is evident but credible intervals remain sensible.
- Occurrence intercepts reflect realistic prevalence (e.g., 09)
- Detection estimates are species-specific but interpretable

Still, community-Level covariates have 95% credible intervals that include zero. This doesnt mean the covariates are unimportant, but it does mean that there are not strong community-wide trends. Instead, focus on species-specific effects 


*Specific Interesting Results*

Community-Level Effects:
No community-level occupancy or detection effects had 95% credible intervals that excluded zero. However, some trends are notable:
- Volume filtered had a positive mean effect on detection (0.88), with a CI approaching significance: [-0.50, 2.32].
- Day also had a strong positive effect on detection (mean = 1.18), but its CI overlapped zero: [-1.08, 3.29].


Species-Level Effects:
ASV_7 (Roughtail stingray): 
- Detection probability increased significantly with sampling day: Mean = 7.80, 95% CI = [3.63, 14.67]  - Statistically significant
Detection also increased with filtration volume: Mean = 1.39, 95% CI = [0.05, 3.03]  Marginally significant
ASV_14:
- Volume filtered had a moderately strong effect (1.36), but CI included zero.
All other species-level covariate effects (occupancy or detection) had wide CIs overlapping zero, indicating uncertainty.


Spatial Covariance:
- Spatial decay parameters (phi-1 and phi-2) are around 400 meters, indicating that spatial autocorrelation decays over relatively short distances.
- These represent the effective range of spatial autocorrelation: samples collected closer than ~400 meters tend to show more similar species occupancy patterns than those collected farther apart
- Interpretation: The presence or activity of an organism could potentially be detected up to ~400 meters away from where it actually was, due to water movement, mixing, or the persistence of eDNA in the water column. This doesnt mean eDNA always travels 400 meters, just that beyond that distance, sites become statistically uncorrelated in the model.

The 400m range could reflect biological processes (e.g., species distributions), eDNA transport, or shared environmental conditions  the model doesn't distinguish.

Model Diagnostics:
- Rhat values are all close to 1.0, indicating convergence.
- ESS values are sufficient for most parameters, although some intercept variances are higher than ideal.



*Text Summary*
In the spatial multi-species occupancy model, no covariates showed statistically significant community-level effects on occupancy, as all 95% credible intervals overlapped zero. At the detection level, none of the covariates reached statistical significance at the community level either, though filtration volume notably had a positive effect (mean = 0.878, 95% CI: 0.50 to 2.32, Rhat = 1.0013, ESS = 5634).

Species-level effects revealed more notable patterns. For ASV_7(roughtail stingray), both filtration volume and sampling day had significant positive effects on detection probability. Filtration volume had a mean effect of 1.39 (95% CI: 0.05 to 3.03, Rhat = 1.0024, ESS = 5004), and sampling day had a strong effect with a mean of 7.80 (95% CI: 3.63 to 14.67, Rhat = 1.0054, ESS = 2472), suggesting increased detectability later in the sampling period. Spatial covariance parameters phi-1 and phi-2 were estimated at 402.7 (95% CI: 19.8 to 780.5, Rhat = 1.0002, ESS = 14,297) and 400.1 (95% CI: 20.5 to 781.4, Rhat = 1.0001, ESS = 14,121), respectively, indicating a range of spatial correlation of approximately 400 meters.

*Discussion* The lack of statistically significant community-level effects in this model suggests that, across all elasmobranch taxa detected, occupancy patterns were not consistently driven by any single covariate, including habitat type or physical variables like depth, temperature, or salinity. However, the positive detection effect of filtration volume and strong day-of-sampling effect for the roughtail stingray suggest that detection probability for this species is sensitive to both sampling effort and temporal variation, possibly reflecting movement that increased detectability later in the sampling window.

Of particular interest is the spatial covariance estimate of approximately 400 meters, indicating the range over which spatial autocorrelation in occupancy patterns persists. This means that sites located more than 400 meters apart tend to exhibit distinct occupancy patterns, reflecting ecological or environmental heterogeneity that is not simply due to spatial proximity. Notably, this pattern emerges even for elasmobranchs, highly mobile species relative to many of the other taxa detected in our study. The fact that we still observe spatially distinct patterns in occupancy suggests that their distribution, or the persistence and transport of their eDNA signal, is shaped by localized environmental conditions or behaviors such as site fidelity or habitat preference. These findings validate the spatial resolution of our sampling and reinforce the importance of broad spatial coverage when monitoring even mobile or wide-ranging species with eDNA within a restricted waterbody such as a shallow water lagoon.


##### Check Model diagnostics
###### 1. Convergence diagnostics

```{r}
plot(sf.elas.out, param = "beta")  
plot(sf.elas.out, param = "alpha")
```

--> All traces look like "fuzzy caterpillars," i.e. overlap among 3 chains and nothing stands out too much from other chains. Density plots are unimodal


###### 2. Posterior predictive checks


```{r}
ppc.out1 <- ppcOcc(sf.elas.out, fit.stat = 'freeman-tukey', group = 1)
summary(ppc.out1)

ppc.out2 <- ppcOcc(sf.elas.out, fit.stat = 'freeman-tukey', group = 2)
summary(ppc.out2)

```


###### 3. WAIC for model comparison

```{r}
waicOcc(sf.elas.out)

```




### MiFish
Run libraries as separate models because there is some overlap in detected species- that would conflate things.

#### Prepare data
Use dfs where controls were already removed. Simplify metadata variables

```{r}
mifish_2024_exp_asv_table_taxbased
mifish_2024_exp_sample_data_taxbased
mifish_2024_exp_tax_table_taxbased
colnames(mifish_2024_exp_sample_data_taxbased)
```


Simplify sample names, remove redundant or unuseful metadata, remove controls (MP_C_4ES2_S25_L001,MP_C_5ES2_S25_L001)
```{r}
# Simplify sample names in count table
mifish_exp_df_counttable <- mifish_2024_exp_asv_table_taxbased %>% select(-c(MP_C_4ES2_S25_L001,MP_C_5ES2_S25_L001))
colnames(mifish_exp_df_counttable) <- str_extract(colnames(mifish_exp_df_counttable), "(?<=MP_)[0-9]+_[0-9]+")

# Simplify sample names in sample data table
mifish_exp_df_sampledata <- mifish_2024_exp_sample_data_taxbased %>% filter(!rownames(.) %in% c("MP_C_4ES2_S25_L001","MP_C_5ES2_S25_L001"))
rownames(mifish_exp_df_sampledata) <- str_extract(rownames(mifish_exp_df_sampledata), "(?<=MP_)[0-9]+_[0-9]+")

# Remove some columns in sample data
mifish_exp_df_sampledata <- mifish_exp_df_sampledata %>%
  select(
    -Name.Deploy_Cartr_Library,
    -Deployment,
    -Cartridge_ID,
    -Site_type,
    -site_altname,
    -controls,
    -Date,
    -End_Time__ET_,
    -Datecode,
    -Year,
    -Month,
    -Name.Deploy_Cartr)

mifish_exp_df_counttable
glimpse(mifish_exp_df_sampledata)

```







Convert counts to presence/ absence 
```{r}
mifish_exp_df_countmatrix <- mifish_exp_df_counttable
mifish_exp_df_countmatrix[] <- ifelse(mifish_exp_df_countmatrix > 0, 1, 0)

# compare
mifish_exp_df_counttable
mifish_exp_df_countmatrix
```


Occupancy and Detection Covariates
```{r}
# Occupancy covariates: one row per unique replicate group
# Don't include spatial variables because that is built into coords later and would lead to redundancy
occ.covs <- mifish_exp_df_sampledata %>%
  group_by(replicates) %>%
  summarize(across(c(Bayside, Habitat, Night__Day, Avg_Depth_m, Avg_Temperature_C, Avg_Salinity_psu, Avg_Chlorophyll_ugL, Avg_Turbidity_NTU), mean_or_mode)) %>%
  as.data.frame()
# remove rep names and use as IDs
rownames(occ.covs) <- occ.covs$replicates
occ.covs$replicates <- NULL


# Detection covariates: list of matrices for each sampling variable. Matrix contains sites and replicates
# Extract and sort by rep
det_grouped <- mifish_exp_df_sampledata[, c("replicates", "Volume_liter", "Start_Time__ET_", "Day")] %>%
  group_by(replicates) %>%
  arrange(replicates) %>%
  mutate(rep_id = row_number()) %>%
  ungroup()

# Reshape to wide format for each covariate
vol_mat <- det_grouped %>%
  select(replicates, rep_id, Volume_liter) %>%
  pivot_wider(names_from = rep_id, values_from = Volume_liter) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

day_mat <- det_grouped %>%
  select(replicates, rep_id, Day) %>%
  pivot_wider(names_from = rep_id, values_from = Day) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

# use decimal hour for time
start_time_mat <- det_grouped %>%
  select(replicates, rep_id, Start_Time__ET_) %>%
  mutate(Start_Time__ET_ = hms::as_hms(as.character(Start_Time__ET_)),
         Start_Time__ET_ = hour(Start_Time__ET_) + 
                           minute(Start_Time__ET_) / 60 + 
                           second(Start_Time__ET_) / 3600) %>%
  pivot_wider(names_from = rep_id, values_from = Start_Time__ET_) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

# Combine into a named list
det.covs <- list(
  Volume_liter = vol_mat,
  Day = day_mat,
  Start_Time = start_time_mat
)

occ.covs
det.covs
```

Drop strongly correlated covariates (leads to errors later)---
```{r}
# Get Pearson correlation matrix for numeric occupancy covariates
num_covs <- occ.covs[, sapply(occ.covs, is.numeric)]
cor_matrix <- cor(num_covs, use = "complete.obs")
round(cor_matrix, 2)
```
None are greater than 0.9. Keep all for now

Check categorical variables to see if any are represented in only 1-2 sites, which can cause errors
```{r}
table(occ.covs$Bayside)
table(occ.covs$Habitat)
```
Clam sanctuary and ousyer reef only represented 4 times each. May come back later and just categorize into "restored" as for Elasmos above. Keep an eye out for now

Change both "CLAM SANCTUARY" and "EELGRASS" to "RESTORED" so they will be one category
```{r}
# occ.covs$Habitat <- as.character(occ.covs$Habitat)  # convert to character if factor
# occ.covs$Habitat[occ.covs$Habitat %in% c("CLAM SANCTUARY", "EELGRASS")] <- "RESTORED"
# occ.covs$Habitat <- factor(occ.covs$Habitat)  # convert back to factor
# 
# occ.covs
```



Transform count matrix to a 3D array y[i,j,k], where:
i indexes species (ASVs),
j indexes sites (i.e., unique replicate groups),
k indexes replicate samples at that site.

Group samples based on replicates column. Reorganize so all replicates are aligned based on `occ.covs`
```{r}
# Unique replicate groups
replicate_groups <- rownames(occ.covs)

# Sort samples by replicate groups
mifish_exp_df_sampledata <- mifish_exp_df_sampledata[order(mifish_exp_df_sampledata$replicates), ]

# Initialize 3D y array
n_species <- nrow(mifish_exp_df_countmatrix)
n_sites <- length(unique(mifish_exp_df_sampledata$replicates)) # "sites" are unique to time/day/location
max_reps <- max(table(mifish_exp_df_sampledata$replicates))

y_array <- array(NA, dim = c(n_species, n_sites, max_reps))

# Populate the array
for (i in seq_along(replicate_groups)) {
  reps <- rownames(mifish_exp_df_sampledata)[mifish_exp_df_sampledata$replicates == replicate_groups[i]]
  for (j in seq_along(reps)) {
    y_array[, i, j] <- mifish_exp_df_countmatrix[, reps[j]]
  }
}

# Retain names
dimnames(y_array) <- list(
  rownames(mifish_exp_df_countmatrix),  # species (rows)
  unique(mifish_exp_df_sampledata$replicates),     # sites (columns)
  as.character(seq_len(dim(y_array)[3]))       # replicate number (slices)
)

data.frame(y_array[,,1])
data.frame(y_array[,,2])
```
I triple checked that these columns match up with the same order of sample/ replicates in the covariate matrices.



and coords- prject in meters
```{r}
# Create coords data frame for spatial transformation
coords_df <- mifish_exp_df_sampledata %>%
  group_by(replicates) %>%
  summarize(across(c(long, lat), mean_or_mode), .groups = "drop")  # .groups avoids grouping issues

# Convert to sf object
coords_sf <- st_as_sf(coords_df, coords = c("long", "lat"), crs = 4326)

# Transform to UTM Zone 18N
coords_utm <- st_transform(coords_sf, crs = 32618)

# Extract projected coordinates in meters
coords_projected <- st_coordinates(coords_utm)

# Set row names to match your occupancy covariates
rownames(coords_projected) <- rownames(occ.covs)


head(coords_projected)


```

Make into one list for spOccupancy
```{r}
# Final data list for spPGOcc
mifish.exp.list <- list(
  y = y_array,
  occ.covs = occ.covs,
  det.covs = det.covs,
  coords = coords_projected
)

str(mifish.exp.list)
```



Grid index the repeated coordinates
```{r}
# Convert coords to data frame with column names
mifish.exp.list$coords <- as.data.frame(mifish.exp.list$coords)
colnames(mifish.exp.list$coords) <- c("long", "lat")

# Create a single string identifier for each coordinate pair
coord_strings <- paste(mifish.exp.list$coords$long, mifish.exp.list$coords$lat)

# Factor to assign unique grid index numbers for each site
grid.index <- as.integer(as.factor(coord_strings))

# Extract unique coordinates in the same order as grid.index levels
coords_unique <- mifish.exp.list$coords[!duplicated(coord_strings), ]

# Make sure coords_unique is in the order of the factor levels
coords_unique <- coords_unique[order(as.integer(factor(coord_strings[!duplicated(coord_strings)]))), ]

# Assign back to data list
mifish.exp.list$grid.index <- grid.index # an integer vector mapping the 48 samples to 38 site coordinates
mifish.exp.list$coords <- coords_unique  # Must match order of grid.index

```


Check dimensions

```{r}
dim(mifish.exp.list$y)[1] # species
dim(mifish.exp.list$y)[2] # sites
dim(mifish.exp.list$y)[3] # replicates
length(mifish.exp.list$grid.index) # one site per sample
nrow(mifish.exp.list$coords) #unique sites
```

#### Run model (Multi-Species with Spatial Effects)

```{r}
# Set OPEN WATER as the reference level. This means the model will calculate occupancy at the other 3 Habitats relative to open water
mifish.exp.list$occ.covs$Habitat <- relevel(factor(mifish.exp.list$occ.covs$Habitat), ref = "OPEN WATER")


set.seed(20250615)

n.batch <- 7500
batch.length <- 10
n.samples <- n.batch * batch.length
n.factors <- 2
n.species <- dim(mifish.exp.list$y)[1]

priors.list <- list(
  beta.comm.normal = list(mean = 0, var = 2.72),
  alpha.comm.normal = list(mean = 0, var = 2.72),
  tau.sq.beta.ig = list(a = 2, b = 2),
  tau.sq.alpha.ig = list(a = 2, b = 2)
)

tuning.list <- list(phi = rep(1, n.factors))

lambda.inits <- matrix(0, nrow = n.species, ncol = n.factors)
for (k in 1:min(n.species, n.factors)) {
  lambda.inits[k, k] <- 1
}
lambda.inits[lower.tri(lambda.inits)] <- rnorm(sum(lower.tri(lambda.inits)))

inits.list <- list(
  alpha.comm = 0,
  beta.comm = 0,
  beta = 0,
  alpha = 0,
  tau.sq.beta = 1,
  tau.sq.alpha = 1,
  phi = rep(3 / mean(dist(mifish.exp.list$coords)), n.factors),
  lambda = lambda.inits,
  z = apply(mifish.exp.list$y, c(1, 2), max, na.rm = TRUE)
)

sf.mifish.out <- sfMsPGOcc(
  occ.formula = ~ Bayside + Habitat + Night__Day + 
                 scale(Avg_Depth_m) + scale(Avg_Temperature_C) + 
                 scale(Avg_Salinity_psu) + scale(Avg_Chlorophyll_ugL) + scale(Avg_Turbidity_NTU),
  det.formula = ~ scale(Volume_liter) + scale(Day) + scale(Start_Time),
  data = list(
    y = mifish.exp.list$y,
    occ.covs = mifish.exp.list$occ.covs,
    det.covs = mifish.exp.list$det.covs,
    coords = mifish.exp.list$coords,
    grid.index = mifish.exp.list$grid.index
  ),
  inits = inits.list,
  priors = priors.list,
  tuning = tuning.list,
  cov.model = "exponential",
  NNGP = TRUE,
  n.neighbors = 20,
  n.factors = n.factors,
  n.batch = n.batch,
  batch.length = batch.length,
  n.burn = round(n.samples / 3),
  n.thin = 10,
  n.chains = 3,
  n.report = 750,
  n.omp.threads = 3,
  verbose = TRUE
)

```



##### Save model
```{r}
# save(sf.mifish.out, file = "figures-expedition/sf.mifish.out")
# load("figures-expedition/sf.mifish.out")
```

##### Examine output

Summarize results
```{r}
# options(max.print = 10000)
# sink(file.path("figures-expedition", "model_out_mifish_spatial_summary.txt"))
# summary(sf.mifish.out)
# sink()
```


*Community Level*
Community-Level Effects
At the community level, several environmental covariates exhibited directional effects on species occupancy, though most estimates were associated with wide credible intervals. The only statistically supported relationship was a positive effect of temperature on occupancy across species, indicating a higher probability of occupancy in warmer waters.
                              Mean     SD    2.5%     50%  97.5%   Rhat ESS
scale(Avg_Temperature_C)    1.0618 0.4884  0.2239  1.0130 2.1361 1.0046 501

All other occupancy covariates had 95% credible intervals overlapping zero. Eelgrass habitats showed a positive effect, while oyster reefs showed a negative effect and depth showed a potential negative trend but none reached statistical significance.
                              Mean     SD    2.5%     50%  97.5%   Rhat ESS
HabitatEELGRASS             0.9625 0.9876 -0.9884  0.9503 2.9802 1.0018 885
HabitatOYSTER REEF         -1.8977 1.2255 -4.3440 -1.8872 0.4814 1.0047 777


Community-level detection probability was most strongly influenced by the start time of sampling, with later sampling times increasing detection likelihood. Filtration volume had a positive mean, thought it was not a significant effect at the community level.
                       Mean     SD    2.5%     50%   97.5%   Rhat  ESS
scale(Volume_liter)  0.1705 0.1520 -0.1178  0.1668  0.4796 1.0017 8070
scale(Start_Time)    0.3399 0.1520  0.0499  0.3355  0.6440 1.0012 5534


*Spatial Covariance*
The spatial latent factors showed strong spatial structuring, with estimated spatial decay parameters () of ~400 meters (phi-1: mean = 399.7, 95% CI: 17.9 to 782.1, Rhat = 1.0001; phi-2: mean = 400.1, 95% CI: 19.1 to 783.1, Rhat = 1.0000). This suggests that species occupancy patterns were autocorrelated at distances up to ~400 meters.
          Mean       SD    2.5%      50%    97.5%   Rhat   ESS
phi-1 399.7029 231.9743 17.9104 400.5571 782.1497 1.0001 12243
phi-2 400.0922 232.1426 19.0812 402.0617 783.1188 1.0000 14085




*Species Level*
There are A LOT of ASVs so first just filter out the ones for which CI does not overlap zero:

```{r}
# Get the summary output
sum_output <- capture.output(summary(sf.mifish.out))

# Locate Species Level section
species_header_line <- grep("^\tSpecies Level$", sum_output)
dashes_lines <- grep("^[-]{5,}$", sum_output)
start_line <- dashes_lines[dashes_lines > species_header_line][1]
end_line <- dashes_lines[dashes_lines > start_line][1]

# Extract the table body
species_occ_lines <- sum_output[(start_line + 2):(end_line - 2)]

# Use regex to extract Parameter and numeric values
pattern <- "^\\s*(.*?)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(\\d+\\.\\d+|NA)\\s+(\\d+|NA)\\s*$"

# Apply pattern line-by-line
parsed <- lapply(species_occ_lines, function(line) {
  if (grepl(pattern, line)) {
    matches <- regmatches(line, regexec(pattern, line))[[1]]
    return(matches[2:9])  # Drop the full match, keep capture groups
  } else {
    return(rep(NA, 8))
  }
})

# Convert to data frame
species_occ_df <- as.data.frame(do.call(rbind, parsed), stringsAsFactors = FALSE)
colnames(species_occ_df) <- c("Parameter", "Mean", "SD", "2.5%", "50%", "95%", "Rhat", "ESS")

# Convert appropriate columns to numeric
species_occ_df[, 2:8] <- lapply(species_occ_df[, 2:8], as.numeric)

# Filter for significant results (credible intervals do not include 0)
species_occ_sig <- species_occ_df %>%
  filter(`2.5%` > 0 | `95%` < 0)

species_occ_sig
```

Very few interesting ones are completely outside of non-overlapping 95% CIs. Instead, look variable by variable to pick out ASVs with interesting patterns.  Choose rows with 95% credible intervals that either do not include zero or are centered far above/below zero to warrant attention (i.e. wide interval is OK but strongly positive or negative mean)...

1) "INTERCEPT" is the baseline model variables (open water/ eastern bay)- 
- Species-level intercepts represent the baseline (i.e., average) occupancy probabilities on the logit scale, after accounting for all covariates. A positive intercept suggests high baseline occurrence, while a wide credible interval spanning zero or negative values indicates uncertainty or rare occurrence.

- While several ASVs have high and stable occupancy, some ASVs (e.g., ASV_17, ASV_31, ASV_44, ASV_95, ASV_248) showed low intercepts with wide credible intervals extending deep into negative values (e.g., ASV_95: Mean = -2.04, CI: -7.30 to 4.11), indicating rare or spatially restricted occupancy.
- All intercept estimates had acceptable Rhat values (<1.06), and most effective sample sizes (ESS) were >500, indicating good MCMC convergence for these parameters.



2) BaysideW
A large proportion of ASVs showed negative BaysideW effects, often with 95% credible intervals that remain negative throughout or strongly skewed negative:
- ASV_8: Mean = 2.29 (CI: 5.91 to 0.88)
- ASV_9: Mean = 2.43 (CI: 6.49 to 1.31)
- ASV_47: Mean = 2.55 (CI: 7.45 to 0.88)
- ASV_38: Mean = 2.46 (CI: 7.70 to 1.14)

Many ASVs had means close to zero and wide intervals that crossed zero, indicating no strong evidence of a difference in occupancy between western and the eastern reference sites:
- ASV_1: Mean = 0.56 (CI: 2.82 to 5.43)
- ASV_2: Mean = 0.49 (CI: 2.62 to 5.04)
- ASV_7: Mean = 0.39 (CI: 2.57 to 3.87)

Few Positive Shifts
- A small number of ASVs show weakly positive mean shifts in occupancy at western sites, but none have credible intervals that exclude zero. Thus, no ASVs exhibit strong, consistent increases in occupancy at Bayside sites.

3) HabitatCLAM SANCTUARY
- No species show statistically significant occupancy effects in the Clam Sanctuary habitat: all 95% credible intervals for the HabitatCLAM SANCTUARY coefficients overlap zero.
- Most species have weakly negative mean estimates, suggesting a potential trend toward lower occupancy in the Clam Sanctuary compared to the reference habitat (Open Water), but with high uncertainty.
- ASV_12 (Spot), 14 (Northern sennet), and 113 (Scombriformes) stand out slightly:
                                      Mean     SD    2.5%     50%   97.5%   Rhat  ESS
HabitatCLAM SANCTUARY-ASV_12        0.5147 1.7641 -2.6276  0.4090  4.3852 1.0037 1637
HabitatCLAM SANCTUARY-ASV_14        0.1770 1.6178 -2.8234  0.1026  3.5299 1.0050 2224
HabitatCLAM SANCTUARY-ASV_113       0.1705 1.7082 -3.0219  0.0919  3.8072 1.0017 1840

Although not statistically significant, they have positive posterior means, contrasting with the generally negative trend for other ASVs. Note, the strongest here (Spot) is a known predator of molluscs.

4) HabitatEELGRASS
- Strong and consistent positive associations: Every ASV has a positive posterior mean for the HabitatEELGRASS coefficient, indicating higher occupancy probability in eelgrass compared to the reference habitat.
- No ASVs show statistically significant differences (i.e., all 95% credible intervals include zero), but many hover near significance and have narrow, informative intervals--
- These ASVs have particularly high mean estimates and credible intervals that lean more strongly positive, suggesting a trend toward eelgrass preference:
                                      Mean     SD    2.5%     50%   97.5%   Rhat  ESS
HabitatEELGRASS-ASV_20              1.5036 1.5000 -1.3133  1.4559  4.6076 1.0009 2005
HabitatEELGRASS-ASV_17              1.4085 1.5445 -1.4978  1.3679  4.5733 1.0004 2018
HabitatEELGRASS-ASV_49              1.4353 1.6633 -1.6434  1.3677  4.9522 1.0018 1933
HabitatEELGRASS-ASV_352             1.3911 1.6231 -1.7332  1.3461  4.6870 1.0013 2150
HabitatEELGRASS-ASV_38              1.3184 1.5386 -1.5931  1.2892  4.4837 1.0002 2067
HabitatEELGRASS-ASV_95              1.3081 1.5885 -1.6968  1.2766  4.4646 1.0007 1953
HabitatEELGRASS-ASV_147             1.3054 1.5981 -1.7208  1.2702  4.6028 1.0020 2013
HabitatEELGRASS-ASV_4               1.1268 1.6099 -2.0547  1.1039  4.4177 1.0005 2062
HabitatEELGRASS-ASV_1               1.0512 1.6006 -2.0007  1.0241  4.2507 1.0006 2422

 1stRepresentative_ASV            CommonName
1                ASV_17     Northern pipefish
2                ASV_20             Anchovies
3                ASV_38 Atlantic bluefin tuna
4                ASV_49                  Scup
5                ASV_95           Skilletfish
6               ASV_147          Striped bass
7               ASV_352             Anchovies
           
5) HabitatOYSTER REEF
- Strong and consistent negative associations: All ASVs have large negative posterior means for the HabitatOYSTER REEF coefficient, indicating substantially lower occupancy in oyster reef habitats relative to the reference condition.
- The magnitude of these effects is larger than for any other habitat type, and the pattern is highly consistent across ASVs.

6) Night__DayNight
- Across all ASVs, posterior means ranged from approximately 0.42 to +0.48, with 95% credible intervals spanning wide ranges that reflect high uncertainty. Model diagnostics were consistently good (Rhat < 1.01, ESS > 1300), indicating convergence and sufficient sampling.
- The effects of sampling time (Night vs. Day) on elasmobranch occupancy showed no consistent directional trend across ASVs, though several taxa exhibited modest positive associations with night sampling. 
- Most Positive Effects (i.e. occupancy at night)
                                      Mean     SD    2.5%     50%   97.5%   Rhat  ESS
Night__DayNight-ASV_1               0.3769 1.1971 -1.8838  0.3387  2.8466 1.0019 2770
Night__DayNight-ASV_2               0.4858 1.1597 -1.7934  0.4681  2.8306 1.0025 2334
Night__DayNight-ASV_31              0.3755 1.2494 -2.0680  0.3458  2.9407 1.0059 2028
Night__DayNight-ASV_58              0.4022 1.2523 -2.0284  0.3807  2.9892 1.0103 1854

  1stRepresentative_ASV        CommonName
1                 ASV_1 Atlantic menhaden
2                 ASV_2     Scombriformes
3                ASV_31         Anchovies
4                ASV_58             Shads

- Most Negative Effect (i.e. occupancy in the day)
                                      Mean     SD    2.5%     50%   97.5%   Rhat  ESS
Night__DayNight-ASV_7              -0.4184 1.1930 -2.8300 -0.4084  1.9084 1.0088 1363
Night__DayNight-ASV_17             -0.3756 1.2625 -3.0218 -0.3439  2.0150 1.0024 2310

  1stRepresentative_ASV        CommonName
1                 ASV_7    Black sea bass
2                ASV_17 Northern pipefish


Note: Shads are considered nocturnal and feed at night (REF). Menhaden larvae swim into upper water column at night ([REF](https://www.sciencedirect.com/science/article/abs/pii/0022098196025841#:~:text=This%20migration%20pattern%20would%20result,tidal%20stream%20transport%20in%20estuaries.)).


7) scale(Avg_Depth_m)
- No species show a statistically significant effect of depth (i.e., 95% CI excluding zero), but a handful show suggestive trends, especially in the negative direction (i.e. a higher occupancy at sites with shallower depths).
- Most negative effect:
                                      Mean     SD    2.5%     50%   97.5%   Rhat  ESS
scale(Avg_Depth_m)-ASV_5           -1.7884 1.1687 -4.4600 -1.6655  0.1995 1.0038 1494

ASV_5 is Atlantic silverside



8) scale(Avg_Temperature_C)
- All species show positive mean coefficients, but the credible intervals overlap zero, indicating no statistically robust effect at the 95% CI level. 
- Some (e.g., ASV_14, ASV_64, ASV_58) show large effects and are candidates for further study or highlighting...

- Most positive effects:
                                      Mean     SD    2.5%     50%   97.5%   Rhat  ESS
scale(Avg_Temperature_C)-ASV_14     1.6767 0.9982 -0.1066  1.6142  3.8418 1.0000 1611
scale(Avg_Temperature_C)-ASV_21     1.4939 1.0859 -0.4618  1.4270  3.8385 1.0017 2129
scale(Avg_Temperature_C)-ASV_30     1.3641 1.0411 -0.5670  1.2996  3.6328 1.0004 1827
scale(Avg_Temperature_C)-ASV_38     1.3787 1.0079 -0.4315  1.3089  3.5822 1.0014 1984
scale(Avg_Temperature_C)-ASV_52     1.4938 1.0254 -0.3604  1.4210  3.7228 1.0008 2219
scale(Avg_Temperature_C)-ASV_58     1.5529 1.0222 -0.2609  1.4860  3.8088 1.0008 1741
scale(Avg_Temperature_C)-ASV_64     1.6590 1.0121 -0.1558  1.5980  3.8023 1.0002 1918

  1stRepresentative_ASV            CommonName
1                ASV_14       Northern sennet
2                ASV_21    Inshore lizardfish
3                ASV_30            Naked goby
4                ASV_38 Atlantic bluefin tuna
5                ASV_52   Atlantic needlefish
6                ASV_58                 Shads
7                ASV_64         Crevalle jack

 --> Some of these are tropical and/or at the northern most end of their geographic range (get more nuance here from Natalia). It makes sense that temperature would be most important for these species in our climatic zone (Long Island is in a transition zone between a humid subtropical climate and a hot-summer humid continental climate). Also the tuna are likely from restaurants and not actual live residents. They might correlate with temperature because runoff water is warmer...
 
9) scale(Avg_Salinity_psu)
- Variable repsonses across species: Some have positive nad some have negative relationship with salinity (makes sense in an estuarine environment).

- Most positive effects:
ASV	  Mean      95% CI
ASV_7	+2.38	[0.28, 5.40]	Strongest positive effect AND significant
ASV_17	+1.52	[-0.79, 4.61]	High positive mean, wide CI
ASV_20	+1.44	[-0.61, 4.30]	Substantial positive effect
ASV_31	+1.27	[-1.12, 4.31]	High mean, CI overlaps 0
ASV_67	+1.14	[-1.44, 4.40]	High upper CI bound
ASV_4	+1.09	[-0.89, 3.32]	Strong central tendency

  1stRepresentative_ASV        CommonName
1                 ASV_4     Seaboard goby
2                 ASV_7    Black sea bass
3                ASV_17 Northern pipefish
4                ASV_20         Anchovies
5                ASV_31         Anchovies
6                ASV_67    Smooth dogfish

- Most negative effect:
ASV	      Mean	  95% CI	
ASV_47	1.09	[3.45, 1.15]	

ASV_47 is Bigeye tuna


--> According to Google search, seaboard goby can live in salinities 0 to 37 and black sea bass prefers higher salinity (>10) but this result is notable/ strange because the range of salinities here is only 27-31 psu.
This suggests that salinity is an important structuring gradient, potentially differentiating communities across the Bay, from inlet to upper-bay locations. BUT given some strange associations, I wonder if salinity is representative of any other variable (mixing? proximity to river? I am not sure what to think here.)

10) scale(Avg_Chlorophyll_ugL)
- All of the ASVs show negative associations with chlorophyll concentration, suggesting reduced occupancy at sites with elevated chlorophyll. 

--> Despite wide credible intervals, the overwhelming directional consensus suggests a community-wide negative response to increasing chlorophyll-a concentrations. This could reflect lower occupancy of key ASVs in bloom-prone areas, potentially due to hypoxia, turbidity, or biological interactions associated with eutrophic conditions. These patterns may indicate either direct ecological stressors from high chlorophyll (e.g., altered light, oxygen, or toxin levels), or habitat differentiation, where these ASVs are associated with less nutrient-enriched zones. The consistency in direction across dozens of ASVs, even if modest, adds weight to this interpretation, especially when interpreted alongside community-level trends.


11) scale(Avg_Turbidity_NTU)
- Most ASVs trend negative, reinforcing the idea that higher turbidity is broadly detrimental to fish occupancy in this system.

- Most negative effects:
ASV   Effect	  95% CI
ASV_6	1.55	[4.65, 0.72]	Strongest negative trend, wide CI
ASV_12	1.21	[4.19, 1.02]	Strongly negative and consistent with others
ASV_4	1.18	[3.98, 1.01]	Narrower CI, similarly strong effect
ASV_20	0.96	[3.65, 1.22]	Possibly sensitive to suspended solids

  1stRepresentative_ASV    CommonName
1                 ASV_4 Seaboard goby
2                 ASV_6     Anchovies
3                ASV_12          Spot
4                ASV_20     Anchovies

--> Across the dataset, the negative skew is consistent, suggesting that turbidity may be broadly suppressing occupancy in a large subset of the community
--> Turbidity yielded a more pronounced and coherent directional signal than chlorophyll or night/day. Nearly all ASVs trend negatively with turbidity, and several, such as ASVs 6, 4, 12, and 20, show large effect sizes and credible intervals skewed well below zero, despite remaining technically inclusive of it. This pattern suggests that turbidity may be a key driver of species turnover or suppression in the system. The ecological explanation could involve 1) reduced detectability of eDNA due to particle binding or sediment interference, 2) physiological stress from high suspended solids, or 3) indirect exclusion due to habitat degradation or shifts in trophic interactions, or 4) a negative association due to mode of feeding for filter feeders. 

- Positive effects are much weaker (highest mean is +0.45)


12) scale(Volume_liter)
- Most ASVs show a small, positive central estimate for the effect of volume (i.e., better detection with more water filtered). However, wide credible intervals mean most effects are not statistically supported.

- Strongest positive effects: Only 1 ASV had a lower CI above zero:
ASV	  Mean	  95% CI	
ASV_9	0.81	[0.00, 1.93]	

ASV_9 is Tautog

- Strongest negative effect: Only 1 ASV had an upper CI below zero:
ASV	    Mean	  95% CI	
ASV_26	0.90	[1.80, 0.09]

ASV_26 is Smallmouth flounder


13) scale(Day)
- No ASV showed a credible interval entirely above or below zero, so all effects should be interpreted with caution. The highest positive value was only 0.47 and strongest negative value was only -0.42.

14) scale(Start_Time)
- ASVs Most Positively Associated with Later Sampling Start Times: These taxa showed increased detection probabilities when sampling occurred later in the day. Some have 95% credible intervals entirely above zero, suggesting credible positive associations:

ASV   Effect	95% CI
ASV_4	0.919	[0.24, 1.70]	Strongest positive effect; CI > 0 
ASV_5	0.754	[0.00, 1.65]	Positive trend, CI just crosses 0
ASV_12	0.683	[0.03, 1.51]	CI nearly excludes 0

  1stRepresentative_ASV          CommonName
1                 ASV_4       Seaboard goby
2                 ASV_5 Atlantic silverside
3                ASV_12                Spot

- Only one ASV had a clearly negative mean effect, but no credible intervals exclude 0:
ASV     Effect	    95% CI
ASV_30	0.274	[1.19, 0.59]

ASV_30 is Naked goby



##### Check Model diagnostics
###### 1. Convergence diagnostics

```{r}
plot(sf.mifish.out, param = "beta")  
plot(sf.mifish.out, param = "alpha")
```

--> All traces look like "fuzzy caterpillars," i.e. overlap among 3 chains and nothing stands out too much from other chains. Density plots are unimodal


###### 2. Posterior predictive checks


```{r}
ppc.out1 <- ppcOcc(sf.mifish.out, fit.stat = 'freeman-tukey', group = 1)
summary(ppc.out1)

ppc.out2 <- ppcOcc(sf.mifish.out, fit.stat = 'freeman-tukey', group = 2)
summary(ppc.out2)

```


###### 3. WAIC for model comparison

```{r}
waicOcc(sf.mifish.out)

```





### CO1


#### Prepare data
Use dfs where controls were already removed. Simplify metadata variables

```{r}
co1_2024_exp_asv_table_taxbased
co1_2024_exp_sample_data_taxbased
co1_2024_exp_tax_table_taxbased
colnames(co1_2024_exp_sample_data_taxbased)
```

Import and join CTD values averaged over each eDNA sampling interval (was only done for Elas and MiFish above)
```{r}
# rename df and remove controls
co1_exp_df_sampledata <- co1_2024_exp_sample_data_taxbased  %>% filter(is.na(controls))

# Store original row names
rownames_original <- rownames(co1_exp_df_sampledata)

# Simplify sample names in sample data table
co1_exp_df_sampledata$Name_Deploy_Cartr <- gsub("_$", "", gsub("\\.", "_", trimws(co1_exp_df_sampledata$Name_Deploy_Cartr)))

# import
sample_metadata2 <- read.csv("/Volumes/easystore/eDNA/shirp-edna/figures-expedition/CTD_avg_eDNA_sample_intervals.csv")

# join
co1_exp_df_sampledata <- left_join(co1_exp_df_sampledata, sample_metadata2, by = c("Name_Deploy_Cartr" = "Name.Deploy_Cartr"))

# Restore row names (for now)
rownames(co1_exp_df_sampledata) <- rownames_original

co1_exp_df_sampledata
```



Simplify sample names, remove redundant or unuseful metadata, remove controls. Data and sample ID format is slightly different than for mifish/ elas02
```{r}
# Rename
co1_exp_df_counttable <- co1_2024_exp_asv_table_taxbased
# Subset to keep only columns present in the filtered sampledata
co1_exp_df_counttable <- co1_exp_df_counttable[, rownames(co1_exp_df_sampledata), drop = FALSE]
# Rename the columns of the count table using the simplified names (note- these were in the same order)
colnames(co1_exp_df_counttable) <- co1_exp_df_sampledata$Name_Deploy_Cartr

# back to sample data- make sample names row names and drop long weird sample names
rownames(co1_exp_df_sampledata) <- co1_exp_df_sampledata$Name_Deploy_Cartr

# Remove some redundant columns in sample data
co1_exp_df_sampledata <- co1_exp_df_sampledata %>%
  select(
    -Name_Deploy_Cartr,
    -Deployment,
    -Cartridge_ID,
    -Site_type,
    -site_altname,
    -Station,
    -controls,
    -Date,
    -End_Time__ET_,
    -Datecode,
    -Year,
    -Month,
    -DO,
    -Temperature,
    -Salinity,
    -Time,
    -Weather)


# view
co1_exp_df_counttable
co1_exp_df_sampledata
glimpse(co1_exp_df_sampledata)

```







Convert counts to presence/ absence 
```{r}
co1_exp_df_countmatrix <- co1_exp_df_counttable
co1_exp_df_countmatrix[] <- ifelse(co1_exp_df_countmatrix > 0, 1, 0)

# compare
co1_exp_df_counttable
co1_exp_df_countmatrix
```


Occupancy and Detection Covariates
```{r}
# Occupancy covariates: one row per unique replicate group
# Don't include spatial variables because that is built into coords later and would lead to redundancy
occ.covs <- co1_exp_df_sampledata %>%
  group_by(replicates) %>%
  summarize(across(c(Bayside, Habitat, Night__Day, Avg_Depth_m, Avg_Temperature_C, Avg_Salinity_psu, Avg_Chlorophyll_ugL, Avg_Turbidity_NTU), mean_or_mode)) %>%
  as.data.frame()
# remove rep names and use as IDs
rownames(occ.covs) <- occ.covs$replicates
occ.covs$replicates <- NULL


# Detection covariates: list of matrices for each sampling variable. Matrix contains sites and replicates
# Extract and sort by rep
det_grouped <- co1_exp_df_sampledata[, c("replicates", "Volume_liter", "Start_Time__ET_", "Day")] %>%
  group_by(replicates) %>%
  arrange(replicates) %>%
  mutate(rep_id = row_number()) %>%
  ungroup()

# Reshape to wide format for each covariate
vol_mat <- det_grouped %>%
  select(replicates, rep_id, Volume_liter) %>%
  pivot_wider(names_from = rep_id, values_from = Volume_liter) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

day_mat <- det_grouped %>%
  select(replicates, rep_id, Day) %>%
  pivot_wider(names_from = rep_id, values_from = Day) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

# use decimal hour for time
start_time_mat <- det_grouped %>%
  select(replicates, rep_id, Start_Time__ET_) %>%
  mutate(Start_Time__ET_ = hms::as_hms(as.character(Start_Time__ET_)),
         Start_Time__ET_ = hour(Start_Time__ET_) + 
                           minute(Start_Time__ET_) / 60 + 
                           second(Start_Time__ET_) / 3600) %>%
  pivot_wider(names_from = rep_id, values_from = Start_Time__ET_) %>%
  column_to_rownames("replicates") %>%
  as.matrix()

# Combine into a named list
det.covs <- list(
  Volume_liter = vol_mat,
  Day = day_mat,
  Start_Time = start_time_mat
)

occ.covs
det.covs
```

Drop strongly correlated covariates (leads to errors later)---
```{r}
# Get Pearson correlation matrix for numeric occupancy covariates
num_covs <- occ.covs[, sapply(occ.covs, is.numeric)]
cor_matrix <- cor(num_covs, use = "complete.obs")
round(cor_matrix, 2)
```
None are greater than 0.9. Keep all for now

Check categorical variables to see if any are represented in only 1-2 sites, which can cause errors
```{r}
table(occ.covs$Bayside)
table(occ.covs$Habitat)
```
Clam sanctuary and ousyer reef only represented 4 times each. May come back later and just categorize into "restored" as for Elasmos above. Keep an eye out for now

Change both "CLAM SANCTUARY" and "EELGRASS" to "RESTORED" so they will be one category
```{r}
# occ.covs$Habitat <- as.character(occ.covs$Habitat)  # convert to character if factor
# occ.covs$Habitat[occ.covs$Habitat %in% c("CLAM SANCTUARY", "EELGRASS")] <- "RESTORED"
# occ.covs$Habitat <- factor(occ.covs$Habitat)  # convert back to factor
# 
# occ.covs
```



Transform count matrix to a 3D array y[i,j,k], where:
i indexes species (ASVs),
j indexes sites (i.e., unique replicate groups),
k indexes replicate samples at that site.

Group samples based on replicates column. Reorganize so all replicates are aligned based on `occ.covs`
```{r}
# Unique replicate groups
replicate_groups <- rownames(occ.covs)

# Sort samples by replicate groups
co1_exp_df_sampledata <- co1_exp_df_sampledata[order(co1_exp_df_sampledata$replicates), ]

# Initialize 3D y array
n_species <- nrow(co1_exp_df_countmatrix)
n_sites <- length(unique(co1_exp_df_sampledata$replicates)) # "sites" are unique to time/day/location
max_reps <- max(table(co1_exp_df_sampledata$replicates))

y_array <- array(NA, dim = c(n_species, n_sites, max_reps))

# Populate the array
for (i in seq_along(replicate_groups)) {
  reps <- rownames(co1_exp_df_sampledata)[co1_exp_df_sampledata$replicates == replicate_groups[i]]
  for (j in seq_along(reps)) {
    y_array[, i, j] <- co1_exp_df_countmatrix[, reps[j]]
  }
}

# Retain names
dimnames(y_array) <- list(
  rownames(co1_exp_df_countmatrix),  # species (rows)
  unique(co1_exp_df_sampledata$replicates),     # sites (columns)
  as.character(seq_len(dim(y_array)[3]))       # replicate number (slices)
)

data.frame(y_array[,,1])
data.frame(y_array[,,2])
data.frame(y_array[,,3])

```
I triple checked that these columns match up with the same order of sample/ replicates in the covariate matrices.



and coords
```{r}
# Create coords data frame for spatial transformation
coords_df <- co1_exp_df_sampledata %>%
  group_by(replicates) %>%
  summarize(across(c(long, lat), mean_or_mode), .groups = "drop")  # .groups avoids grouping issues

# Convert to sf object
coords_sf <- st_as_sf(coords_df, coords = c("long", "lat"), crs = 4326)

# Transform to UTM Zone 18N
coords_utm <- st_transform(coords_sf, crs = 32618)

# Extract projected coordinates in meters
coords_projected <- st_coordinates(coords_utm)

# Set row names to match your occupancy covariates
rownames(coords_projected) <- rownames(occ.covs)


head(coords_projected)


```

Make into one list for spOccupancy
```{r}
# Final data list for spPGOcc
co1.exp.list <- list(
  y = y_array,
  occ.covs = occ.covs,
  det.covs = det.covs,
  coords = as.matrix(coords_projected)
)

str(co1.exp.list)
```



Grid index the repeated coordinates
```{r}
# Convert coords to data frame with column names
co1.exp.list$coords <- as.data.frame(co1.exp.list$coords)
colnames(co1.exp.list$coords) <- c("long", "lat")

# Create a single string identifier for each coordinate pair
coord_strings <- paste(co1.exp.list$coords$long, co1.exp.list$coords$lat)

# Factor to assign unique grid index numbers for each site
grid.index <- as.integer(as.factor(coord_strings))

# Extract unique coordinates in the same order as grid.index levels
coords_unique <- co1.exp.list$coords[!duplicated(coord_strings), ]

# Make sure coords_unique is in the order of the factor levels
coords_unique <- coords_unique[order(as.integer(factor(coord_strings[!duplicated(coord_strings)]))), ]

# Assign back to data list
co1.exp.list$grid.index <- grid.index # an integer vector mapping the samples to site coordinates
co1.exp.list$coords <- as.matrix(coords_unique)  # Must match order of grid.index

```


Check dimensions

```{r}
dim(co1.exp.list$y)[1] # species
dim(co1.exp.list$y)[2] # samples
dim(co1.exp.list$y)[3] # replicates
length(co1.exp.list$grid.index) # one site per sample
nrow(co1.exp.list$coords) #unique locations
```
Note- CO1 has many more ASVs than MiFish or Elas02. Might need to adjust some parameters to get reasonable run times...


#### Run model (Multi-Species with Spatial Effects)
I changed some parameters here to make the model a little bit faster since this dataset is much bigger. Keep iteratively adjusting the batch length, the n factors, the n batch, and the number of neighbors in order to optimize R-hat and ESS and balance that with run time. 



```{r}
# Set OPEN WATER as the reference level. This means the model will calculate occupancy at the other 3 Habitats relative to open water
co1.exp.list$occ.covs$Habitat <- relevel(factor(co1.exp.list$occ.covs$Habitat), ref = "OPEN WATER")


set.seed(20250615)

n.batch <- 7500
batch.length <- 10
# n.samples <- 75000
n.factors <- 1
n.thin <-  10
n.burn <-  25000
n.species <- dim(co1.exp.list$y)[1]

# keep phi in reasonable range to help convergence
a <- 1 / max(dist(co1.exp.list$coords))
b <- 1 / min(dist(co1.exp.list$coords[!duplicated(co1.exp.list$grid.index), ]))

priors.list <- list(
  beta.comm.normal = list(mean = 0, var = 2.72),
  alpha.comm.normal = list(mean = 0, var = 2.72),
  tau.sq.beta.ig = list(a = 2, b = 2),
  tau.sq.alpha.ig = list(a = 2, b = 2)
#  ,phi.unif = list(a = a, b = b) Actually when NGGP = TRUE, the models sets its own phi bounds and sfMsPGOcc is currently only implemented for NNGPs, not full Gaussian Processes. NNGP has to be TRUE
)

tuning.list <- list(phi = rep(1, n.factors))

lambda.inits <- matrix(0, nrow = n.species, ncol = n.factors)
for (k in 1:min(n.species, n.factors)) {
  lambda.inits[k, k] <- 1
}
lambda.inits[lower.tri(lambda.inits)] <- rnorm(sum(lower.tri(lambda.inits)))

inits.list <- list(
  alpha.comm = 0,
  beta.comm = 0,
  beta = 0,
  alpha = 0,
  tau.sq.beta = 1,
  tau.sq.alpha = 1,
  phi = rep(3 / mean(dist(co1.exp.list$coords)), n.factors),
  lambda = lambda.inits,
  z = apply(co1.exp.list$y, c(1, 2), max, na.rm = TRUE)
)

sf.co1.out <- sfMsPGOcc(
  occ.formula = ~ Bayside + Habitat + Night__Day + 
                 scale(Avg_Depth_m) + scale(Avg_Temperature_C) + 
                 scale(Avg_Salinity_psu) + scale(Avg_Chlorophyll_ugL) + scale(Avg_Turbidity_NTU),
  det.formula = ~ scale(Volume_liter) + scale(Day) + scale(Start_Time),
  data = list(
    y = co1.exp.list$y,
    occ.covs = co1.exp.list$occ.covs,
    det.covs = co1.exp.list$det.covs,
    coords = co1.exp.list$coords,
    grid.index = co1.exp.list$grid.index
  ),
  inits = inits.list,
  prior = priors.list,
  tuning = tuning.list,
  cov.model = "exponential",
  NNGP = TRUE,
  n.neighbors = 20,
  n.factors = n.factors,
  n.batch = n.batch,
  batch.length = batch.length,
  n.burn = n.burn,
  n.thin = n.thin,
  n.chains = 3,
  n.report = 750,
  n.omp.threads = 4,
  verbose = TRUE
)

```







##### Save model
```{r}
# save(sf.co1.out, file = "figures-expedition/sf.co1.out")
# load("figures-expedition/sf.co1.out")
```


##### Examine output

Summarize results
```{r}
options(max.print = 100000)
sink(file.path("figures-expedition", "model_out_co1_spatial_summary.txt"))
summary(sf.co1.out)
sink()
```




There are A LOT of ASVs so first just filter out the ones for which CI does not overlap zero:

```{r}
# Get the summary output
options(max.print = 100000)
sum_output <- capture.output(summary(sf.co1.out))

# Locate Species Level section
species_header_line <- grep("^\tSpecies Level$", sum_output)
dashes_lines <- grep("^[-]{5,}$", sum_output)
start_line <- dashes_lines[dashes_lines > species_header_line][1]
end_line <- dashes_lines[dashes_lines > start_line][1]

# Extract the table body
species_occ_lines <- sum_output[(start_line + 2):(end_line - 2)]

# Use regex to extract Parameter and numeric values
pattern <- "^\\s*(.*?)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(-?\\d+\\.\\d+|NA)\\s+(\\d+\\.\\d+|NA)\\s+(\\d+|NA)\\s*$"

# Apply pattern line-by-line
parsed <- lapply(species_occ_lines, function(line) {
  if (grepl(pattern, line)) {
    matches <- regmatches(line, regexec(pattern, line))[[1]]
    return(matches[2:9])  # Drop the full match, keep capture groups
  } else {
    return(rep(NA, 8))
  }
})

# Convert to data frame
species_occ_df <- as.data.frame(do.call(rbind, parsed), stringsAsFactors = FALSE)
colnames(species_occ_df) <- c("Parameter", "Mean", "SD", "2.5%", "50%", "95%", "Rhat", "ESS")

# Convert appropriate columns to numeric
species_occ_df[, 2:8] <- lapply(species_occ_df[, 2:8], as.numeric)

# Filter for significant results (credible intervals do not include 0)
species_occ_sig <- species_occ_df %>%
  filter(`2.5%` > 0 | `95%` < 0)

species_occ_sig

# save as file
write.table(species_occ_sig, 
            file = "figures-expedition/model_out_co1_spatial_summary_95sig_sp_level_only.txt",
            sep = "\t", 
            row.names = FALSE, 
            quote = FALSE)
```

Pull out sig results and match to taxonomy tables to see what these ASVs are:
```{r}
# Remove "Intercept" rows- these are baseline species level results and don't say anything about the impacts of env parameters on the ASVs)
species_occ_sig <- species_occ_sig %>%
  filter(!str_detect(Parameter, "Intercept")) %>%
  mutate(
    ASV = str_extract(Parameter, "ASV_\\d+"),
    suffix = str_extract(Parameter, "_[abc]$") %>% str_remove("_"),
    covariate = str_remove(Parameter, "-ASV_\\d+_[abc]$")
  )

# Read in CO1 taxonomy files
tax_a <- read_tsv("results-revamp-2024-CO1-expeditiona_Chordata_reBLAST/results-revamp-2024-CO1-expeditiona_Updated_asvTaxonomyTable.txt") %>%
  mutate(suffix = "a")

tax_b <- read_tsv("results-revamp-2024-CO1-expeditionb_Chordata_reBLAST/results-revamp-2024-CO1-expeditionb_Updated_asvTaxonomyTable.txt") %>%
  mutate(suffix = "b")

tax_c <- read_tsv("results-revamp-2024-CO1-expeditionc_Chordata_reBLAST/results-revamp-2024-CO1-expeditionc_Updated_asvTaxonomyTable.txt") %>%
  mutate(suffix = "c")

# Combine all tax tables
all_tax <- bind_rows(tax_a, tax_b, tax_c)

# Merge significant results with taxonomy
sig_results_tax <- species_occ_sig %>%
  left_join(all_tax, by = c("ASV", "suffix"))

# Reorder to inspect
sig_results_tax <-
  sig_results_tax %>%
  mutate(sign = ifelse(Mean >= 0, "positive", "negative"),
    abs_mean = abs(Mean)) %>%
  arrange(covariate, sign, desc(abs_mean)) %>%
  select(Parameter, covariate, sign, Mean, abs_mean, ASV, suffix, Kingdom:Species, everything(), -abs_mean)%>%
  View()  

```



"Western bay" variable:

86 ASVs have strong negative ocurrence patterns in the west (i.e. preference for the east). Many are phytoplankton or macroalgae (diatoms, dinoflagellates, haptophytes, chlorophytes, rhodophytes, raphidophytes, streptophytes) as well as other protists (e.g. rotifers, amoebas, ). Here are some of the more interesting ones:

- ASV_1860_a, Order Actiniaria (Sea anemones!)- overall strongest association with eastern bay
- ASV_15697_a (order Octopoda)
- ASV_480_b Mercenaria mercenaria (hard clam)- interesting because west is where the clam sanctuaries are!
- ASV_1967_c Mytilus trossulus - Pacific blue mussel that is also found in Maine and Canada. Can hybridize with Mytilus edulis
- ASV_509_c, Macoploma tenta, (Elongated macoma clam)
- ASV_1247_c, Petricolaria pholadiformis (False angelwing clam)
- ASV_1684_a, Sclerodactyla briareus (Hairy sea cucumber)
- ASV_3903_a, Styela canopus (Rough Sea Squirt)
- ASV_613_c, Anisolpidiales (Order of fungus-like parasites of algae)
- Potential HABs: 
  - ASV_13_a Aureococcus anophagefferens, ASV_10303_b (genus Aureococcus), ASV_9042_a (Aureococcus family)
  - ASV_610_c, Pseudochattonella farcimen (icthyotoxic)
  - ASV_185_c, Fibrocapsa japonica
  - ASV_801_c, Eucampia zodiacus (harmful to seaweeds?)
  - ASV_544_c, Takayama genus (in the same family as Karenia brevis)
  - ASV_2008_a, Odontella regia, harmful diatom
  - ASV_2229_c, Gymnodiniaceae family contains several HAB species (but not further identified)
  - ASV_2908_a, Phaeocystaceae family contains several HAB species (but not further identified)


Only 4 ASVs have a strong positive occurrence (preference for the west)

- ASV_1036_a Gemma gemma (Amethyst gem clam; competitor with other small bivalves/ clams- [wins out in competition in the presence of green crab](https://invasions.si.edu/nemesis/species_summary/81511))
- ASV_134_a Haminella solitaria (Solitary glassy bubble snail)
- ASV_1092_a Cylindrotheca genus (diatom)
- ASV_923_a Grateloupia genus - [likely invasive red algae](https://invasions.si.edu/nemesis/species_summary/12572), species turuturu?


Only 3 species had strong positive occurrence with Eelgrass habitat

- ASV_694_a Cheilostomatida, order of bryozoans ([there is at least 1 species in this order that is an epibiont of eelgrass](https://bioone.org/journals/zoological-science/volume-32/issue-5/zs150079/Cribrilina-mutabilis-n-sp-an-Eelgrass-Associated-Bryozoan-Gymnolaemata/10.2108/zs150079.full))
- ASV_4483_a, Anaulales, order of diatoms
- ASV_3893_a, Entomophthoromycetes, rare fungal infection of humans (ew)- tropical disease


292 ASVs had a strong negative occurrence with Night (e.g. preference for the day)- makes logical sense for photosynthesizers. The majority of these are phytoplankton. Some species of note:

- HABs:
  - ASV_1197_c, Levanderina fissa
  - ASV_276_a, Takayama acrotrocha
  - ASV_896_a, Phaeocystis pouchetii
  - ASV_45_a, Biecheleria cincta
  - ASV_185_c, Fibrocapsa japonica
  - ASV_94_a, Nusuttodinium genus- contains some harmful species
  - ASV_610_c, Pseudochattonella farcimen (icthyotoxic)


No significant positive occurences at night


1 ASV had a significant positive occurence with Chlorophyll-

- ASV_2356_a, Toxariales, an order of diatoms


No significant negative occurences with Chlorophyll


3 ASVs had a significant negative occurrence with salinity-
- ASV_4654_c (Clathria prolifera) - [a sponge that has a wide salinity tolerance, including down to 15 psu](https://invasions.si.edu/nemesis/species_summary/47997)
- ASV_708_b (Halichondriidae family of sponges)- also seem to have wide salinity tolerance
- ASV_1572_b Chloropicales - order of chlorophytes/ green algae. Some species can thrive at lower salinities (10-15 psu)


128 ASVs had a significant positive occurrence with salinity. These were all over the phylogenetic tree.- Phytoplankton, sponges, arthropods, fungi, bivalves and other mollusks, protists, cnidaria, ctenophores, annelids, etc etc


No significant negative occurences with temperature


74 ASVs had a significant positive occurrence with temperature. Similar to salinity, these were all over the phylogenetic tree. Some can be considered tropical or to favor warmer waters, for example:

- ASV_35_c Mnemiopsis leidyi, comb jelly - wide temperature tolerance but range [is spreading due to climate change](https://www.sciencedirect.com/science/article/pii/S0272771419305955)
- ASV_276_a, Takayama acrotrocha- also a HAB dinoflagellate. [Blooms are driven by high SST/ low SS](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=5282212)
- ASV_1823_a	Gracilaria tikvahiae (red alga)	[Native to the western Atlantic tropics/sub-tropics; commercial cultivation in Florida & the Caribbean at 25  30 C.	[large termperature tolerance and indicative of eutrophication](https://www.hawaii.edu/reefalgae/invasive_algae/pdf%20files/gracilaria_tikvahiae.pdf)
- ASV_2622_a	Hypnea musciformis (red alga)	[Cosmopolitan but considered a tropical macroalga; optimum growth temperatures 25-30C](https://www.cabidigitallibrary.org/doi/full/10.5555/20033183355)- can't get full text of this book chapter but cited [here](https://www.tandfonline.com/doi/full/10.1080/26388081.2021.2011412#d1e166)
- ASV_736 _c	Lithodesmium intricatum (diatom)	[Cosmopolitan; found in warm to temperate water](https://www.inaturalist.org/guide_taxa/1636894#:~:text=Classification%204,CC%20BY%2DNC%2DSA))
- ASV_614_c	Chaetoceros tenuissimus (diatom) [Considered a tropical species and is highly adaptable to futher warming](https://www.nature.com/articles/s41598-018-36091-y)
- ASV_913_c	Thoracosphaera heimii (calcifying dinophyte)	[Found in Sargasso Sea, Gulf of Mexico, equatorial Atlantic. Used as a warm-water/paleo-tropical marker in marine sediments](https://www.sciencedirect.com/science/article/abs/pii/0377839882900020)
- ASV_112_a	Skeletonema menzelii (diatom)	[Recorded mainly from warm coastal waters of the Atlantic, Mediterranean, and S America.](https://www.sciencedirect.com/science/article/pii/S1434461007000739#fig4);  [Among the Skeletonema, has ability to grow at very high temperatures (35C)](https://www.tandfonline.com/doi/full/10.1080/09670262.2011.565128#d1e565)



8 ASVs (all are phytoplankton) had negative occurrence with day-of-sampling and 10 (mostloy animals) had positive occurrence.
- Phytoplankton (Cryptophyceae, Rhodophyta, Phaeophyceae, Haptophyta) taxa decline across the sampling period-- might reflect a bloom decay or changing water masses.
- ASV_41_b, Nanomia bijuga (Siphonophore): a pelagic cnidarian. increased detection over time might reflect oceanic influx (mirroring phytoplankton pattern)

26 ASVs had positive occurrence with volume of water filtered, indicating that detection probability increases with graeter volume filtered




##### Check Model diagnostics
###### 1. Convergence diagnostics

```{r}
plot(sf.co1.out, param = "beta")  
plot(sf.co1.out, param = "alpha")
```

--> 

###### 2. Posterior predictive checks


```{r}
ppc.out1 <- ppcOcc(sf.co1.out, fit.stat = 'freeman-tukey', group = 1)
summary(ppc.out1)

ppc.out2 <- ppcOcc(sf.co1.out, fit.stat = 'freeman-tukey', group = 2)
summary(ppc.out2)

```



###### 3. WAIC for model comparison

```{r}
waicOcc(sf.co1.out)

```


1. Posterior Predictive Checks (Bayesian p-values)
- Community-level Bayesian p-values:
  - Group 1 (sites): 0.3845
  - Group 2 (replicates): 0.4296
- These Bayesian p-values are near the ideal midpoint (~0.5), suggesting a good model fit at the community level. Values between about 0.20.8 generally indicate a healthy balance between model fit and complexity. Thus, your community-level checks are quite good.
- Species-level Bayesian p-values:
  - Almost all species have Bayesian p-values comfortably in the acceptable range (mostly between 0.10.8). A few species have values closer to extremes (e.g., ASV_89 = 0.086), but these are relatively rare. This slight deviation is expected with a large number of species.
- The errors related to memory limits (vector memory limit of 96.0 Gb reached) are computational in nature and do not reflect a conceptual flaw in the model or analysis. They simply indicate you might need to break down certain diagnostics into smaller chunks if repeated.

2. WAIC Diagnostic
WAIC: 44447.282
Effective number of parameters (pD): 3849.936
- A large WAIC is normal given your large dataset and the complexity of a spatial multispecies occupancy model with 993 species. The high effective number of parameters (pD) reflects the complexity of the model and the many latent variables involved (e.g., species-level random effects, spatial factors).

3. MCMC Diagnostics (from your earlier output)
- R (Rhat): mostly close to 1.0 (1.05) across community-level parameters.
- Effective Sample Size (ESS): Most values are comfortably above ~500, with many in the thousands. This suggests good chain convergence and stable parameter estimates.
- The spatial covariance () parameter ESS > 10,000 is reassuring, indicating stable estimation of spatial structure.







#### Save environment
```{r}
# the data objects and model outputs are huge and slow up saving/loading. remove before running- will just have to rerun making the lists every time. models saved above as their own separate files

# rm(elas.exp.list,mifish.exp.list,co1.exp.list,model_out_elas,sf.elas.out,sf.mifish.out,sf.co1.out)

# save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_spOccupancyModels.RData")
```


```{r}
# load(file = "figures-expedition/exp_ecol_analysis_environment_upto_spOccupancyModels.RData")
```






# Species Richness- 
## Avg sample richness
```{r}
# species_counts_exp_mifish <- ps2024_exp_mifish_glommed_df %>%
#   group_by(Sample, Deployment, sites, replicates, site_altname, Volume_liter, lat, long, Bayside_f, Habitat, Site_type, Night__Day, Datecode, Month, Day, Avg_Depth_m, Avg_Temperature_C, Avg_Salinity_psu, Avg_Chlorophyll_ugL, Avg_Turbidity_NTU) %>%
#   summarise(UniqueSpeciesCount = n_distinct(Species), .groups = "drop")

species_counts_exp_mifish <- ps2024_exp_mifish_glommed_df %>%
  group_by(Sample, Habitat, Bayside_f) %>%
  summarise(UniqueSpeciesCount = n_distinct(Species), .groups = "drop")

species_counts_exp_elas <- ps2024_exp_elas_glommed_df %>%
  group_by(Sample, Habitat, Bayside_f) %>%
  summarise(UniqueSpeciesCount = n_distinct(Species), .groups = "drop")


species_counts_exp_mifish
species_counts_exp_elas

```

Calculate averages
```{r}
species_stats_by_habitat_mifish <- species_counts_exp_mifish %>%
  group_by(Habitat) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_habitat_mifish

species_stats_by_bayside_mifish <- species_counts_exp_mifish %>%
  group_by(Bayside_f) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_bayside_mifish


species_stats_by_habitat_elas <- species_counts_exp_elas %>%
  group_by(Habitat) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_habitat_elas

species_stats_by_bayside_elas <- species_counts_exp_elas %>%
  group_by(Bayside_f) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_bayside_elas
```
### Plot
```{r}
# Put line breaks in x-tick labels for plotting
species_stats_by_habitat_mifish$Habitat <- gsub(" ", "\n", species_stats_by_habitat_mifish$Habitat)
species_stats_by_habitat_elas$Habitat <- gsub(" ", "\n", species_stats_by_habitat_elas$Habitat)


#Plot
species_richness_by_habitat_mifish_plot <- 
ggplot(species_stats_by_habitat_mifish, aes(x = Habitat, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_habitat_mifish_plot

species_richness_by_bayside_mifish_plot <- 
ggplot(species_stats_by_bayside_mifish, aes(x = Bayside_f, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("cornflowerblue", "coral"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_bayside_mifish_plot

species_richness_by_habitat_elas_plot <- 
ggplot(species_stats_by_habitat_elas, aes(x = Habitat, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("azure2", "chartreuse4", "cadetblue1"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_habitat_elas_plot

species_richness_by_bayside_elas_plot <- 
ggplot(species_stats_by_bayside_elas, aes(x = Bayside_f, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("cornflowerblue", "coral"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_bayside_elas_plot


```

save
```{r}
ggsave(plot = species_richness_by_habitat_mifish_plot, filename = "figures-expedition/species_richness_by_habitat_mifish_plot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = species_richness_by_bayside_mifish_plot, filename = "figures-expedition/species_richness_by_bayside_mifish_plot.jpg", width = 2.5, height = 4, units = "in")

ggsave(plot = species_richness_by_habitat_elas_plot, filename = "figures-expedition/species_richness_by_habitat_elas_plot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = species_richness_by_bayside_elas_plot, filename = "figures-expedition/species_richness_by_bayside_elas_plot.jpg", width = 2.5, height = 4, units = "in")

```


## Overall richness by site type
```{r}
species_counts_bayside_mifish <- ps2024_exp_mifish_glommed_df %>%
  group_by(Bayside_f) %>%
  summarise(UniqueSpeciesCount = n_distinct(`Species.CommonName`), .groups = "drop")

species_counts_bayside_mifish

species_counts_habitat_mifish <- ps2024_exp_mifish_glommed_df %>%
  group_by(Habitat) %>%
  summarise(UniqueSpeciesCount = n_distinct(`Species.CommonName`), .groups = "drop")

species_counts_habitat_mifish

species_counts_bayside_elas <- ps2024_exp_elas_glommed_df %>%
  group_by(Bayside_f) %>%
  summarise(UniqueSpeciesCount = n_distinct(`Species.CommonName`), .groups = "drop")

species_counts_bayside_elas

species_counts_habitat_elas <- ps2024_exp_elas_glommed_df %>%
  group_by(Habitat) %>%
  summarise(UniqueSpeciesCount = n_distinct(`Species.CommonName`), .groups = "drop")

species_counts_habitat_elas

```

### Plot
```{r}
# Put line breaks in x-tick labels for plotting
species_counts_habitat_mifish$Habitat <- gsub(" ", "\n", species_counts_habitat_mifish$Habitat)
species_counts_habitat_elas$Habitat <- gsub(" ", "\n", species_counts_habitat_elas$Habitat)


#Plot
species_counts_by_habitat_mifish_plot <- 
ggplot(species_counts_habitat_mifish, aes(x = Habitat, y = UniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"), alpha = 0.8) +  
  labs(x = "", y = "Total Species Richness") +  
  theme_minimal() 
species_counts_by_habitat_mifish_plot

species_counts_by_bayside_mifish_plot <- 
ggplot(species_counts_bayside_mifish, aes(x = Bayside_f, y = UniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("cornflowerblue", "coral"), alpha = 0.8) +  
  labs(x = "", y = "Total Species Richness") +  
  theme_minimal() 
species_counts_by_bayside_mifish_plot

species_counts_by_habitat_elas_plot <- 
ggplot(species_counts_habitat_elas, aes(x = Habitat, y = UniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("azure2", "chartreuse4", "cadetblue1"), alpha = 0.8) +  
  labs(x = "", y = "Total Species Richness") +  
  theme_minimal() 
species_counts_by_habitat_elas_plot

species_counts_by_bayside_elas_plot <- 
ggplot(species_counts_bayside_elas, aes(x = Bayside_f, y = UniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("cornflowerblue", "coral"), alpha = 0.8) +  
  labs(x = "", y = "Total Species Richness") +  
  theme_minimal() 
species_counts_by_bayside_elas_plot


```

save
```{r}
ggsave(plot = species_counts_by_habitat_mifish_plot, filename = "figures-expedition/species_counts_by_habitat_mifish_plot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = species_counts_by_bayside_mifish_plot, filename = "figures-expedition/species_counts_by_bayside_mifish_plot.jpg", width = 2.5, height = 4, units = "in")

ggsave(plot = species_counts_by_habitat_elas_plot, filename = "figures-expedition/species_counts_by_habitat_elas_plot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = species_counts_by_bayside_elas_plot, filename = "figures-expedition/species_counts_by_bayside_elas_plot.jpg", width = 2.5, height = 4, units = "in")

```


# Multivariate Analysis

## PCA
PCA is essentially a type of PCoA using the Euclidean distance matrix as input. When combined with a log-ratio transformation of the count table, this is appropriate for compositional datasets. It is also recommended as a first step in exploratory analyses of sequencing datasets.

### CLR Transformation
First do a CLR, centered log ratio transformation of the absolute abundance data (after filtering), as suggested by [Gloor et al. 2017](https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2017.02224/full) for compositional data

# Estimate covariance matrix for CLR-transformed ASV table
`compositions::clr` expects samples in columns, species in rows.
Multivariate analyses cannot handle zeroes but clr should remove them... check for zeroes after transformation.
```{r}

# replace zeroes with very very small numbers
ps2024_exp_mifish_glommed_zerosubbed <- as.data.frame(otu_table(ps2024_exp_mifish_glommed))
ps2024_exp_mifish_glommed_zerosubbed[ps2024_exp_mifish_glommed_zerosubbed == 0] <- 1e-10 

ps2024_exp_elas_glommed_zerosubbed <- as.data.frame(otu_table(ps2024_exp_elas_glommed))
ps2024_exp_elas_glommed_zerosubbed[ps2024_exp_elas_glommed_zerosubbed == 0] <- 1e-10 

ps2024_exp_co1_glommed_zerosubbed <- as.data.frame(otu_table(ps2024_exp_co1_glommed))
ps2024_exp_co1_glommed_zerosubbed[ps2024_exp_co1_glommed_zerosubbed == 0] <- 1e-10 


exp_mifish_glommed_clr <- data.frame(compositions::clr(ps2024_exp_mifish_glommed_zerosubbed))
exp_elas_glommed_clr <- data.frame(compositions::clr(ps2024_exp_elas_glommed_zerosubbed))
exp_co1_glommed_clr <- data.frame(compositions::clr(ps2024_exp_co1_glommed_zerosubbed))

any(exp_mifish_glommed_clr==0)
any(exp_elas_glommed_clr==0)
any(exp_co1_glommed_clr==0)

```


Generate PCA. Leaving scaling off (default) because these are already scaled by CLR.
`prcomp` generally expects samples are rows and variables are columns.
```{r}
lograt_pca_mifish_exp <- prcomp(t(exp_mifish_glommed_clr), rank. = 10)
lograt_pca_elas_exp <- prcomp(t(exp_elas_glommed_clr), rank. = 10)
lograt_pca_co1_exp <- prcomp(t(exp_co1_glommed_clr), rank. = 10)

summary(lograt_pca_mifish_exp)
summary(lograt_pca_elas_exp)
summary(lograt_pca_co1_exp)
```

Screeplots
```{r}
lograt_variances_mifish_exp <- as.data.frame(lograt_pca_mifish_exp$sdev^2/sum(lograt_pca_mifish_exp$sdev^2)) %>% #Extract axes
  # Format to plot
  select(PercVar = 'lograt_pca_mifish_exp$sdev^2/sum(lograt_pca_mifish_exp$sdev^2)') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
head(lograt_variances_mifish_exp)

ggplot(lograt_variances_mifish_exp, aes(x = as.numeric(PCaxis), y = PercVar)) + 
  geom_bar(stat = "identity", fill = "grey", color = "black") +
  theme_minimal() +
  theme(axis.title = element_text(color = "black", face = "bold", size = 10),
        axis.text.y = element_text(color = "black", face = "bold"),
        axis.text.x = element_blank()) +
  labs(x = "PC axis", y = "% Variance", title = "Log-Ratio PCA Screeplot, CLR Tranformation- MiFish/ Expedition")


lograt_variances_elas_exp <- as.data.frame(lograt_pca_elas_exp$sdev^2/sum(lograt_pca_elas_exp$sdev^2)) %>% #Extract axes
  # Format to plot
  select(PercVar = 'lograt_pca_elas_exp$sdev^2/sum(lograt_pca_elas_exp$sdev^2)') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
head(lograt_variances_elas_exp)

ggplot(lograt_variances_elas_exp, aes(x = as.numeric(PCaxis), y = PercVar)) + 
  geom_bar(stat = "identity", fill = "grey", color = "black") +
  theme_minimal() +
  theme(axis.title = element_text(color = "black", face = "bold", size = 10),
        axis.text.y = element_text(color = "black", face = "bold"),
        axis.text.x = element_blank()) +
  labs(x = "PC axis", y = "% Variance", title = "Log-Ratio PCA Screeplot, CLR Tranformation- Elas02/ Expedition")

lograt_variances_co1_exp <- as.data.frame(lograt_pca_co1_exp$sdev^2/sum(lograt_pca_co1_exp$sdev^2)) %>% #Extract axes
  # Format to plot
  select(PercVar = 'lograt_pca_co1_exp$sdev^2/sum(lograt_pca_co1_exp$sdev^2)') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
head(lograt_variances_co1_exp)

ggplot(lograt_variances_co1_exp, aes(x = as.numeric(PCaxis), y = PercVar)) + 
  geom_bar(stat = "identity", fill = "grey", color = "black") +
  theme_minimal() +
  theme(axis.title = element_text(color = "black", face = "bold", size = 10),
        axis.text.y = element_text(color = "black", face = "bold"),
        axis.text.x = element_blank()) +
  labs(x = "PC axis", y = "% Variance", title = "Log-Ratio PCA Screeplot, CLR Tranformation- CO1/ Expedition")


```
First two axes are moderate for MiFish and CO1 and better for Elas


Extract variances from the clr pca
(Note [this issue](https://github.com/joey711/phyloseq/issues/1322) with converting rownames to column from a phyloseq object.... it was giving me a square 28x28 table rather than the full table. Needed to modify)
```{r}
pca_lograt_mifish_exp_df <- data.frame(lograt_pca_mifish_exp$x) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata into the pca data table
pca_lograt_mifish_exp_df <- sample_data(ps2024_exp_mifish_glommed) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
right_join(pca_lograt_mifish_exp_df, by = "SampleID")

pca_lograt_mifish_exp_df


pca_lograt_elas_exp_df <- data.frame(lograt_pca_elas_exp$x) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata into the pca data table
pca_lograt_elas_exp_df <- sample_data(ps2024_exp_elas_glommed) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
right_join(pca_lograt_elas_exp_df, by = "SampleID")

pca_lograt_elas_exp_df

pca_lograt_co1_exp_df <- data.frame(lograt_pca_co1_exp$x) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata into the pca data table
pca_lograt_co1_exp_df <- sample_data(ps2024_exp_co1_glommed) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
right_join(pca_lograt_co1_exp_df, by = "SampleID")

pca_lograt_co1_exp_df

```

Select eigenvalues from dataframe, round to 4 places and multiply by 100 for plotting. These will be the axes for the 3-D plot
```{r}
eigenvalues_pca_mifish_exp <-round(lograt_variances_mifish_exp[,2], digits = 4)*100
eigenvalues_pca_elas_exp <-round(lograt_variances_elas_exp[,2], digits = 4)*100
eigenvalues_pca_co1_exp <-round(lograt_variances_co1_exp[,2], digits = 4)*100


```

### Plot PCA
Use 1st two axes 
```{r}
pca_mifish_exp_bayside_plot <- ggplot(pca_lograt_mifish_exp_df, aes(x = PC1, y = PC2, color = Bayside)) +
  geom_point(size = 3, alpha = 0.8) + 
  labs(x = paste0('PC1 ',eigenvalues_pca_mifish_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_mifish_exp[2],'%'), color = "Bayside") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
  labs(title="PCA: MiFish") +
  scale_color_manual(values = c("coral","cornflowerblue"))
pca_mifish_exp_bayside_plot

pca_elas_exp_bayside_plot <- ggplot(pca_lograt_elas_exp_df, aes(x = PC1, y = PC2, color = Bayside)) +
  geom_point(size = 3, alpha = 0.8) + 
  labs(x = paste0('PC1 ',eigenvalues_pca_elas_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_elas_exp[2],'%'), color = "Bayside") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
    labs(title="PCA: Elas02") +
  scale_color_manual(values = c("coral","cornflowerblue"))
pca_elas_exp_bayside_plot

pca_co1_exp_bayside_plot <- ggplot(pca_lograt_co1_exp_df, aes(x = PC1, y = PC2, color = Bayside)) +
  geom_point(size = 3, alpha = 0.8) + 
  labs(x = paste0('PC1 ',eigenvalues_pca_co1_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_co1_exp[2],'%'), color = "Bayside") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
  labs(title="PCA: CO1") +
  scale_color_manual(values = c("coral","cornflowerblue"))
pca_co1_exp_bayside_plot

#

pca_mifish_exp_habitat_plot <- ggplot(pca_lograt_mifish_exp_df, aes(x = PC1, y = PC2, fill = Habitat)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(x = paste0('PC1 ',eigenvalues_pca_mifish_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_mifish_exp[2],'%'), color = "Habitat") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
    labs(title="PCA: MiFish") +
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
pca_mifish_exp_habitat_plot


pca_elas_exp_habitat_plot <- ggplot(pca_lograt_elas_exp_df, aes(x = PC1, y = PC2, fill = Habitat)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(x = paste0('PC1 ',eigenvalues_pca_elas_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_elas_exp[2],'%'), color = "Habitat") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
    labs(title="PCA: Elas02") +
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1"))
pca_elas_exp_habitat_plot


pca_co1_exp_habitat_plot <- ggplot(pca_lograt_co1_exp_df, aes(x = PC1, y = PC2, fill = Habitat)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(x = paste0('PC1 ',eigenvalues_pca_co1_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_co1_exp[2],'%'), color = "Habitat") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
    labs(title="PCA: CO1") +
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
pca_co1_exp_habitat_plot


#

pca_mifish_exp_nightday_plot <- ggplot(pca_lograt_mifish_exp_df, aes(x = PC1, y = PC2, fill = Night__Day)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(
    x = paste0('PC1 ', eigenvalues_pca_mifish_exp[1], '%'), 
    y = paste0('PC2 ', eigenvalues_pca_mifish_exp[2], '%'), 
    fill = ""
  ) +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
    labs(title="PCA: MiFish") +
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
pca_mifish_exp_nightday_plot

pca_elas_exp_nightday_plot <- ggplot(pca_lograt_elas_exp_df, aes(x = PC1, y = PC2, fill = Night__Day)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(
    x = paste0('PC1 ', eigenvalues_pca_elas_exp[1], '%'), 
    y = paste0('PC2 ', eigenvalues_pca_elas_exp[2], '%'), 
    fill = ""
  ) +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
    labs(title="PCA: Elas02") +
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
pca_elas_exp_nightday_plot


pca_co1_exp_nightday_plot <- ggplot(pca_lograt_co1_exp_df, aes(x = PC1, y = PC2, fill = Night__Day)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(
    x = paste0('PC1 ', eigenvalues_pca_co1_exp[1], '%'), 
    y = paste0('PC2 ', eigenvalues_pca_co1_exp[2], '%'), 
    fill = ""
  ) +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
    labs(title="PCA: CO1") +
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
pca_co1_exp_nightday_plot

```





### Env Fit
#### MiFish
```{r}
# sort metadata in same order as the pca matrix
metadata_mifish_exp <- sample_data(ps2024_exp_mifish) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% arrange(factor(SampleID, levels = rownames(lograt_pca_mifish_exp$x)))

# Fix time and date so it can be used as input

metadata_mifish_exp$DateTime <- as.POSIXct(force_tz((mdy_hms(paste(metadata_mifish_exp$Date, metadata_mifish_exp$Start_Time__ET_))), tzone = "America/New_York"))

# Convert Start Time (from RocSI filtration time) to continuous variable rather than factor. This will be a "time-of-day" variable
metadata_mifish_exp$Time_numeric <- as.numeric(as_hms(as.character(metadata_mifish_exp$Start_Time__ET_)))

# Convert Datecode to numeric
metadata_mifish_exp$Datecode <- as.numeric(metadata_mifish_exp$Datecode)

# Remove some columns of metadate- only retain the env data that I am testing. Leave out the repetitive ID columns, etc.
metadata_mifish_exp <- select(metadata_mifish_exp, -"Name.Deploy_Cartr_Library", -Deployment, -Cartridge_ID, -sites, -replicates, -site_altname, -controls, -Date, -Year, -Month, -Day, -Name.Deploy_Cartr, -Start_Time__ET_, -End_Time__ET_)

# fit environmental factors and save stats output
pca_envfit_mifish_exp <- envfit(lograt_pca_mifish_exp, metadata_mifish_exp, permutations = 10000, na.rm = TRUE)

pca_envfit_mifish_exp
capture.output(pca_envfit_mifish_exp, file = "figures-expedition/pca_envfit_mifish_exp.txt")

```
*SUMMARY- MiFish EnvFir Results that were sig*
Salinity, Time, Site_type



#### Elas02
```{r}
# sort metadata in same order as the pca matrix
metadata_elas_exp <- sample_data(ps2024_exp_elas) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% arrange(factor(SampleID, levels = rownames(lograt_pca_elas_exp$x)))

# Fix time and date so it can be used as input
metadata_elas_exp$DateTime <- as.POSIXct(force_tz((mdy_hms(paste(metadata_elas_exp$Date, metadata_elas_exp$Start_Time__ET_))), tzone = "America/New_York"))

# Convert Start Time (from RocSI filtration time) to continuous variable rather than factor. This will be a "time-of-day" variable
metadata_elas_exp$Time_numeric <- as.numeric(as_hms(as.character(metadata_elas_exp$Start_Time__ET_)))

# Convert Datecode to numeric
metadata_elas_exp$Datecode <- as.numeric(metadata_elas_exp$Datecode)

# Remove some columns of metadata- only retain the env data that I am testing. Leave out the repetitive ID columns, etc.
metadata_elas_exp <- select(metadata_elas_exp, -Deployment, -Cartridge_ID, -sites, -replicates, -controls, -Date, -Year, -Month, -Day, -Name_Deploy_Cartr_Library, -Start_Time__ET_, -End_Time__ET_)

# fit environmental factors and save stats output
pca_envfit_elas_exp <- envfit(lograt_pca_elas_exp, metadata_elas_exp, permutations = 10000, na.rm = TRUE)

pca_envfit_elas_exp
capture.output(pca_envfit_elas_exp, file = "figures-expedition/pca_envfit_elas_exp.txt")

```

*Summary of Envfit to Elas02 library*
Date, salinity, site_type, night-day



#### CO1
the `sample_data(ps2024_exp_co1)` does not have the calculated avg temp, etc. Grab it from the metadata file 
```{r}
# Import complete table with temp, etc
metadata_co1_exp_complete <- read.csv(file = "figures-expedition/CTD_avg_eDNA_sample_intervals.csv")


# sort metadata in same order as the pca matrix, remove empty columns
metadata_co1_exp <- sample_data(ps2024_exp_co1) %>% 
  data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
  arrange(factor(SampleID, levels = rownames(lograt_pca_co1_exp$x))) %>%
  select(-DO, -Temperature, -Salinity, -Time, -Weather) %>%
  mutate(Name.Deploy_Cartr = gsub("\\.", "_", gsub("_$", "", as.character(Name_Deploy_Cartr))))


# join
metadata_co1_exp <- full_join(metadata_co1_exp, metadata_co1_exp_complete, by = join_by(Name.Deploy_Cartr))


# Fix time and date so it can be used as input
metadata_co1_exp$DateTime <- as.POSIXct(force_tz((mdy_hms(paste(metadata_co1_exp$Date, metadata_co1_exp$Start_Time__ET_))), tzone = "America/New_York"))

# Convert Start Time (from RocSI filtration time) to continuous variable rather than factor. This will be a "time-of-day" variable
metadata_co1_exp$Time_numeric <- as.numeric(as_hms(as.character(metadata_co1_exp$Start_Time__ET_)))

# Convert Datecode to numeric
metadata_co1_exp$Datecode <- as.numeric(metadata_co1_exp$Datecode)

# Remove some columns of metadata- only retain the env data that I am testing. Leave out the repetitive ID columns, etc.
metadata_co1_exp <- select(metadata_co1_exp, -"Name_Deploy_Cartr", -Station, -Deployment, -Cartridge_ID, -sites, -replicates, -site_altname, -controls, -Date, -Year, -Month, -Day, -Start_Time__ET_, -End_Time__ET_, -"Name.Deploy_Cartr", -"Deploy_Cartr_Library")

# fit environmental factors and save stats output
pca_envfit_co1_exp <- envfit(lograt_pca_co1_exp, metadata_co1_exp, permutations = 10000, na.rm = TRUE)

pca_envfit_co1_exp
capture.output(pca_envfit_co1_exp, file = "figures-expedition/pca_envfit_co1_exp.txt")

```

Volume, lat, long, date, temp, salinity, bayside, habitat, night/day are all significant





##### Plot EnvFit, Categorical 
```{r}
pca_mifish_exp_bayside_plot <- pca_mifish_exp_bayside_plot +
  stat_ellipse(aes(color = factor(Bayside)), level = 0.99)

pca_elas_exp_bayside_plot <- pca_elas_exp_bayside_plot +
  stat_ellipse(aes(color = factor(Bayside)), level = 0.99)

pca_co1_exp_bayside_plot <- pca_co1_exp_bayside_plot +
  stat_ellipse(aes(color = factor(Bayside)), level = 0.99)


pca_mifish_exp_habitat_plot <- pca_mifish_exp_habitat_plot +
  stat_ellipse(aes(color = factor(Habitat)), level = 0.99) +
  scale_color_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey")) 


pca_elas_exp_habitat_plot <- pca_elas_exp_habitat_plot +
  stat_ellipse(aes(color = factor(Habitat)), level = 0.99) +
    scale_color_manual(values = c("azure2", "chartreuse4", "cadetblue1")) 

pca_co1_exp_habitat_plot <- pca_co1_exp_habitat_plot +
  stat_ellipse(aes(color = factor(Habitat)), level = 0.99) +
    scale_color_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey")) 


pca_mifish_exp_nightday_plot <- pca_mifish_exp_nightday_plot +
  stat_ellipse(aes(color = factor(Night__Day)), level = 0.99) +
  scale_color_manual(values = c("cornsilk", "antiquewhite4")) +
  guides(color = "none") 


pca_elas_exp_nightday_plot <- pca_elas_exp_nightday_plot +
  stat_ellipse(aes(color = factor(Night__Day)), level = 0.99) +
  scale_color_manual(values = c("cornsilk", "antiquewhite4")) +
  guides(color = "none") 

pca_co1_exp_nightday_plot <- pca_co1_exp_nightday_plot +
  stat_ellipse(aes(color = factor(Night__Day)), level = 0.99) +
  scale_color_manual(values = c("cornsilk", "antiquewhite4")) +
  guides(color = "none") 



pca_mifish_exp_bayside_plot
pca_elas_exp_bayside_plot
pca_co1_exp_bayside_plot

pca_mifish_exp_habitat_plot
pca_elas_exp_habitat_plot
pca_co1_exp_habitat_plot

pca_mifish_exp_nightday_plot
pca_elas_exp_nightday_plot
pca_co1_exp_nightday_plot

```

###### Plot Envfit + Numerical: Mifish
Add sig vectors from envfit ontop of categ vars

```{r}
pca_mifish_sig_vectors <- as.data.frame(pca_envfit_mifish_exp$vectors$arrows) %>%
  rownames_to_column(var = "EnvVar") %>%
  filter(EnvVar %in% c("Avg_Salinity_psu", "Time_numeric"))  %>%  
  mutate(EnvVar = ifelse(EnvVar == "Avg_Salinity_psu", "Salinity", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Time_numeric", "Time of Day", EnvVar))

# Plot vectors ontop of ellipse plot of bayside
pca_mifish_exp_bayside_date_salinity_plot <- pca_mifish_exp_bayside_plot +
  geom_segment(data = pca_mifish_sig_vectors,
               aes(x = 0, y = 0, xend = PC1, yend = PC2),
               arrow = arrow(type = "closed", length = unit(0.2, "inches")),
               color = "black") +  
  geom_text(data = pca_mifish_sig_vectors,
            aes(x = PC1, y = PC2, label = EnvVar),
            size = 4, color = "black", vjust = -0.5)

pca_mifish_exp_bayside_date_salinity_plot


```
save
```{r}
ggsave(plot = pca_mifish_exp_bayside_date_salinity_plot, filename = "figures-expedition/pca_mifish_exp_bayside_latlong_date_salinity_turbid_plot.jpg", width = 7, height = 5, units = "in")

ggsave(plot = pca_mifish_exp_nightday_plot, filename = "figures-expedition/pca_mifish_exp_nightday_plot.jpg", width = 7, height = 5, units = "in")

```

###### Plot Envfit + Numerical: Elas02
Add sig vectors from envfit ontop of categorical vars 
Date, salinity, night-day

```{r}
pca_elas_sig_vectors <- as.data.frame(pca_envfit_elas_exp$vectors$arrows) %>%
  rownames_to_column(var = "EnvVar") %>%
  filter(EnvVar %in% c("Avg_Salinity_psu","Datecode")) %>%
  mutate(EnvVar = ifelse(EnvVar == "Avg_Salinity_psu", "Salinity", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Datecode", "Date", EnvVar))

    
# Plot vectors ontop of ellipse plot of Bayside
pca_elas_exp_nightday_salinity <- pca_elas_exp_nightday_plot +
  geom_segment(data = pca_elas_sig_vectors,
               aes(x = 0, y = 0, xend = PC1, yend = PC2),
               arrow = arrow(type = "closed", length = unit(0.2, "inches")),
               color = "black",
               inherit.aes = FALSE) +  
  geom_text(data = pca_elas_sig_vectors,
            aes(x = PC1, y = PC2, label = EnvVar),
            size = 4, color = "black", vjust = -0.5, inherit.aes = FALSE)
pca_elas_exp_nightday_salinity


```

save
```{r}
ggsave(plot = pca_elas_exp_nightday_salinity, filename = "figures-expedition/pca_elas_exp_nightday_salinity.jpg", width = 7, height = 5, units = "in")

ggsave(plot = pca_elas_exp_habitat_plot, filename = "figures-expedition/pca_elas_exp_habitat_plot.jpg", width = 7, height = 5, units = "in")

```




###### Plot Envfit + Numerical: CO1
Add sig vectors from envfit ontop of categ vars
Volume, lat, long, date, temp, salinity, bayside, habitat, night/day are all significant
```{r}
pca_co1_sig_vectors <- as.data.frame(pca_envfit_co1_exp$vectors$arrows) %>%
  rownames_to_column(var = "EnvVar") %>%
  filter(EnvVar %in% c("Volume_liter","lat","long","Datecode", "Avg_Temperature_C","Avg_Salinity_psu"))  %>%  
  mutate(EnvVar = ifelse(EnvVar == "Volume_liter", "Volume", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Avg_Salinity_psu", "Salinity", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Datecode", "Date", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Avg_Temperature_C", "Temp", EnvVar)) 


# Plot vectors ontop of ellipse plot of bayside
pca_co1_exp_bayside_date_salinity_plot <- pca_co1_exp_bayside_plot +
  geom_segment(data = pca_co1_sig_vectors,
               aes(x = 0, y = 0, xend = PC1, yend = PC2),
               arrow = arrow(type = "closed", length = unit(0.2, "inches")),
               color = "black") +  
  geom_text(data = pca_co1_sig_vectors,
            aes(x = PC1, y = PC2, label = EnvVar),
            size = 4, color = "black", vjust = -0.5)

pca_co1_exp_bayside_date_salinity_plot


```
save
```{r}
ggsave(plot = pca_co1_exp_bayside_date_salinity_plot, filename = "figures-expedition/pca_co1_exp_bayside_latlong_date_salinity_turbid_plot.jpg", width = 7, height = 5, units = "in")

ggsave(plot = pca_co1_exp_nightday_plot, filename = "figures-expedition/pca_co1_exp_nightday_plot.jpg", width = 7, height = 5, units = "in")

```


#### Save

```{r}
save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_pca.RData")
```


```{r}
load(file = "figures-expedition/exp_ecol_analysis_environment_upto_pca.RData")
```


## Dissimilarity across geographic distance usiung and Mantel tests
Jaccard similarity (or its complement, Jaccard distance) can be used to examine spatial autocorrelation, especially when paired with [Mantel tests](https://uw.pressbooks.pub/appliedmultivariatestatistics/chapter/mantel-test/)

Bray Curtis can similarly be compared to geo distance and might be more appropriate for metabarcoding data. 

The clr/ Euclidean distance matrix is more apprpriate for compositional data

Try all...



[Example 1](https://fireecology.springeropen.com/articles/10.4996/fireecology.0502116): Mantel test compares two distance matrices (one based on the variable of interest and the other based on the geo distance between sampling units). From the paper:

- The normalized statistic behaves similar to the Pearson correlation coefficient, varying between 1 and +1, so that coefficients can be compared to other variables at the same site or to similar variables at other sites. We determined the overall significance of spatial relationships by permutation testing
- We subjected each of these measures within each experimental unit to a Mantel test using Euclidean distances, with 10 000 permutations used to establish significance ( = 0.05). Although the large number of tests we performed would argue for an adjustment of the critical value, we wanted these tests to be as liberal as possible to search for evidence of significant spatial autocorrelation (e.g., if we used a Bonferroni correction for our 144 tests, we would have a critical value of  = 0.05  144 = 0.00035, (a value not surpassed by any of our tests).

[Example 2](https://sciendo.com/article/10.2478/sg-2018-0013)

[Example 3, Wen et al.](https://onlinelibrary.wiley.com/doi/full/10.1002/ece3.1962) - further evidence (or non evidence) of spatial autocorrelation

[Example 4](https://www.jstor.org/stable/pdf/48703667.pdf)


### Elas02
#### Mantel tests
```{r}
# Jaccard uses  presence/absence
# use already-made presence/absence tables from SpOccupancy analysis
elas_exp_df_countmatrix

elas_jaccard_dist <- vegdist(t(elas_exp_df_countmatrix), method = "jaccard")


# for Bray, use relative abundance calculated from read abundance table
elas_exp_df_counttable
elas_exp_df_counttable_relabun <- sweep(elas_exp_df_counttable, 2, colSums(elas_exp_df_counttable), FUN = "/")

elas_bray_dist <- vegdist(t(elas_exp_df_counttable_relabun), method = "bray")


# for compositional approach, use clr transformed data and euclidean distance (aka Aitchison distance)
elas_exp_df_counttable_relabun_clr <- elas_exp_df_counttable
elas_exp_df_counttable_relabun_clr[elas_exp_df_counttable_relabun_clr == 0] <- 1e-10 # replace zeros using a small pseudocount
elas_exp_df_counttable_relabun_clr <- data.frame(compositions::clr(elas_exp_df_counttable_relabun_clr))

elas_aitchison_dist <- vegdist(t(elas_exp_df_counttable_relabun), method = "euclidean")

# extract and order coordinate rows to match matrix sample order
sample_ids <- colnames(elas_exp_df_countmatrix)

# Use coordinates form sampledata df. Note caveat that, bc samples were collected multiple times at same location on different days, there will be places with "zero" geographic distance that might be very different bc of time. Address that later
geo_dist_matrix <- elas_exp_df_sampledata[sample_ids, c("long", "lat")]


# calculate pairwise geographic distances
geo_dist_matrix <- distm(geo_dist_matrix, fun = distHaversine) # computes distances between points in meters using "Haversine" method, which assumes a spherical earth
geo_dist <- as.dist(geo_dist_matrix)

# compare Jaccard vs geographic distance
mantel(elas_jaccard_dist, as.dist(geo_dist), permutations = 10000)
mantel(elas_bray_dist, as.dist(geo_dist), permutations = 10000)
mantel(elas_aitchison_dist, as.dist(geo_dist), permutations = 10000)



```


#### Plot
```{r}
# Convert to vectors that should match in length
jaccard_vec <- as.vector(elas_jaccard_dist)
bray_vec <- as.vector(elas_bray_dist)
aitchison_vec <- as.vector(elas_aitchison_dist)
geo_vec <- as.vector(geo_dist)


# Combine into data frame. Note I converted to SIMILARITES so the plots are easier to understand
# Jaccard and Bray are bounded to 1
# Aitshison is unbounded, so use different calc
elas_similarities <- data.frame(Geo_Distance_km = geo_vec / 1000,
  Jaccard_Similarity = 1-jaccard_vec,
  BrayCurtis_Similarity = 1-bray_vec,
  Aitchison_Similarity = 1/(1+aitchison_vec))

elas_distvsjaccard <- ggplot(elas_similarities, aes(x = Geo_Distance_km, y = Jaccard_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Jaccard, Elas02",
    x = "Geographic distance (km)",
    y = "1 - Jaccard dissimilarity") +
  theme_minimal()


elas_distvsbray <- ggplot(elas_similarities, aes(x = Geo_Distance_km, y = BrayCurtis_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Bray-Curtis, Elas02",
    x = "Geographic distance (km)",
    y = "1 - Bray Curtis dissimilarity") +
  theme_minimal()


elas_distvsaitchison <- ggplot(elas_similarities, aes(x = Geo_Distance_km, y = Aitchison_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Aitchison, Elas02",
    x = "Geographic distance (km)",
    y = "1/(1+Aitchison distance)") +
  theme_minimal()


elas_distvsjaccard
elas_distvsbray
elas_distvsaitchison
```
No relationship between Jaccard distance and geographic distance- suggests Elas02 presence/absence is spatially correlated 
HOWEVER both Bray Curtis and Aitchison suggests there is a relationship between dissimilarity and geo distance in which dissimilarity increases with geo distance (i.e. samples closer together are less dissimilar/ more similar)

CAVEAT- cannot include samples with all zeroes for Jaccard. Therefore this analysis only compares those stations at which at least one elasmobranch was detected. There were 33 stations (of 94) which had zero elasmobranchs. Also there is a very small number of ASVs (only 6) in this dataset compated to the others.

save
```{r}
ggsave(plot = elas_distvsjaccard, filename = "figures-expedition/elas_distvsjaccard.jpg", width = 7, height = 5, units = "in")

ggsave(plot = elas_distvsbray, filename = "figures-expedition/elas_distvsbray.jpg", width = 7, height = 5, units = "in")

ggsave(plot = elas_distvsaitchison, filename = "figures-expedition/elas_distvsaitchison.jpg", width = 7, height = 5, units = "in")

```


### MiFish
#### Mantel tests
```{r}
# Jaccard uses  presence/absence
# use already-made presence/absence tables from SpOccupancy analysis
mifish_exp_df_countmatrix

mifish_jaccard_dist <- vegdist(t(mifish_exp_df_countmatrix), method = "jaccard")


# for Bray, use relative abundance calculated from read abundance table
mifish_exp_df_counttable
mifish_exp_df_counttable_relabun <- sweep(mifish_exp_df_counttable, 2, colSums(mifish_exp_df_counttable), FUN = "/")

mifish_bray_dist <- vegdist(t(mifish_exp_df_counttable_relabun), method = "bray")


# for compositional approach, use clr transformed data and euclidean distance (aka Aitchison distance)
mifish_exp_df_counttable_relabun_clr <- mifish_exp_df_counttable
mifish_exp_df_counttable_relabun_clr[mifish_exp_df_counttable_relabun_clr == 0] <- 1e-10 # replace zeros using a small pseudocount
mifish_exp_df_counttable_relabun_clr <- data.frame(compositions::clr(mifish_exp_df_counttable_relabun_clr))

mifish_aitchison_dist <- vegdist(t(mifish_exp_df_counttable_relabun), method = "euclidean")

# extract and order coordinate rows to match matrix sample order
sample_ids <- colnames(mifish_exp_df_countmatrix)

# Use coordinates form sampledata df. Note caveat that, bc samples were collected multiple times at same location on different days, there will be places with "zero" geographic distance that might be very different bc of time. Address that later
geo_dist_matrix <- mifish_exp_df_sampledata[sample_ids, c("long", "lat")]


# calculate pairwise geographic distances
geo_dist_matrix <- distm(geo_dist_matrix, fun = distHaversine) # computes distances between points in meters using "Haversine" method, which assumes a spherical earth
geo_dist <- as.dist(geo_dist_matrix)

# compare Jaccard vs geographic distance
mantel(mifish_jaccard_dist, as.dist(geo_dist), permutations = 10000)
mantel(mifish_bray_dist, as.dist(geo_dist), permutations = 10000)
mantel(mifish_aitchison_dist, as.dist(geo_dist), permutations = 10000)



```

#### Plot
```{r}
# Convert to vectors that should match in length
jaccard_vec <- as.vector(mifish_jaccard_dist)
bray_vec <- as.vector(mifish_bray_dist)
aitchison_vec <- as.vector(mifish_aitchison_dist)
geo_vec <- as.vector(geo_dist)


# Combine into data frame. Note I converted to SIMILARITES so the plots are easier to understand
# Jaccard and Bray are bounded to 1
# Aitshison is unbounded, so use different calc
mifish_similarities <- data.frame(Geo_Distance_km = geo_vec / 1000,
  Jaccard_Similarity = 1-jaccard_vec,
  BrayCurtis_Similarity = 1-bray_vec,
  Aitchison_Similarity = 1/(1+aitchison_vec))

mifish_distvsjaccard <- ggplot(mifish_similarities, aes(x = Geo_Distance_km, y = Jaccard_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Jaccard, MiFish",
    x = "Geographic distance (km)",
    y = "1 - Jaccard dissimilarity") +
  theme_minimal()


mifish_distvsbray <- ggplot(mifish_similarities, aes(x = Geo_Distance_km, y = BrayCurtis_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Bray-Curtis, MiFish",
    x = "Geographic distance (km)",
    y = "1 - Bray Curtis dissimilarity") +
  theme_minimal()


mifish_distvsaitchison <- ggplot(mifish_similarities, aes(x = Geo_Distance_km, y = Aitchison_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Aitchison, MiFish",
    x = "Geographic distance (km)",
    y = "1/(1+Aitchison distance)") +
  theme_minimal()


mifish_distvsjaccard
mifish_distvsbray
mifish_distvsaitchison
```

None are significant, i.e. there is no relationship between geo distance and compositional similarity for MiFish


save
```{r}
ggsave(plot = mifish_distvsjaccard, filename = "figures-expedition/mifish_distvsjaccard.jpg", width = 7, height = 5, units = "in")

ggsave(plot = mifish_distvsbray, filename = "figures-expedition/mifish_distvsbray.jpg", width = 7, height = 5, units = "in")

ggsave(plot = mifish_distvsaitchison, filename = "figures-expedition/mifish_distvsaitchison.jpg", width = 7, height = 5, units = "in")

```


### CO1
#### Mantel tests
```{r}
# Jaccard uses  presence/absence
# use already-made presence/absence tables from SpOccupancy analysis
co1_exp_df_countmatrix

co1_jaccard_dist <- vegdist(t(co1_exp_df_countmatrix), method = "jaccard")


# for Bray, use relative abundance calculated from read abundance table
co1_exp_df_counttable
co1_exp_df_counttable_relabun <- sweep(co1_exp_df_counttable, 2, colSums(co1_exp_df_counttable), FUN = "/")

co1_bray_dist <- vegdist(t(co1_exp_df_counttable_relabun), method = "bray")


# for compositional approach, use clr transformed data and euclidean distance (aka Aitchison distance)
co1_exp_df_counttable_relabun_clr <- co1_exp_df_counttable
co1_exp_df_counttable_relabun_clr[co1_exp_df_counttable_relabun_clr == 0] <- 1e-10 # replace zeros using a small pseudocount
co1_exp_df_counttable_relabun_clr <- data.frame(compositions::clr(co1_exp_df_counttable_relabun_clr))

co1_aitchison_dist <- vegdist(t(co1_exp_df_counttable_relabun), method = "euclidean")

# extract and order coordinate rows to match matrix sample order
sample_ids <- colnames(co1_exp_df_countmatrix)

# Use coordinates form sampledata df. Note caveat that, bc samples were collected multiple times at same location on different days, there will be places with "zero" geographic distance that might be very different bc of time. Address that later
geo_dist_matrix <- co1_exp_df_sampledata[sample_ids, c("long", "lat")]


# calculate pairwise geographic distances
geo_dist_matrix <- distm(geo_dist_matrix, fun = distHaversine) # computes distances between points in meters using "Haversine" method, which assumes a spherical earth
geo_dist <- as.dist(geo_dist_matrix)

# compare Jaccard vs geographic distance
mantel(co1_jaccard_dist, as.dist(geo_dist), permutations = 10000)
mantel(co1_bray_dist, as.dist(geo_dist), permutations = 10000)
mantel(co1_aitchison_dist, as.dist(geo_dist), permutations = 10000)



```


#### Plot
```{r}
# Convert to vectors that should match in length
jaccard_vec <- as.vector(co1_jaccard_dist)
bray_vec <- as.vector(co1_bray_dist)
aitchison_vec <- as.vector(co1_aitchison_dist)
geo_vec <- as.vector(geo_dist)


# Combine into data frame. Note I converted to SIMILARITES so the plots are easier to understand
# Jaccard and Bray are bounded to 1
# Aitshison is unbounded, so use different calc
co1_similarities <- data.frame(Geo_Distance_km = geo_vec / 1000,
  Jaccard_Similarity = 1-jaccard_vec,
  BrayCurtis_Similarity = 1-bray_vec,
  Aitchison_Similarity = 1/(1+aitchison_vec))

co1_distvsjaccard <- ggplot(co1_similarities, aes(x = Geo_Distance_km, y = Jaccard_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Jaccard, CO1",
    x = "Geographic distance (km)",
    y = "1 - Jaccard dissimilarity") +
  theme_minimal()


co1_distvsbray <- ggplot(co1_similarities, aes(x = Geo_Distance_km, y = BrayCurtis_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Bray-Curtis, CO1",
    x = "Geographic distance (km)",
    y = "1 - Bray Curtis dissimilarity") +
  theme_minimal()


co1_distvsaitchison <- ggplot(co1_similarities, aes(x = Geo_Distance_km, y = Aitchison_Similarity)) +
  geom_point(alpha = 0.4) +
  geom_smooth(method = "loess", se = TRUE, color = "blue") +
  labs(title = "Distance-decay of community similarity, Aitchison, CO1",
    x = "Geographic distance (km)",
    y = "1/(1+Aitchison distance)") +
  theme_minimal()


co1_distvsjaccard
co1_distvsbray
co1_distvsaitchison
```

By all 3 distance metrics, there is a significant relationship between geo distance and compositional similarity. This makes sense bc this dataset is highly composed of single-celled phytoplankton and other planktonic things (larvae, zooplankton, protists, etc) which are dispersal limited and not mobile. Their position is thus controlled by mixing of currents, etc



save
```{r}
ggsave(plot = co1_distvsjaccard, filename = "figures-expedition/co1_distvsjaccard.jpg", width = 7, height = 5, units = "in")

ggsave(plot = co1_distvsbray, filename = "figures-expedition/co1_distvsbray.jpg", width = 7, height = 5, units = "in")

ggsave(plot = co1_distvsaitchison, filename = "figures-expedition/co1_distvsaitchison.jpg", width = 7, height = 5, units = "in")

```


### Save

```{r}
save.image(file = "figures-expedition/exp_ecol_analysis_environment_upto_dissimilarity.RData")
```


```{r}
load(file = "figures-expedition/exp_ecol_analysis_environment_upto_dissimilarity.RData")
```


## NMDS
Use compositional approach, an NMDS on Aitchison distance matrix (aka Euclidean distances on clr-trasnformed abundance. Already computed above)
To compare to PCA

### Run NMDS
```{r}
elas_aitchison_nmds <- metaMDS(elas_aitchison_dist, k=2, autotransform=FALSE) 

mifish_aitchison_nmds <- metaMDS(mifish_aitchison_dist, k=2, autotransform=FALSE) 

co1_aitchison_nmds <- metaMDS(co1_aitchison_dist, k=2, autotransform=FALSE) 

```


### Stress
```{r}
elas_aitchison_nmds$stress

mifish_aitchison_nmds$stress

co1_aitchison_nmds$stress
```

Stress less than 0.1 is good. Less than 0.05 is better. Elas02 is OK but the other two are a bit high. 

Can also try NMDS up to 3 dimensions. See if that's a better fit...

### 3D NMDS
```{r}
elas_aitchison_nmds_3d <- metaMDS(elas_aitchison_dist, k=3, autotransform=FALSE) 

mifish_aitchison_nmds_3d <- metaMDS(mifish_aitchison_dist, k=3, autotransform=FALSE) 
co1_aitchison_nmds_3d <- metaMDS(co1_aitchison_dist, k=3, autotransform=FALSE) 

```


### Stress
```{r}
elas_aitchison_nmds_3d$stress

mifish_aitchison_nmds_3d$stress

co1_aitchison_nmds_3d$stress
```

This improved, proceed with 3 dimensions


### Prepare data frames
Extract points and merge to metadata
Elas02
```{r}
# Extract points from nmds and merge into data frame with metadata 
elas_aitchison_nmds_3d_df <- data.frame(elas_aitchison_nmds_3d$points) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata 
elas_sample_data <- sample_data(ps2024_exp_elas_glommed) %>%
  data.frame() %>%
  rename("SampleID" = "Name.Deploy_Cartr")

elas_aitchison_nmds_3d_df <- full_join(elas_aitchison_nmds_3d_df, elas_sample_data)

elas_aitchison_nmds_3d_df
```


MiFish
```{r}
# Extract points from nmds and merge into data frame with metadata 
mifish_aitchison_nmds_3d_df <- data.frame(mifish_aitchison_nmds_3d$points) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata 
mifish_sample_data <- sample_data(ps2024_exp_mifish_glommed) %>%
  data.frame() %>%
  rename("SampleID" = "Name.Deploy_Cartr")

mifish_aitchison_nmds_3d_df <- full_join(mifish_aitchison_nmds_3d_df, mifish_sample_data)

mifish_aitchison_nmds_3d_df
```

CO1
```{r}
# Extract points from nmds and merge into data frame with metadata 
co1_aitchison_nmds_3d_df <- data.frame(co1_aitchison_nmds_3d$points) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata 
co1_sample_data <- sample_data(ps2024_exp_co1_glommed) %>%
  data.frame() %>%
  mutate(SampleID = gsub("_$", "", gsub("\\.", "_", Name_Deploy_Cartr)))

co1_aitchison_nmds_3d_df <- full_join(co1_aitchison_nmds_3d_df, co1_sample_data)

co1_aitchison_nmds_3d_df
```

## Plot

```{r}
elas_aitchison_nmds_3d_plot <- plot_ly(data = elas_aitchison_nmds_3d_df,
  x    = ~MDS1,
  y    = ~MDS2,
  z    = ~MDS3,
  color  = ~Habitat,
  symbol = ~Night__Day,
  symbols = c('circle','square'),   
  marker = list(size = 5)
) %>%
  layout(
    title = paste0("Elas02 3D NMDS (stress = ", round(elas_aitchison_nmds_3d$stress, 2), ")"),
    scene = list(
      xaxis = list(title = "NMDS1"),
      yaxis = list(title = "NMDS2"),
      zaxis = list(title = "NMDS3")
    ),
    legend = list(title = list(text = "<b>Habitat / Night__Day</b>"))
  )

# save 
withr::with_dir('figures-expedition', htmlwidgets::saveWidget(as_widget(elas_aitchison_nmds_3d_plot), file="elas_aitchison_nmds_3d_plot.html", selfcontained = F))
```


<iframe src="elas_aitchison_nmds_3d_plot.html" width="800" height="600"></iframe> 

--> NMDS1 is major axis. What variable does it correlate to? (it's not habitat nor night/day)
--> Mostly open water sites so differentiation among habitats is difficult but there is clear separation between eelgrass and clam sanctuary



## Things to do
- Compare NMDS results to PCA
- 
- Follow most commonly sampled station (15? 20?)- diversity/ composition change through all 2 weeks, comparing day/night especially. follow composition (by bar?) and richness across all samples. annotate important events (day night, rain event, temp)
- Day night comparison: present in a way which removes samples that were only sampled in the day
- Present analysis of impact of RNAlater preservation (in 2024 trawl MiFish samples)
- Think about ways to plot SpOccupancy results (box and whisker plots of sig ASVs - covering mean and CI of logit score. axis shows night/day or east/west as neg/pos, etc)
- What is average distance across stations? Min? Max?




