---
title: "Expedition-eDNA-Ecol-analysis"
author: "Liz Suter"
date: "2025-03-27"
output: html_document
---

# Load packages

```{r}
# rm(list = ls())
library(tidyverse)
library(phyloseq)
library(vegan)
library(RColorBrewer)
library(microbiome)
library(metagMisc)
library(fantaxtic)
library(ggrepel)
library(paletteer)
library(pals)
library(compositions)
library(hms)
library(ggvegan)
```

# Load data from QC and Filtering Workbook
```{r}
load("figures-expedition/figures-expedition/environment_upto_export.RData")
```


# Visualizations

## Bar plots of top species from whole dataset
### Prepare data
#### Expedition/ Mifish
```{r}
# Aggregate by Common Name
ps2024_exp_mifish_species <- tax_glom(ps2024_exp_mifish, taxrank = "Species.CommonName") 

# Summarize total abundances for each species across the whole dataset
exp_mifish_species_abundance <- data.frame(otu_table(ps2024_exp_mifish_species)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_mifish_species))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))


exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Retain only most abundant sp
exp_mifish_species_abundance <- exp_mifish_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 12) # Show top 12 species

# Remove ASV column
exp_mifish_species_abundance <- exp_mifish_species_abundance %>% select(-ASV)

# Add columns indicating which library and sample type
exp_mifish_species_abundance$library <- "MiFish"
exp_mifish_species_abundance$sample_type <- "USV"


exp_mifish_species_abundance
```


#### Expedition/ Elas
```{r}
# Aggregate at the "Species" level
ps2024_exp_elas_species <- tax_glom(ps2024_exp_elas, taxrank = "Species.CommonName") 

# Summarize total abundances for each species across the whole dataset
exp_elas_species_abundance <- data.frame(otu_table(ps2024_exp_elas_species)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
exp_elas_species_abundance <- exp_elas_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_elas_species))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))

exp_elas_species_abundance <- exp_elas_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Plot only most abundant
exp_elas_species_abundance <- exp_elas_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 2) # Show top 2 species

# Remove ASV column
exp_elas_species_abundance <- exp_elas_species_abundance %>% select(-ASV)

# Add columns indicating which library and sample type
exp_elas_species_abundance$library <- "Elas02"
exp_elas_species_abundance$sample_type <- "USV"

exp_elas_species_abundance
```

#### Expedition/ CO1

There is too much going on here to make a similar figure of top species...

First, make a figure for top 10 phyla
```{r}
# Aggregate at the "Species" level
ps2024_exp_co1_phyla <- tax_glom(ps2024_exp_co1, taxrank = "Phylum") 

# Summarize total abundances for each species across the whole dataset
exp_co1_phyla_abundance <- data.frame(otu_table(ps2024_exp_co1_phyla)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_phyla))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))

exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Plot only most abundant
exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 10) # Show top 10 phyla

# Remove ASV column
exp_co1_phyla_abundance <- exp_co1_phyla_abundance %>% select(-ASV)

# Add columns indicating which library and sample type
exp_co1_phyla_abundance$library <- "CO1"
exp_co1_phyla_abundance$sample_type <- "USV"

exp_co1_phyla_abundance
```


Then separate out all Actinopteri and elasmos and aggregate at species level
```{r}
# Remove control samples
ps2024_exp_co1_nocontrols <- subset_samples(ps2024_exp_co1, is.na(controls))

# Filter to just Actinopteri
ps2024_exp_co1_actinopteri_elasmos <- subset_taxa(ps2024_exp_co1_nocontrols, Class %in% c("Actinopteri","Chondrichthyes"))

# Convert to dataframe
exp_co1_actinopteri_elasmos_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_actinopteri_elasmos)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_actinopteri_elasmos)) %>%
  mutate(ASV = rownames(.))

exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Group and sum
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  group_by(Species.CommonName) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()

# Remove unclassified Actinopteri
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  filter(!Species.CommonName == "Actinopteri")

# Calculate relative abundances across the whole dataset
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Plot only most abundant
exp_co1_actinopteri_elasmos_sp_abundance <- exp_co1_actinopteri_elasmos_sp_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 14) # Show top 14 species


# Add columns indicating which library and sample type
exp_co1_actinopteri_elasmos_sp_abundance$library <- "CO1"
exp_co1_actinopteri_elasmos_sp_abundance$sample_type <- "USV"



exp_co1_actinopteri_elasmos_sp_abundance
```

Make similar df for elasmobranchs
```{r}
# Filter to just Elasmos
ps2024_exp_co1_elasmos <- subset_taxa(ps2024_exp_co1_nocontrols, )

# Convert to dataframe
exp_co1_elasmo_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_elasmos)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_elasmos)) %>%
  mutate(ASV = rownames(.))

exp_co1_elasmo_sp_abundance <- exp_co1_elasmo_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Group and sum
exp_co1_elasmo_sp_abundance <- exp_co1_elasmo_sp_abundance %>%
  group_by(Species.CommonName) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()


# Calculate relative abundances across the whole dataset
exp_co1_elasmo_sp_abundance <- exp_co1_elasmo_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add columns indicating which library and sample type
exp_co1_elasmo_sp_abundance$library <- "CO1"
exp_co1_elasmo_sp_abundance$sample_type <- "USV"

exp_co1_elasmo_sp_abundance
```


Check other Chordates for anything interesting
```{r}
# Filter 
ps2024_exp_co1_chordates <- subset_taxa(ps2024_exp_co1_nocontrols,
  Phylum == "Chordata" & Class != "Actinopteri" & Class != "Chondrichthyes")

# Convert to dataframe
exp_co1_chordates_sp_abundance <- data.frame(otu_table(ps2024_exp_co1_chordates)) %>%
  mutate(ASV = rownames(.))

# Add back in tax information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_exp_co1_chordates)) %>%
  mutate(ASV = rownames(.))

exp_co1_chordates_sp_abundance <- exp_co1_chordates_sp_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))


# Group and Sum
exp_co1_chordates_sp_abundance <- exp_co1_chordates_sp_abundance %>%
  group_by(Species) %>%
  summarise(TotalAbundance = sum(across(where(is.numeric)))) %>%
  ungroup()


# Calculate relative abundances across the whole dataset
exp_co1_chordates_sp_abundance <- exp_co1_chordates_sp_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add columns indicating which library and sample type
exp_co1_chordates_sp_abundance$library <- "CO1"
exp_co1_chordates_sp_abundance$sample_type <- "USV"

exp_co1_chordates_sp_abundance

```
Mostly tunicates



#### Trawl/ Mifish
```{r}
# Aggregate at the "Species" level
ps2024_trawl_mifish_species <- tax_glom(ps2024_trawl_mifish, taxrank = "Species.CommonName") 

# Summarize total abundances for each species across the whole dataset
trawl_mifish_species_abundance <- data.frame(otu_table(ps2024_trawl_mifish_species)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_trawl_mifish_species))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))


trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Plot only most abundant
trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 12) # Show top 12 species

# Remove ASV column
trawl_mifish_species_abundance <- trawl_mifish_species_abundance %>% select(-ASV)

# Add columns indicating which library and sample type
trawl_mifish_species_abundance$library <- "MiFish"
trawl_mifish_species_abundance$sample_type <- "Hand Sample"

trawl_mifish_species_abundance
```

#### Trawl/ Elas
```{r}
# Aggregate at the "Species" level
ps2024_trawl_elas_species <- tax_glom(ps2024_trawl_elas, taxrank = "Species.CommonName") 

# Summarize total abundances for each species across the whole dataset
trawl_elas_species_abundance <- data.frame(otu_table(ps2024_trawl_elas_species)) %>%
  rowSums() %>%
  as.data.frame() %>%
  rownames_to_column(var = "ASV") %>%
  rename(TotalAbundance = ".")

# Calculate relative abundances across the whole dataset
trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance))

# Add back taxonomic information
rm(tax_info)
tax_info <- as.data.frame(tax_table(ps2024_trawl_elas_species))
tax_info <- tax_info %>%
  mutate(ASV = rownames(tax_info))

trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  left_join(tax_info, by = c("ASV" = "ASV"))

# Plot only most abundant
trawl_elas_species_abundance <- trawl_elas_species_abundance %>%
  arrange(desc(RelativeAbundance)) %>%
  slice_head(n = 2) # Show top 2 species

# Remove ASV column
trawl_elas_species_abundance <- trawl_elas_species_abundance %>% select(-ASV)

# Add columns indicating which library and sample type
trawl_elas_species_abundance$library <- "Elas02"
trawl_elas_species_abundance$sample_type <- "Hand Sample"

trawl_elas_species_abundance
```
#### Summary of top Species
Sum what % of total the top species comprise
```{r}
sum(exp_mifish_species_abundance$RelativeAbundance)
sum(trawl_mifish_species_abundance$RelativeAbundance)

sum(exp_elas_species_abundance$RelativeAbundance)
sum(trawl_elas_species_abundance$RelativeAbundance)

sum(exp_co1_phyla_abundance$RelativeAbundance)
sum(exp_co1_actinopteri_elasmos_sp_abundance$RelativeAbundance)
sum(exp_co1_elasmo_sp_abundance$RelativeAbundance)
sum(exp_co1_chordates_sp_abundance$RelativeAbundance)

```

### Visualize
Append the 5 datasets of actinopteri and elasmos for comparison
```{r}
most_abund_all_datasets <- exp_mifish_species_abundance %>%
  bind_rows(trawl_mifish_species_abundance) %>%
  bind_rows(exp_elas_species_abundance) %>%
  bind_rows(trawl_elas_species_abundance) %>% 
  bind_rows(exp_co1_actinopteri_elasmos_sp_abundance) %>% 
  select (-Color, -CommonName) # this formatting got messed up in some of the rows

commonnames_mod <- commonnames %>% select(-Species) %>% distinct()

# Add back in colors and common name
most_abund_all_datasets <- most_abund_all_datasets %>% 
  left_join(commonnames_mod, by = c("Species.CommonName" = "Species.CommonName"))


most_abund_all_datasets
```



Fix colors so they remain constant in following plots
Some [predefined color names](https://sape.inf.usi.ch/quick-reference/ggplot2/colour) for reference
I started a colors column in commonnames database to keep assignment consistent across notebooks, but they got lost from the df when aggregating. Join back in and format for plotting
```{r}

myColors_sciname <- setNames(commonnames$Color, commonnames$Species)
myColors_comname <- setNames(commonnames$Color, commonnames$CommonName)

head(myColors_sciname)
head(myColors_comname)
```

Plot stacked barplot
For report: MiFish and Elas02 only
```{r}
# Plot a stacked barplot- 
top_species_abundance_plot <- ggplot(most_abund_all_datasets %>% filter(!library == "CO1"), aes(x = sample_type, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::percent_format()) + 
  scale_fill_manual(values = myColors_comname) +
  facet_grid(~library) +
  labs(x = NULL, y = "Relative Abundance (%)", title = "Relative Abundance of Top Species: Expedition vs. Hand Samples") +
  theme_minimal() +
  guides(fill = guide_legend(ncol = 2)) +  # Adjust # of legend columns
  theme(legend.title = element_blank())
  
  

top_species_abundance_plot

ggsave(plot = top_species_abundance_plot, filename = "figures-expedition/top_species_abundance_plot.jpg")

```



For presentation: MiFish and Elas02 alongside CO1
```{r}
# Plot a stacked barplot- 
top_species_abundance_plot_CO1 <- ggplot(most_abund_all_datasets, aes(x = sample_type, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  scale_y_continuous(labels = scales::percent_format()) + 
  scale_fill_manual(values = myColors_comname) +
  facet_grid(~library, scales = "free_x", space = "free") +
  labs(x = NULL, y = "Relative Abundance (%)", title = "Relative Abundance of Top Species: Expedition vs. Hand Samples") +
  theme_minimal() +
  guides(fill = guide_legend(ncol = 2)) +  # Adjust # of legend columns
  theme(legend.title = element_blank())
  
  

top_species_abundance_plot_CO1

ggsave(plot = top_species_abundance_plot_CO1, filename = "figures-expedition/top_species_abundance_plot_with_CO1.jpg")

```




## Barplots by sample

### Exp/MiFish-

raw reads
```{r}
ps2024_exp_mifish_barplot <- plot_bar(ps2024_exp_mifish, x = "Name.Deploy_Cartr_Library", fill = "Order")
ps2024_exp_mifish_barplot
```

relative abundances
```{r}
ps2024_exp_ra_mifish_barplot <- plot_bar(ps2024_exp_mifish_ra, x = "Name.Deploy_Cartr_Library", fill = "Order")
ps2024_exp_ra_mifish_barplot
```

### Exp/Elas02-

raw reads
```{r}
ps2024_exp_elas_barplot <- plot_bar(ps2024_exp_elas, x = "Name_Deploy_Cartr_Library", fill = "Species")
ps2024_exp_elas_barplot
```

relative abundances
```{r}
ps2024_exp_ra_elas_barplot <- plot_bar(ps2024_exp_elas_ra, x = "Name_Deploy_Cartr_Library", fill = "Species")
ps2024_exp_ra_elas_barplot
```

### Exp/CO1-

raw reads
```{r}
ps2024_exp_co1_barplot <- plot_bar(ps2024_exp_co1, x = "Name_Deploy_Cartr", fill = "Phylum")
ps2024_exp_co1_barplot
```

relative abundances
```{r}
# ps2024_exp_ra_co1_barplot <- plot_bar(ps2024_exp_co1_ra, x = "Name_Deploy_Cartr", fill = "Phylum")
# ps2024_exp_ra_co1_barplot
```


### Trawl/MiFish-

raw reads
```{r}
ps2024_trawl_mifish_barplot <- plot_bar(ps2024_trawl_mifish, fill = "Order")
ps2024_trawl_mifish_barplot
```

relatve abundances
```{r}
ps2024_trawl_ra_mifish_barplot <- plot_bar(ps2024_trawl_mifish_ra, fill = "Order")
ps2024_trawl_ra_mifish_barplot
```

### Trawl/Elas02-

raw reads
```{r}
ps2024_trawl_elas_barplot <- plot_bar(ps2024_trawl_elas, fill = "Species")
ps2024_trawl_elas_barplot
```

relatve abundances
```{r}
ps2024_trawl_ra_elas_barplot <- plot_bar(ps2024_trawl_elas_ra, fill = "Species")
ps2024_trawl_ra_elas_barplot
```


## Examine controls 

### Prepare data for positive controls
Extract positive controls and convert to dataframe
NOTE- there are not Elasmo pos controls in this dataset
```{r}
ps2024_exp_mifish_controls <- subset_samples(ps2024_exp_mifish, controls == "positive")
ps2024_exp_co1_controls <- subset_samples(ps2024_exp_co1, controls == "positive")
ps2024_trawl_mifish_controls <- subset_samples(ps2024_trawl_mifish, controls == "positive")

ps2024_exp_mifish_controls_long_df <- ps2024_exp_mifish_controls %>%
  psmelt() %>%
  as_tibble()

ps2024_exp_co1_controls_long_df <- ps2024_exp_co1_controls %>%
  psmelt() %>%
  as_tibble()

ps2024_trawl_mifish_controls_long_df <- ps2024_trawl_mifish_controls %>%
  psmelt() %>%
  as_tibble()

ps2024_exp_mifish_controls_long_df
ps2024_exp_co1_controls_long_df
ps2024_trawl_mifish_controls_long_df
```


#### Exp/MiFish-

Plot a stacked barplot- 
```{r}
# Sum by common name, calculate total abundance and relative abundance
ps2024_exp_mifish_controls_long_df_sum <- ps2024_exp_mifish_controls_long_df %>%
  group_by(CommonName, Name.Deploy_Cartr_Library) %>%
  summarise(TotalAbundance = sum(Abundance, na.rm = TRUE), .groups = "drop_last") %>%
  group_by(Name.Deploy_Cartr_Library) %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(TotalAbundance > 0) %>%
  arrange(desc(TotalAbundance))

# Pos Controls by read abundance
ps2024_exp_mifish_posctl_barplot <- ggplot(ps2024_exp_mifish_controls_long_df_sum, 
       aes(x = Name.Deploy_Cartr_Library, y = TotalAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_mifish_posctl_barplot

# Pos Controls by relative abundance
ps2024_exp_mifish_posctl_ra_barplot <- ggplot(ps2024_exp_mifish_controls_long_df_sum, 
       aes(x = Name.Deploy_Cartr_Library, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Relative Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_mifish_posctl_ra_barplot


ggsave(plot = ps2024_exp_mifish_posctl_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_barplot.jpg", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_ra_barplot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = ps2024_exp_mifish_posctl_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_barplot.eps", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_exp_mifish_posctl_ra_barplot.eps", width = 4, height = 4, units = "in")

```

#### Exp/CO1-
(Note- there are no pos ctls for elasmobranchs in 2024)
Plot a stacked barplot- 
```{r}
# Sum by common name, calculate total abundance and relative abundance
ps2024_exp_co1_controls_long_df_sum <- ps2024_exp_co1_controls_long_df %>%
  group_by(CommonName, Name_Deploy_Cartr) %>%
  summarise(TotalAbundance = sum(Abundance, na.rm = TRUE), .groups = "drop_last") %>%
  filter(!is.na(CommonName)) %>%  # Remove NAs BEFORE calculating relative abundance
  group_by(Name_Deploy_Cartr) %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(TotalAbundance > 0) %>%
  arrange(desc(TotalAbundance))


# Pos Controls by read abundance
ps2024_exp_co1_posctl_barplot <- ggplot(ps2024_exp_co1_controls_long_df_sum, 
       aes(x = Name_Deploy_Cartr, y = TotalAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_co1_posctl_barplot

# Pos Controls by relative abundance
ps2024_exp_co1_posctl_ra_barplot <- ggplot(ps2024_exp_co1_controls_long_df_sum, 
       aes(x = Name_Deploy_Cartr, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 Expedition, Pos. Controls: Relative Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_exp_co1_posctl_ra_barplot


ggsave(plot = ps2024_exp_co1_posctl_barplot, filename = "figures-expedition/2024_exp_co1_posctl_barplot.jpg", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_co1_posctl_ra_barplot, filename = "figures-expedition/2024_exp_co1_posctl_ra_barplot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = ps2024_exp_co1_posctl_barplot, filename = "figures-expedition/2024_exp_co1_posctl_barplot.eps", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_exp_co1_posctl_ra_barplot, filename = "figures-expedition/2024_exp_co1_posctl_ra_barplot.eps", width = 4, height = 4, units = "in")

```


#### Trawl/MiFish-
(again, no Elasmo pos controls from hand samples)

Plot a stacked barplot- 
```{r}
# Sum by common name, calculate total abundance and relative abundance
ps2024_trawl_mifish_controls_long_df_sum <- ps2024_trawl_mifish_controls_long_df %>%
  group_by(CommonName, replicates) %>%
  summarise(TotalAbundance = sum(Abundance, na.rm = TRUE), .groups = "drop_last") %>%
  group_by(replicates) %>%
  mutate(RelativeAbundance = TotalAbundance / sum(TotalAbundance, na.rm = TRUE)) %>%
  ungroup() %>%
  filter(TotalAbundance > 0) %>%
  arrange(desc(TotalAbundance))

# Pos Controls by read abundance
ps2024_trawl_mifish_posctl_barplot <- ggplot(ps2024_trawl_mifish_controls_long_df_sum, 
       aes(x = replicates, y = TotalAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 trawl, Pos. Controls: Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_trawl_mifish_posctl_barplot

# Pos Controls by relative abundance
ps2024_trawl_mifish_posctl_ra_barplot <- ggplot(ps2024_trawl_mifish_controls_long_df_sum, 
       aes(x = replicates, y = RelativeAbundance, fill = CommonName)) +
  geom_bar(stat = "identity") +
  labs(title = "2024 trawl, Pos. Controls: Relative Read Abund.",
       x = NULL, fill = NULL) +
  scale_y_continuous(name="Read Abundance") +
  theme(axis.title.x=element_blank(), legend.title = element_blank(), axis.text.x = element_blank()) +
  scale_fill_manual(values = myColors_comname) +
  theme_minimal()
ps2024_trawl_mifish_posctl_ra_barplot


ggsave(plot = ps2024_trawl_mifish_posctl_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_barplot.jpg", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_trawl_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_ra_barplot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = ps2024_trawl_mifish_posctl_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_barplot.eps", width = 4, height = 4, units = "in")
ggsave(plot = ps2024_trawl_mifish_posctl_ra_barplot, filename = "figures-expedition/2024_trawl_mifish_posctl_ra_barplot.eps", width = 4, height = 4, units = "in")

```



### CONTINUE HERE, 3/27/25

Remove Pos and Neg  Controls from all phyloseq objects:
(Note, they were already removed from Elas objects just based on quality filtering above)

ASV-based, reads:
ps2024_exp_mifish

ASV-based, relative abundance:
ps2024_exp_mifish_ra

Taxonomy (species)- based, reads:
ps2024_exp_mifish_glommed


Taxonomy (species)- based, relative abundance:
ps2024_exp_mifish_glommed_ra


```{r}
ps2024_exp_mifish <- subset_samples(ps2024_exp_mifish, is.na(controls))
ps2024_exp_mifish_ra <- subset_samples(ps2024_exp_mifish_ra, is.na(controls))
ps2024_exp_mifish_glommed <- subset_samples(ps2024_exp_mifish_glommed, is.na(controls))
ps2024_exp_mifish_glommed_ra <- subset_samples(ps2024_exp_mifish_glommed_ra, is.na(controls))
```





## Bubble plots

Based on my previous [scripts](https://github.com/lizsuter/Cariaco_Euk)

Work with tax-based dataframes (rather than ASV-based):

Taxonomy (species)- based, reads:
ps2024_exp_mifish_glommed

Taxonomy (species)- based, relative abundance:
ps2024_exp_mifish_glommed_ra

### Expedition/MiFish-
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_mifish_glommed_df <- psmelt(ps2024_exp_mifish_glommed)
ps2024_exp_mifish_glommed_ra_df <- psmelt(ps2024_exp_mifish_glommed_ra)

# replace zeroes in the table with NA
ps2024_exp_mifish_glommed_df[ps2024_exp_mifish_glommed_df == 0] <- NA
ps2024_exp_mifish_glommed_ra_df[ps2024_exp_mifish_glommed_ra_df == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_mifish_glommed_df <-  filter(ps2024_exp_mifish_glommed_df, !is.na(Abundance))
ps2024_exp_mifish_glommed_ra_df <-  filter(ps2024_exp_mifish_glommed_ra_df, !is.na(Abundance))


# and view
ps2024_exp_mifish_glommed_df
ps2024_exp_mifish_glommed_ra_df
```

Elas02-
```{r}
# convert ps object to dataframe using phyloseq's psmelt
ps2024_exp_elas_glommed_df <- psmelt(ps2024_exp_elas_glommed)
ps2024_exp_elas_glommed_ra_df <- psmelt(ps2024_exp_elas_glommed_ra)

# replace zeroes in the table with NA
ps2024_exp_elas_glommed_df[ps2024_exp_elas_glommed_df == 0] <- NA
ps2024_exp_elas_glommed_ra_df[ps2024_exp_elas_glommed_ra_df == 0] <- NA

# and remove rows with NAs (so they don't appear as small dots in plot)
ps2024_exp_elas_glommed_df <-  filter(ps2024_exp_elas_glommed_df, !is.na(Abundance))
ps2024_exp_elas_glommed_ra_df <-  filter(ps2024_exp_elas_glommed_ra_df, !is.na(Abundance))

# and view
ps2024_exp_elas_glommed_df
ps2024_exp_elas_glommed_ra_df
```



Plot species-common name by site

MiFish-

```{r}

# First sort the site names in a sensible way
levels(unique(c(ps2024_exp_mifish_glommed_df$sites)))
sitelevels <- as.character(sort(as.numeric(levels(unique(c(ps2024_exp_mifish_glommed_df$sites))))))
sitelevels

# Order E-W in sensible way by making it a factor
ps2024_exp_mifish_glommed_df$Bayside_f= factor(ps2024_exp_mifish_glommed_df$Bayside, levels=c('W','E'))

```



Plot by bayside
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_bayside <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Bayside_f, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_bayside



bubbleplot_comname_2024_expedition_mifish_Bayside <- ggplot(ps2024_exp_mifish_glommed_df_bayside, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_Bayside


  
  
# save
ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Bayside.eps", width = 10, height = 7, units = "in")

ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Bayside.jpg", width = 10, height = 7, units = "in")
```

Plot by DayNight
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_daynight <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Night__Day, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_daynight


bubbleplot_comname_2024_expedition_mifish_DayNight <- ggplot(ps2024_exp_mifish_glommed_df_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(~Night__Day, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_DayNight


ggsave(plot = bubbleplot_comname_2024_expedition_mifish_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_DayNight.eps", width = 10, height = 7, units = "in")
```

Plot by Habitat
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_habitat <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Habitat, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_habitat


bubbleplot_comname_2024_expedition_mifish_Habitat <- ggplot(ps2024_exp_mifish_glommed_df_habitat, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(~Habitat, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_Habitat

ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Habitat, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Habitat.eps", width = 12, height = 7, units = "in")
```
Plot by Bayside and Day-Night
```{r}
# First average replicates at same sites
ps2024_exp_mifish_glommed_df_bayside_daynight <- ps2024_exp_mifish_glommed_df %>%
    group_by(sites, Bayside_f, Night__Day, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_mifish_glommed_df_bayside_daynight


bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight <- ggplot(ps2024_exp_mifish_glommed_df_bayside_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(Night__Day~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight




ggsave(plot = bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_mifish_Bayside_DayNight.eps", width = 10, height = 9, units = "in")

```





### Elas02-

```{r}

# First sort the site names in a sensible way
levels(unique(c(ps2024_exp_elas_glommed_df$sites)))
sitelevels <- as.character(sort(as.numeric(levels(unique(c(ps2024_exp_elas_glommed_df$sites))))))
sitelevels

# Order E-W in sensible way by making it a factor
ps2024_exp_elas_glommed_df$Bayside_f= factor(ps2024_exp_elas_glommed_df$Bayside, levels=c('W','E'))

```



Plot by bayside
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_bayside <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Bayside_f, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_bayside



bubbleplot_comname_2024_expedition_elas_Bayside <- ggplot(ps2024_exp_elas_glommed_df_bayside, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_Bayside

# save
ggsave(plot = bubbleplot_comname_2024_expedition_elas_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Bayside.eps", width = 8, height = 2.5, units = "in")

ggsave(plot = bubbleplot_comname_2024_expedition_elas_Bayside, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Bayside.jpg", width = 8, height = 2.5, units = "in")
```

Plot by DayNight
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_daynight <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Night__Day, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_daynight


bubbleplot_comname_2024_expedition_elas_DayNight <- ggplot(ps2024_exp_elas_glommed_df_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(~Night__Day, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_DayNight


ggsave(plot = bubbleplot_comname_2024_expedition_elas_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_DayNight.eps", width = 10, height = 2.5, units = "in")
```

Plot by Habitat
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_habitat <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Habitat, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_habitat


bubbleplot_comname_2024_expedition_elas_Habitat <- ggplot(ps2024_exp_elas_glommed_df_habitat, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(~Habitat, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_Habitat

ggsave(plot = bubbleplot_comname_2024_expedition_elas_Habitat, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Habitat.eps", width = 10, height = 2.5, units = "in")
```
Plot by Bayside and Day-Night
```{r}
# First average replicates at same sites
ps2024_exp_elas_glommed_df_bayside_daynight <- ps2024_exp_elas_glommed_df %>%
    group_by(sites, Bayside_f, Night__Day, Species, CommonName, `Species.CommonName`) %>%
    dplyr::summarize(Abundance = mean(Abundance))
ps2024_exp_elas_glommed_df_bayside_daynight


bubbleplot_comname_2024_expedition_elas_Bayside_DayNight <- ggplot(ps2024_exp_elas_glommed_df_bayside_daynight, aes(x = factor(sites, levels = sitelevels), y = fct_rev(`Species.CommonName`))) + # the fancy stuff around y (`Species-CommonName`) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = `CommonName`), color = "black", pch = 21) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance", fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank(), axis.text.x = element_text(size = 12), axis.text.y = element_text(size = 9)) +
  guides(fill="none") +
  facet_grid(Night__Day~Bayside_f, scales = "free", space = "free", drop= TRUE)
bubbleplot_comname_2024_expedition_elas_Bayside_DayNight




ggsave(plot = bubbleplot_comname_2024_expedition_elas_Bayside_DayNight, filename = "figures-expedition/bubbleplot_comname_2024_expedition_elas_Bayside_DayNight.eps", width = 10, height = 4, units = "in")

```






## Fractional Abundance Plots

Summarize fractional abundances across whole dataset-
```{r}
ps2024_exp_mifish_whole <- ps2024_exp_mifish_glommed_df %>%
  group_by(Species.CommonName) %>%
    summarize(total_reads = sum(Abundance)) %>%
  mutate(rel_abun = total_reads/sum(across(total_reads))) %>%
  mutate(rel_abun_percent = rel_abun*100) %>%
  mutate(csum = rev(cumsum(rev(rel_abun))), 
         pos = rel_abun/2 + lead(csum, 1),
         pos = if_else(is.na(pos), rel_abun/2, pos))

ps2024_exp_mifish_whole

```



Apparently there is no geom for pie charts but you can [build a stacked geom_bar and make polar coordinates](https://r-graph-gallery.com/piechart-ggplot2.html)

I also found a color palette package with a few options for large number of diverging cartegories, [pals](https://github.com/kwstat/pals)

```{r}
# first filter out anything <1% rel abun
ps2024_exp_mifish_whole_forplot <- ps2024_exp_mifish_whole %>%
  filter(!rel_abun_percent < 1)

# Turn names into factor for plotting in correct order
ps2024_exp_mifish_whole_forplot$CommonName_f= factor(ps2024_exp_mifish_whole_forplot$Species.CommonName, levels=sort(unique(ps2024_exp_mifish_whole_forplot$Species.CommonName)))

# add "Other" row at bottom of factors which sums all taxa <1%
ps2024_exp_mifish_whole_forplot <- ps2024_exp_mifish_whole_forplot %>%
  add_row(`Species.CommonName` = "Other",
          `CommonName_f` = "Other",
          rel_abun = 1-sum(ps2024_exp_mifish_whole_forplot$rel_abun), 
          rel_abun_percent = 100-sum(ps2024_exp_mifish_whole_forplot$rel_abun_percent))
ps2024_exp_mifish_whole_forplot


# plot
piechart_2024_exp_mifish_all <- ggplot(ps2024_exp_mifish_whole_forplot, 
                                aes(x= "", 
                                    y = rel_abun_percent, 
                                    fill=factor(`Species.CommonName`, levels = `CommonName_f`))) +
  geom_bar(stat="identity", width=1) +
  coord_polar("y", start=0) +
  xlab("")+
  ylab("")+
  labs(fill = "")+
  theme_bw() +
  scale_fill_manual(values = myColors_comname)+
  theme(axis.title.x=element_blank()) +
  geom_text(aes(label = round(rel_abun_percent,1)),
            position = position_stack(vjust = 0.5))

piechart_2024_exp_mifish_all


ggsave(plot = piechart_2024_exp_mifish_all, filename = "figures-expedition/piechart_2024_exp_mifish_all.eps", width = 12, height = 7, units = "in")

```


Alternatively, Krona plot
Following [this](https://github.com/markschl/embed_krona/blob/main/example.Rmd)
Compatible with phyloseq objects
Must run in the console:
```{r}
 # source('https://raw.githubusercontent.com/markschl/embed_krona/master/embed_krona.R')
 # plot_krona(ps2024_exp_mifish_glommed)
 # plot_krona(ps2024_exp_elas_glommed)
 
```

Manually changed the names in folder to 
Generates the [MiFish krona plot](krona_files/krona_1.html) and [Elas02 krona plot](krona_files/krona_2.html) as an html file in current working directory. 




## Replication
How do replicates compare?

```{r}
sample_data(ps2024_exp_mifish)$replicates
# 90 samples, with 73 unique levels. So there are 17 samples which were collected in duplicate

duplicate_IDs <- sample_data(ps2024_exp_mifish)$replicates[duplicated(sample_data(ps2024_exp_mifish)$replicates)]
duplicate_IDs

# make new phyloseq object with only replicates
ps2024_exp_reps <- subset_samples(ps2024_exp_mifish_glommed, replicates %in% duplicate_IDs)
ps2024_exp_reps_ra <- subset_samples(ps2024_exp_mifish_glommed_ra, replicates %in% duplicate_IDs)


ps2024_exp_barplot_replicates <- plot_bar(ps2024_exp_reps, x = "Name.Deploy_Cartr_Library", fill = "CommonName")+
  scale_fill_manual(values = myColors_comname)
ps2024_exp_barplot_replicates

ps2024_exp_ra_barplot_replicates_ra <- plot_bar(ps2024_exp_reps_ra, x = "Name.Deploy_Cartr_Library", fill = "CommonName")+
  scale_fill_manual(values = myColors_comname)
ps2024_exp_ra_barplot_replicates_ra

# view replicates pair-by-pair
ps2024_exp_ra_barplot_replicates_ra <- ps2024_exp_ra_barplot_replicates_ra +
    facet_grid(~replicates, scales = "free", space = "free", drop= TRUE)+
    scale_fill_manual(values = myColors_comname)+
    theme(legend.position = "none", axis.title.x = element_blank(), axis.text.x = element_text(size = 12), strip.text = element_text(size = 12))
ps2024_exp_ra_barplot_replicates_ra


ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside.eps", width = 12, height = 7, units = "in")

ggsave(plot = ps2024_exp_ra_barplot_replicates_ra, filename = "figures-expedition/barplot_2024_expedition_replicates_sidebyside.jpg", width = 12, height = 7, units = "in")
```

- Some replicates look very close to each other (~10 of the 17) while others do not. I manually checked the Sample Inventory. One issue may be filtration volume- these are not always consistent bc of the issues with overpressuring
- I also notice that the more diverse the sample is, the less similar the replicates are. eg. 5_19 and 8_11a

### Aitchison dissimilarity among repilcates


For [compositional data](https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2017.02224/full), use Aitchison distance on CLR-transformed data


From [vegan](https://rdrr.io/cran/vegan/man/vegdist.html) documentation, "aitchison" method in `vegdist` automatically does the clr transformation before calculating the distance matrix. Phyloseq has [some options](https://joey711.github.io/phyloseq/distance.html) to do run vegdist with a phyloseq object using a `distance` function, however aitchison is not listed:
```{r}
dist_methods <- unlist(distanceMethodList)
print(dist_methods)
```

On the other hand, from the vegan documentation, Aitchison distance is just the Euclidean distance on clr-transformed data, so I can do a transformation first and then the Euclidean distance calculation:
```{r}
ps2024_exp_reps_clr <- microbiome::transform(ps2024_exp_reps, transform = 'clr')
ps2024_exp_reps_dist <- distance(ps2024_exp_reps_clr, method= "euclidean")

ps2024_exp_reps_dist_mat <- as.matrix(ps2024_exp_reps_dist)

ps2024_exp_reps_dist_df <- as.data.frame(ps2024_exp_reps_dist_mat)
ps2024_exp_reps_dist_df

write.csv(ps2024_exp_reps_dist_df,"figures-expedition/mifish_2024_exp_aitchisondistance_replicates.csv", row.names = TRUE)
```
Analyzed the above in Excel in the file, [mifish_2024_exp_aitchisondistance_replicates.xlsx](figures-expedition/mifish_2024_exp_aitchisondistance_replicates.xlsx)
- Highlighted dissimilarity index between replicates
- Calculated average dissimilarity between replicates = 12.18479861; SD = 2.75176089

Get dissimilarity indices and their average across whole dataset, just for comparison:
```{r}
ps2024_exp_clr <- microbiome::transform(ps2024_exp_mifish, transform = 'clr')
ps2024_exp_dist <- distance(ps2024_exp_clr, method= "euclidean")

ps2024_exp_dist_mat <- as.matrix(ps2024_exp_dist)

ps2024_exp_dist_df <- as.data.frame(ps2024_exp_dist_mat)
ps2024_exp_dist_df

# mean of all distance indices?
mean(ps2024_exp_dist)
sd(ps2024_exp_dist)

```
On average, across the whole dataset, the dissimilarity is 13.4 (SD 2.1) while the average dissimilarity between replicates is 12.2 (SD 2.8). It's good that the replicates are less dissimilar than the on average but these numbers don't seem very far apart, especially taking into account the standard deviation


## Site Occupancy

Present Presence/ Absence as site occupancy, eg. across all samples from XX habitat type or east/west etc etc, how many times did species X appear. Check out [Lawson Handley et al. 2019](https://onlinelibrary.wiley.com/doi/full/10.1002/edn3.5)

### MiFIsh

```{r}
# How many total samples taken in each habitat type?
Total_samples_habitat <- as.data.frame(samples_2024_exp_mifish) %>%
  group_by(Habitat) %>%
  count(name = "Habitat_sample_total") %>%
  na.omit()

# in east vs west?
Total_samples_EW <- as.data.frame(samples_2024_exp_mifish) %>%
  group_by(Bayside) %>%
  count(name = "Bayside_sample_total") %>%
  filter(!Bayside=='Control')

# in day vs night?
Total_samples_DayNight <- as.data.frame(samples_2024_exp_mifish) %>%
  group_by(Night__Day) %>%
  count(name = "Night_day_sample_total") %>%
  na.omit()

Total_samples_habitat
Total_samples_EW
Total_samples_DayNight
```

```{r}
# join sample totals to dataframe so there is a denomenator for the calculation
ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df, Total_samples_habitat, by = "Habitat")
ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Total_samples_EW, by = "Bayside")
ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Total_samples_DayNight, by = "Night__Day")


ps2024_exp_mifish_glommed_df_2

```

Calculate site occupancy for each species
```{r}
# Habitat
Species_occurences_habitat <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat) %>%
  count(name = "Sp_occ_in_habitat")

ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Species_occurences_habitat, by = c("Habitat"="Habitat","Species.CommonName"="Species.CommonName"))

#Bayside
Species_occurences_bayside <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside) %>%
  count(name = "Sp_occ_in_bayside")

ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Species_occurences_bayside, by = c("Bayside"="Bayside","Species.CommonName"="Species.CommonName"))

#Daynight
Species_occurences_daynight <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day) %>%
  count(name = "Sp_occ_in_DayNight")

ps2024_exp_mifish_glommed_df_2 <- full_join(ps2024_exp_mifish_glommed_df_2, Species_occurences_daynight, by = c("Night__Day"="Night__Day","Species.CommonName"="Species.CommonName"))


# Calculate occurrence over sample total
Species_occurences_habitat <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat, Habitat_sample_total, Sp_occ_in_habitat) %>%
  mutate(Species_occurences_in_habitat = Sp_occ_in_habitat/Habitat_sample_total) %>%
  select(Species.CommonName, CommonName, Habitat, Species_occurences_in_habitat) %>%
  distinct()
Species_occurences_habitat

Species_occurences_bayside <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside, Bayside_sample_total, Sp_occ_in_bayside) %>%
  mutate(Species_occurences_in_bayside = Sp_occ_in_bayside/Bayside_sample_total) %>%
  select(Species_occurences_in_bayside, CommonName, Bayside, Species_occurences_in_bayside) %>%
  distinct()
Species_occurences_bayside

Species_occurences_daynight <- ps2024_exp_mifish_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day, Night_day_sample_total, Sp_occ_in_DayNight) %>%
  mutate(Species_occurences_in_daynight = Sp_occ_in_DayNight/Night_day_sample_total) %>%
  select(Species.CommonName, CommonName, Night__Day, Species_occurences_in_daynight) %>%
  distinct()
Species_occurences_daynight


```

Plot
```{r}
speciesoccurrence_habitat <- ggplot(data = Species_occurences_habitat, aes(x = CommonName, y = Species_occurences_in_habitat, fill = Habitat))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
speciesoccurrence_habitat

speciesoccurrence_daynight <- ggplot(data = Species_occurences_daynight, aes(x = CommonName, y = Species_occurences_in_daynight, fill = Night__Day))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
speciesoccurrence_daynight

speciesoccurrence_bayside <- ggplot(data = Species_occurences_bayside, aes(x = CommonName, y = Species_occurences_in_bayside, fill = Bayside))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("coral", "cornflowerblue"))
speciesoccurrence_bayside


ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat.eps", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight.eps", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside.eps", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat.jpg", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight.jpg", width = 12, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside.jpg", width = 12, height = 5, units = "in")

```

### Elas02

```{r}
# How many total samples taken in each habitat type?
Total_samples_habitat <- as.data.frame(samples_2024_exp_elas) %>%
  group_by(Habitat) %>%
  count(name = "Habitat_sample_total") %>%
  na.omit()

# in east vs west?
Total_samples_EW <- as.data.frame(samples_2024_exp_elas) %>%
  group_by(Bayside) %>%
  count(name = "Bayside_sample_total") %>%
  filter(!Bayside=='Control')

# in day vs night?
Total_samples_DayNight <- as.data.frame(samples_2024_exp_elas) %>%
  group_by(Night__Day) %>%
  count(name = "Night_day_sample_total") %>%
  na.omit()

Total_samples_habitat
Total_samples_EW
Total_samples_DayNight
```



```{r}
# join sample totals to dataframe so there is a denomenator for the calculation
ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df, Total_samples_habitat, by = "Habitat")
ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Total_samples_EW, by = "Bayside")
ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Total_samples_DayNight, by = "Night__Day")


# there are no elasmos detected at oyster reef sites so there is an empty row. Delete
ps2024_exp_elas_glommed_df_2 <- ps2024_exp_elas_glommed_df_2 %>% filter(!is.na(OTU))


ps2024_exp_elas_glommed_df_2

```


Calculate site occupancy for each species
```{r}
# Habitat
Species_occurences_habitat <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat) %>%
  count(name = "Sp_occ_in_habitat")

ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Species_occurences_habitat, by = c("Habitat"="Habitat","Species.CommonName"="Species.CommonName"))

#Bayside
Species_occurences_bayside <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside) %>%
  count(name = "Sp_occ_in_bayside")

ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Species_occurences_bayside, by = c("Bayside"="Bayside","Species.CommonName"="Species.CommonName"))

#Daynight
Species_occurences_daynight <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day) %>%
  count(name = "Sp_occ_in_DayNight")

ps2024_exp_elas_glommed_df_2 <- full_join(ps2024_exp_elas_glommed_df_2, Species_occurences_daynight, by = c("Night__Day"="Night__Day","Species.CommonName"="Species.CommonName"))


# Calculate occurrence over sample total
Species_occurences_habitat <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Habitat, Habitat_sample_total, Sp_occ_in_habitat) %>%
  mutate(Species_occurences_in_habitat = Sp_occ_in_habitat/Habitat_sample_total) %>%
  select(Species.CommonName, CommonName, Habitat, Species_occurences_in_habitat) %>%
  distinct()
Species_occurences_habitat

Species_occurences_bayside <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Bayside, Bayside_sample_total, Sp_occ_in_bayside) %>%
  mutate(Species_occurences_in_bayside = Sp_occ_in_bayside/Bayside_sample_total) %>%
  select(Species.CommonName, CommonName, Bayside, Species_occurences_in_bayside) %>%
  distinct()
Species_occurences_bayside

Species_occurences_daynight <- ps2024_exp_elas_glommed_df_2 %>%
  group_by(Species.CommonName, Night__Day, Night_day_sample_total, Sp_occ_in_DayNight) %>%
  mutate(Species_occurences_in_daynight = Sp_occ_in_DayNight/Night_day_sample_total) %>%
  select(Species.CommonName, CommonName, Night__Day, Species_occurences_in_daynight) %>%
  distinct()
Species_occurences_daynight


```



Plot
```{r}
speciesoccurrence_habitat <- ggplot(data = Species_occurences_habitat, aes(x = CommonName, y = Species_occurences_in_habitat, fill = Habitat))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
speciesoccurrence_habitat

speciesoccurrence_daynight <- ggplot(data = Species_occurences_daynight, aes(x = CommonName, y = Species_occurences_in_daynight, fill = Night__Day))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
speciesoccurrence_daynight

speciesoccurrence_bayside <- ggplot(data = Species_occurences_bayside, aes(x = CommonName, y = Species_occurences_in_bayside, fill = Bayside))+
  geom_bar(stat = "identity", position = position_dodge(preserve = "single"), color = "black", linewidth = 0.25)+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        legend.title = element_blank(),
        text = element_text(size = 14)) +
  ylab("Species Occurrence")+
  xlab("")+
  scale_fill_manual(values = c("coral", "cornflowerblue"))
speciesoccurrence_bayside


ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat_elas.eps", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight_elas.eps", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside_elas.eps", width = 5, height = 5, units = "in")




ggsave(plot = speciesoccurrence_habitat, filename = "figures-expedition/speciesoccurrence_habitat_elas.jpg", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_daynight, filename = "figures-expedition/speciesoccurrence_daynight_elas.jpg", width = 5, height = 5, units = "in")

ggsave(plot = speciesoccurrence_bayside, filename = "figures-expedition/speciesoccurrence_bayside_elas.jpg", width = 5, height = 5, units = "in")

```


# Species Richness- 
```{r}
species_counts_exp_mifish <- ps2024_exp_mifish_glommed_df %>%
  group_by(Sample, Deployment, sites, replicates, site_altname, Volume_liter, lat, long, Bayside_f, Habitat, Site_type, Night__Day, Datecode, Month, Day, Avg_Depth_m, Avg_Temperature_C, Avg_Salinity_psu, Avg_Chlorophyll_ugL, Avg_Turbidity_NTU) %>%
  summarise(UniqueSpeciesCount = n_distinct(Species), .groups = "drop")

species_counts_exp_elas <- ps2024_exp_elas_glommed_df %>%
  group_by(Sample, Deployment, sites, replicates, Volume_liter, lat, long, Bayside_f, Habitat, Site_type, Night__Day, Datecode, Month, Day, Avg_Depth_m, Avg_Temperature_C, Avg_Salinity_psu, Avg_Chlorophyll_ugL, Avg_Turbidity_NTU) %>%
  summarise(UniqueSpeciesCount = n_distinct(Species), .groups = "drop")


species_counts_exp_mifish
species_counts_exp_elas

```


## Plot avg sp richness by bayside, habitat, etc.
First calculate averages
```{r}
species_stats_by_habitat_mifish <- species_counts_exp_mifish %>%
  group_by(Habitat) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_habitat_mifish

species_stats_by_bayside_mifish <- species_counts_exp_mifish %>%
  group_by(Bayside_f) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_bayside_mifish


species_stats_by_habitat_elas <- species_counts_exp_elas %>%
  group_by(Habitat) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_habitat_elas

species_stats_by_bayside_elas <- species_counts_exp_elas %>%
  group_by(Bayside_f) %>%
  summarise(
    AverageUniqueSpeciesCount = mean(UniqueSpeciesCount, na.rm = TRUE),
    StdDevUniqueSpeciesCount = sd(UniqueSpeciesCount, na.rm = TRUE)
  )

species_stats_by_bayside_elas
```
Plot
```{r}
# Put line breaks in x-tick labels for plotting
species_stats_by_habitat_mifish$Habitat <- gsub(" ", "\n", species_stats_by_habitat_mifish$Habitat)
species_stats_by_habitat_elas$Habitat <- gsub(" ", "\n", species_stats_by_habitat_elas$Habitat)


#Plot
species_richness_by_habitat_mifish_plot <- 
ggplot(species_stats_by_habitat_mifish, aes(x = Habitat, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_habitat_mifish_plot

species_richness_by_bayside_mifish_plot <- 
ggplot(species_stats_by_bayside_mifish, aes(x = Bayside_f, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("cornflowerblue", "coral"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_bayside_mifish_plot

species_richness_by_habitat_elas_plot <- 
ggplot(species_stats_by_habitat_elas, aes(x = Habitat, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("azure2", "chartreuse4", "cadetblue1"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_habitat_elas_plot

species_richness_by_bayside_elas_plot <- 
ggplot(species_stats_by_bayside_elas, aes(x = Bayside_f, y = AverageUniqueSpeciesCount)) +
  geom_bar(stat = "identity", fill = c("cornflowerblue", "coral"), alpha = 0.8) +  
  geom_errorbar(aes(ymin = AverageUniqueSpeciesCount - StdDevUniqueSpeciesCount,
                    ymax = AverageUniqueSpeciesCount + StdDevUniqueSpeciesCount), 
                width = 0.2, color = "black") +  # Error bars
  labs(x = "", y = "Avg Species Richness") +  
  theme_minimal() 
species_richness_by_bayside_elas_plot


```

save
```{r}
ggsave(plot = species_richness_by_habitat_mifish_plot, filename = "figures-expedition/species_richness_by_habitat_mifish_plot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = species_richness_by_bayside_mifish_plot, filename = "figures-expedition/species_richness_by_bayside_mifish_plot.jpg", width = 2.5, height = 4, units = "in")

ggsave(plot = species_richness_by_habitat_elas_plot, filename = "figures-expedition/species_richness_by_habitat_elas_plot.jpg", width = 4, height = 4, units = "in")

ggsave(plot = species_richness_by_bayside_elas_plot, filename = "figures-expedition/species_richness_by_bayside_elas_plot.jpg", width = 2.5, height = 4, units = "in")

```

# Multivariate Analysis

## PCA
PCA is essentially a type of PCoA using the Euclidean distance matrix as input. When combined with a log-ratio transformation of the count table, this is deemed appropriate for compositional datasets. It is also recommended as a first step in exploratory analyses of sequencinging datasets.

First do a CLR, centered log ratio transformation of the absolute abundance data (after filtering), as suggested by [Gloor et al. 2017](https://www.frontiersin.org/journals/microbiology/articles/10.3389/fmicb.2017.02224/full) for compositional data
```{r}
exp_mifish_glommed_clr <- data.frame(compositions::clr(otu_table(ps2024_exp_mifish)))
exp_elas_glommed_clr <- data.frame(compositions::clr(otu_table(ps2024_exp_elas)))

```

Generate PCA
```{r}
lograt_pca_mifish_exp <- prcomp(t(exp_mifish_glommed_clr)) 
lograt_pca_elas_exp <- prcomp(t(exp_elas_glommed_clr))

```

Screeplots
```{r}
lograt_variances_mifish_exp <- as.data.frame(lograt_pca_mifish_exp$sdev^2/sum(lograt_pca_mifish_exp$sdev^2)) %>% #Extract axes
  # Format to plot
  select(PercVar = 'lograt_pca_mifish_exp$sdev^2/sum(lograt_pca_mifish_exp$sdev^2)') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
head(lograt_variances_mifish_exp)

ggplot(lograt_variances_mifish_exp, aes(x = as.numeric(PCaxis), y = PercVar)) + 
  geom_bar(stat = "identity", fill = "grey", color = "black") +
  theme_minimal() +
  theme(axis.title = element_text(color = "black", face = "bold", size = 10),
        axis.text.y = element_text(color = "black", face = "bold"),
        axis.text.x = element_blank()) +
  labs(x = "PC axis", y = "% Variance", title = "Log-Ratio PCA Screeplot, CLR Tranformation- MiFish/ Expedition")


lograt_variances_elas_exp <- as.data.frame(lograt_pca_elas_exp$sdev^2/sum(lograt_pca_elas_exp$sdev^2)) %>% #Extract axes
  # Format to plot
  select(PercVar = 'lograt_pca_elas_exp$sdev^2/sum(lograt_pca_elas_exp$sdev^2)') %>% 
  rownames_to_column(var = "PCaxis") %>% 
  data.frame
head(lograt_variances_elas_exp)

ggplot(lograt_variances_elas_exp, aes(x = as.numeric(PCaxis), y = PercVar)) + 
  geom_bar(stat = "identity", fill = "grey", color = "black") +
  theme_minimal() +
  theme(axis.title = element_text(color = "black", face = "bold", size = 10),
        axis.text.y = element_text(color = "black", face = "bold"),
        axis.text.x = element_blank()) +
  labs(x = "PC axis", y = "% Variance", title = "Log-Ratio PCA Screeplot, CLR Tranformation- Elas02/ Expedition")


```
First two axes are quite strong for both libraries


Extract variances from the clr pca
(Note [this issue](https://github.com/joey711/phyloseq/issues/1322) with converting rownames to column from a phyloseq object.... it was giving me a swuare 28x28 table rather than the full table. Needed to modify)
```{r}
pca_lograt_mifish_exp_df <- data.frame(lograt_pca_mifish_exp$x) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata into the pca data table
pca_lograt_mifish_exp_df <- sample_data(ps2024_exp_mifish_glommed) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
right_join(pca_lograt_mifish_exp_df, by = "SampleID")

pca_lograt_mifish_exp_df


pca_lograt_elas_exp_df <- data.frame(lograt_pca_elas_exp$x) %>% 
  rownames_to_column(var = "SampleID")

# Merge metadata into the pca data table
pca_lograt_elas_exp_df <- sample_data(ps2024_exp_elas_glommed) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% 
right_join(pca_lograt_elas_exp_df, by = "SampleID")

pca_lograt_elas_exp_df
```

Select eigenvalues from dataframe, round to 4 places and multiply by 100 for plotting. These will be the axes for the 3-D plot
```{r}
eigenvalues_pca_mifish_exp <-round(lograt_variances_mifish_exp[,2], digits = 4)*100
eigenvalues_pca_elas_exp <-round(lograt_variances_elas_exp[,2], digits = 4)*100
```

### Plot PCA
Use two axes since first two PCs explain large proportion of variance
```{r}
pca_mifish_exp_bayside_plot <- ggplot(pca_lograt_mifish_exp_df, aes(x = PC1, y = PC2, color = Bayside)) +
  geom_point(size = 3, alpha = 0.8) + 
  labs(x = paste0('PC1 ',eigenvalues_pca_mifish_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_mifish_exp[2],'%'), color = "Bayside") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
  scale_color_manual(values = c("coral","cornflowerblue"))
pca_mifish_exp_bayside_plot

pca_elas_exp_bayside_plot <- ggplot(pca_lograt_elas_exp_df, aes(x = PC1, y = PC2, color = Bayside)) +
  geom_point(size = 3, alpha = 0.8) + 
  labs(x = paste0('PC1 ',eigenvalues_pca_elas_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_elas_exp[2],'%'), color = "Bayside") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
  scale_color_manual(values = c("coral","cornflowerblue"))
pca_elas_exp_bayside_plot


pca_mifish_exp_habitat_plot <- ggplot(pca_lograt_mifish_exp_df, aes(x = PC1, y = PC2, fill = Habitat)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(x = paste0('PC1 ',eigenvalues_pca_mifish_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_mifish_exp[2],'%'), color = "Habitat") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey"))
pca_mifish_exp_habitat_plot


pca_elas_exp_habitat_plot <- ggplot(pca_lograt_elas_exp_df, aes(x = PC1, y = PC2, fill = Habitat)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(x = paste0('PC1 ',eigenvalues_pca_elas_exp[1],'%'), y = paste0('PC2 ',eigenvalues_pca_elas_exp[2],'%'), color = "Habitat") +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14) #
  ) +
  scale_fill_manual(values = c("azure2", "chartreuse4", "cadetblue1"))
pca_elas_exp_habitat_plot



pca_mifish_exp_nightday_plot <- ggplot(pca_lograt_mifish_exp_df, aes(x = PC1, y = PC2, fill = Night__Day)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(
    x = paste0('PC1 ', eigenvalues_pca_mifish_exp[1], '%'), 
    y = paste0('PC2 ', eigenvalues_pca_mifish_exp[2], '%'), 
    fill = ""
  ) +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
pca_mifish_exp_nightday_plot

pca_elas_exp_nightday_plot <- ggplot(pca_lograt_elas_exp_df, aes(x = PC1, y = PC2, fill = Night__Day)) +
  geom_point(size = 3, shape = 21, stroke = 0.5, color = "black") + 
  labs(
    x = paste0('PC1 ', eigenvalues_pca_elas_exp[1], '%'), 
    y = paste0('PC2 ', eigenvalues_pca_elas_exp[2], '%'), 
    fill = ""
  ) +  
  theme_minimal() + 
  theme(
    axis.title = element_text(size = 14),  
    axis.text = element_text(size = 12),   
    legend.text = element_text(size = 12), 
    legend.title = element_text(size = 14)
  ) +
  scale_fill_manual(values = c("cornsilk", "antiquewhite4"))
pca_elas_exp_nightday_plot

```

### Env Fit
#### MiFish
```{r}
# sort metadata in same order as the pca matrix
metadata_mifish_exp <- sample_data(ps2024_exp_mifish) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% arrange(factor(SampleID, levels = rownames(lograt_pca_mifish_exp$x)))

# Fix time and date so it can be used as input

metadata_mifish_exp$DateTime <- as.POSIXct(force_tz((mdy_hms(paste(metadata_mifish_exp$Date, metadata_mifish_exp$Start_Time__ET_))), tzone = "America/New_York"))

# Convert Start Time (from RocSI filtration time) to continuous variable rather than factor. This will be a "time-of-day" variable
metadata_mifish_exp$Time_numeric <- as.numeric(as_hms(as.character(metadata_mifish_exp$Start_Time__ET_)))

# Convert Datecode to numeric
metadata_mifish_exp$Datecode <- as.numeric(metadata_mifish_exp$Datecode)

# Remove some columns of metadate- only retain the env data that I am testing. Leave out the repetitive ID columns, etc.
metadata_mifish_exp <- select(metadata_mifish_exp, -"Name.Deploy_Cartr_Library", -Deployment, -Cartridge_ID, -sites, -replicates, -site_altname, -controls, -Date, -Year, -Month, -Day, -Name.Deploy_Cartr, -Start_Time__ET_, -End_Time__ET_)

# fit environmental factors and save stats output
pca_envfit_mifish_exp <- envfit(lograt_pca_mifish_exp, metadata_mifish_exp, permutations = 10000, na.rm = TRUE)

pca_envfit_mifish_exp
capture.output(pca_envfit_mifish_exp, file = "figures-expedition/pca_envfit_mifish_exp.txt")

```
*SUMMARY- MiFish EnvFir Results that were sig*
Numeric:
                         PC1      PC2     r2   Pr(>r)   
lat                  0.37267  0.92796 0.1441 0.00150 **
Datecode            -0.93766  0.34756 0.0918 0.01900 * 
Avg_Salinity_psu    -0.69632 -0.71773 0.0538 0.09549 . 
Avg_Turbidity_NTU    0.84728 -0.53114 0.0552 0.08269 . 


Categorical variables that are signficant:
               r2  Pr(>r)  
Bayside    0.0413 0.02820 *
Night__Day 0.0326 0.05789 .


#### Elas02
```{r}
# sort metadata in same order as the pca matrix
metadata_elas_exp <- sample_data(ps2024_exp_elas) %>% data.frame() %>% 
  rownames_to_column(var = "SampleID") %>% arrange(factor(SampleID, levels = rownames(lograt_pca_elas_exp$x)))

# Fix time and date so it can be used as input
metadata_elas_exp$DateTime <- as.POSIXct(force_tz((mdy_hms(paste(metadata_elas_exp$Date, metadata_elas_exp$Start_Time__ET_))), tzone = "America/New_York"))

# Convert Start Time (from RocSI filtration time) to continuous variable rather than factor. This will be a "time-of-day" variable
metadata_elas_exp$Time_numeric <- as.numeric(as_hms(as.character(metadata_elas_exp$Start_Time__ET_)))

# Convert Datecode to numeric
metadata_elas_exp$Datecode <- as.numeric(metadata_elas_exp$Datecode)

# Remove some columns of metadata- only retain the env data that I am testing. Leave out the repetitive ID columns, etc.
metadata_elas_exp <- select(metadata_elas_exp, -Deployment, -Cartridge_ID, -sites, -replicates, -controls, -Date, -Year, -Month, -Day, -Name_Deploy_Cartr_Library, -Start_Time__ET_, -End_Time__ET_)

# fit environmental factors and save stats output
pca_envfit_elas_exp <- envfit(lograt_pca_elas_exp, metadata_elas_exp, permutations = 10000, na.rm = TRUE)

pca_envfit_elas_exp
capture.output(pca_envfit_elas_exp, file = "figures-expedition/pca_envfit_elas_exp.txt")

```

*Summary of Envfit to Elas02 library*
Vectors-
                         PC1      PC2     r2    Pr(>r)    
lat                  0.90589 -0.42352 0.3855 9.999e-05 ***
long                 0.94451 -0.32848 0.2780    0.0002 ***

Categorical variables-
                         r2    Pr(>r)    
Bayside              0.2379 9.999e-05 ***
Site_type            0.0830   0.05379 .  


#### Plot EnvFit - Categorical Vars
```{r}
pca_mifish_exp_bayside_plot <- pca_mifish_exp_bayside_plot +
  stat_ellipse(aes(color = factor(Bayside)), level = 0.99)

pca_elas_exp_bayside_plot <- pca_elas_exp_bayside_plot +
  stat_ellipse(aes(color = factor(Bayside)), level = 0.99)

pca_mifish_exp_habitat_plot <- pca_mifish_exp_habitat_plot +
  stat_ellipse(aes(color = factor(Habitat)), level = 0.99) +
  scale_color_manual(values = c("azure2", "chartreuse4", "cadetblue1", "darkgrey")) 


pca_elas_exp_habitat_plot <- pca_elas_exp_habitat_plot +
  stat_ellipse(aes(color = factor(Habitat)), level = 0.99) +
    scale_color_manual(values = c("azure2", "chartreuse4", "cadetblue1")) 


pca_mifish_exp_nightday_plot <- pca_mifish_exp_nightday_plot +
  stat_ellipse(aes(color = factor(Night__Day)), level = 0.99) +
  scale_color_manual(values = c("cornsilk", "antiquewhite4")) +
  guides(color = "none") 



pca_elas_exp_nightday_plot <- pca_elas_exp_nightday_plot +
  stat_ellipse(aes(color = factor(Night__Day)), level = 0.99) +
  scale_color_manual(values = c("cornsilk", "antiquewhite4")) +
  guides(color = "none") 


pca_mifish_exp_bayside_plot
pca_elas_exp_bayside_plot
pca_mifish_exp_habitat_plot
pca_elas_exp_habitat_plot
pca_mifish_exp_nightday_plot
pca_elas_exp_nightday_plot

```

##### Mifish
Add sig vectors from envfit ontop of categ vars
```{r}
pca_mifish_sig_vectors <- as.data.frame(pca_envfit_mifish_exp$vectors$arrows) %>%
  rownames_to_column(var = "EnvVar") %>%
  filter(EnvVar %in% c("lat","Datecode", "Avg_Salinity_psu", "Avg_Turbidity_NTU"))  %>%  
  mutate(EnvVar = ifelse(EnvVar == "Avg_Salinity_psu", "Salinity", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Avg_Turbidity_NTU", "Turbidity", EnvVar)) %>%
  mutate(EnvVar = ifelse(EnvVar == "Datecode", "Date", EnvVar))

# Plot vectors ontop of ellipse plot of bayside
pca_mifish_exp_bayside_date_salinity_plot <- pca_mifish_exp_bayside_plot +
  geom_segment(data = pca_mifish_sig_vectors,
               aes(x = 0, y = 0, xend = PC1, yend = PC2),
               arrow = arrow(type = "closed", length = unit(0.2, "inches")),
               color = "black") +  
  geom_text(data = pca_mifish_sig_vectors,
            aes(x = PC1, y = PC2, label = EnvVar),
            size = 4, color = "black", vjust = -0.5)

pca_mifish_exp_bayside_date_salinity_plot


```
save
```{r}
ggsave(plot = pca_mifish_exp_bayside_date_salinity_plot, filename = "figures-expedition/pca_mifish_exp_bayside_latlong_date_salinity_turbid_plot.jpg", width = 7, height = 5, units = "in")

ggsave(plot = pca_mifish_exp_nightday_plot, filename = "figures-expedition/pca_mifish_exp_nightday_plot.jpg", width = 7, height = 5, units = "in")

```

##### Elas02
Add sig vectors from envfit ontop of categorical vars (both bayside and habitat significant in this case)
```{r}
pca_elas_sig_vectors <- as.data.frame(pca_envfit_elas_exp$vectors$arrows) %>%
  rownames_to_column(var = "EnvVar") %>%
  filter(EnvVar %in% c("lat", "long"))
    
# Plot vectors ontop of ellipse plot or Bayside
pca_elas_exp_bayside_latlong <- pca_elas_exp_bayside_plot +
  geom_segment(data = pca_elas_sig_vectors,
               aes(x = 0, y = 0, xend = PC1, yend = PC2),
               arrow = arrow(type = "closed", length = unit(0.2, "inches")),
               color = "black") +  
  geom_text(data = pca_elas_sig_vectors,
            aes(x = PC1, y = PC2, label = EnvVar),
            size = 4, color = "black", vjust = -0.5)
pca_elas_exp_bayside_latlong




```

save
```{r}
ggsave(plot = pca_elas_exp_bayside_latlong, filename = "figures-expedition/pca_elas_exp_bayside_latlong.jpg", width = 7, height = 5, units = "in")

ggsave(plot = pca_elas_exp_habitat_plot, filename = "figures-expedition/pca_elas_exp_habitat_plot.jpg", width = 7, height = 5, units = "in")

```



# Station time series- CONTINUE HERE

## Station 20
follow composition (by bar?) and richness across all samples. annotate important events (day night, rain event, temp)

## Station 15
same



## CONTINUE EDITING above 
Things to do
- Follow most commonly sampled station (15? 20?)- diversity/ composition change through all 2 weeks, comparing day/night especially
- Day night comparison: present in a way which removes samples that were only sampled in the day
- Present analysis of impact of RNAlater preservation (in 2024 trawl MiFish samples)
- Multivariate anlayses: determine if there is influence of time of day, location, etc. 
- Separate flounder figure?
- Make pie charts for bayside?
- Presnet mifish and elasmos together?




