---
title: "Data Clean up - all eDNA Hand Samples (MiFish, Elas02, mlCO1): 2020-2025"
output: html_notebook
---

Updated 1/6/2025 to include 2025 data and to update min read abundance filtering.


<br/>
<br/>

 

# Load packages

```{r}
# rm(list = ls())
library(tidyverse)
library(phyloseq)
library(vegan)
library(RColorBrewer)
library(microbiome)
library(metagMisc)
library(fantaxtic)
library(ggrepel)
library(paletteer)
library(pals)
```

#  Import and prepare data
## MiFish
### 2020 MiFish data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2020-MiFIsh/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2020-MiFIsh/ASV2Taxonomy/results-revamp-2020-MiFIsh_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2020-MiFIsh/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_mifish_2020 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_mifish_2020 = tax_table(asv_taxonomy_mat)
samples_mifish_2020 = sample_data(sample_metadata)

phylo_mifish_2020 <- phyloseq(ASV_mifish_2020, TAX_mifish_2020, samples_mifish_2020)

```


### 2021 MiFish data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-MiFIsh/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-MiFIsh/ASV2Taxonomy/results-revamp-2021-MiFIsh_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-MiFIsh/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_mifish_2021 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_mifish_2021 = tax_table(asv_taxonomy_mat)
samples_mifish_2021 = sample_data(sample_metadata)

phylo_mifish_2021 <- phyloseq(ASV_mifish_2021, TAX_mifish_2021, samples_mifish_2021)

```

### 2022 MiFish data
Analyzed in two batches. Import a and b, then join into one table-
```{r}
asv_counta <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2022-MiFisha/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
asv_countb <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2022-MiFishb/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)

# ASV numbers are unique to each of the subsets of data, a and b. Marker into ASV name to indicate whether it's from library a or b
asv_counta <- asv_counta %>% mutate(x = paste0(x, "_a"))
asv_countb <- asv_countb %>% mutate(x = paste0(x, "_b"))

asv_count <- full_join(asv_counta, asv_countb)

row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomya <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2022-MiFisha/ASV2Taxonomy/results-revamp-2022-MiFisha_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
asv_taxonomyb <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2022-MiFishb/ASV2Taxonomy/results-revamp-2022-MiFishb_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)

# ASV numbers are unique to each of the subsets of data, a and b. Marker into ASV name to indicate whether it's from library a or b
asv_taxonomya <- asv_taxonomya %>% mutate(ASV = paste0(ASV, "_a"))
asv_taxonomyb <- asv_taxonomyb %>% mutate(ASV = paste0(ASV, "_b"))

asv_taxonomy <- full_join(asv_taxonomya, asv_taxonomyb)

row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadataa <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2022-MiFisha/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
sample_metadatab <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2022-MiFishb/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)

sample_metadata <- full_join(sample_metadataa, sample_metadatab)

row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_mifish_2022 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_mifish_2022 = tax_table(asv_taxonomy_mat)
samples_mifish_2022 = sample_data(sample_metadata)

# set NAs to zeroes
ASV_mifish_2022[is.na(ASV_mifish_2022)] <- 0



phylo_mifish_2022 <- phyloseq(ASV_mifish_2022, TAX_mifish_2022, samples_mifish_2022)

```


### 2023 MiFish data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-MiFIsh/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-MiFIsh/ASV2Taxonomy/results-revamp-2023-MiFIsh_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-MiFIsh/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_mifish_2023 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_mifish_2023 = tax_table(asv_taxonomy_mat)
samples_mifish_2023 = sample_data(sample_metadata)

phylo_mifish_2023 <- phyloseq(ASV_mifish_2023, TAX_mifish_2023, samples_mifish_2023)

```

### 2024 MiFish data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-MiFIsh/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-MiFIsh/ASV2Taxonomy/results-revamp-2024-MiFIsh_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-MiFIsh/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_mifish_2024 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_mifish_2024 = tax_table(asv_taxonomy_mat)
samples_mifish_2024 = sample_data(sample_metadata)

phylo_mifish_2024 <- phyloseq(ASV_mifish_2024, TAX_mifish_2024, samples_mifish_2024)

```

### 2025 MiFish data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-MiFIsh/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-MiFIsh/ASV2Taxonomy/results-revamp-2025-MiFIsh_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-MiFIsh/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_mifish_2025 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_mifish_2025 = tax_table(asv_taxonomy_mat)
samples_mifish_2025 = sample_data(sample_metadata)

phylo_mifish_2025 <- phyloseq(ASV_mifish_2025, TAX_mifish_2025, samples_mifish_2025)

```



## Elas02
### 2021 data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-Elas02/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-Elas02/ASV2Taxonomy/results-revamp-2021-Elas02_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-Elas02/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_elas_2021 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_elas_2021 = tax_table(asv_taxonomy_mat)
samples_elas_2021 = sample_data(sample_metadata)

phylo_elas_2021 <- phyloseq(ASV_elas_2021, TAX_elas_2021, samples_elas_2021)

```

### 2023 data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-Elas02/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-Elas02/ASV2Taxonomy/results-revamp-2023-Elas02_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-Elas02/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_elas_2023 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_elas_2023 = tax_table(asv_taxonomy_mat)
samples_elas_2023 = sample_data(sample_metadata)

phylo_elas_2023 <- phyloseq(ASV_elas_2023, TAX_elas_2023, samples_elas_2023)

```


### 2024 data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-Elas02/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-Elas02/ASV2Taxonomy/results-revamp-2024-Elas02_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-Elas02/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_elas_2024 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_elas_2024 = tax_table(asv_taxonomy_mat)
samples_elas_2024 = sample_data(sample_metadata)

phylo_elas_2024 <- phyloseq(ASV_elas_2024, TAX_elas_2024, samples_elas_2024)

```

### 2025 data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-Elas02/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-Elas02/ASV2Taxonomy/results-revamp-2025-Elas02_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-Elas02/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_elas_2025 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_elas_2025 = tax_table(asv_taxonomy_mat)
samples_elas_2025 = sample_data(sample_metadata)

phylo_elas_2025 <- phyloseq(ASV_elas_2025, TAX_elas_2025, samples_elas_2025)

```

## CO1
### 2024 data
Need to combine the 3 CO1 runs: a, b, c
```{r}
asv_counta <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1a/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
asv_countb <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1b/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
asv_countc <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1c/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)

# ASV numbers are unique to each of the subsets of data and overlap across datasets. Put a marker in ASV name to indicate whether it's from library a, b, or c
asv_counta <- asv_counta %>% mutate(x = paste0(x, "_a"))
asv_countb <- asv_countb %>% mutate(x = paste0(x, "_b"))
asv_countc <- asv_countc %>% mutate(x = paste0(x, "_c"))

asv_count <- full_join(asv_counta, asv_countb) %>%
  full_join(asv_countc)

row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)


asv_taxonomya <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1a/ASV2Taxonomy/results-revamp-2024-CO1a_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
asv_taxonomyb <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1b/ASV2Taxonomy/results-revamp-2024-CO1b_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
asv_taxonomyc <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1c/ASV2Taxonomy/results-revamp-2024-CO1c_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)


# ASV numbers are unique to each of the subsets of data, a and b. Marker into ASV name to indicate whether it's from library a or b
asv_taxonomya <- asv_taxonomya %>% mutate(ASV = paste0(ASV, "_a"))
asv_taxonomyb <- asv_taxonomyb %>% mutate(ASV = paste0(ASV, "_b"))
asv_taxonomyc <- asv_taxonomyc %>% mutate(ASV = paste0(ASV, "_c"))

asv_taxonomy <- full_join(asv_taxonomya, asv_taxonomyb) %>%
  full_join(asv_taxonomyc)

row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadataa <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1a/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
sample_metadatab <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1b/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
sample_metadatac <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2024-CO1c/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)

sample_metadata_co1 <- full_join(sample_metadataa, sample_metadatab) %>%
  full_join(sample_metadatac)

row.names(sample_metadata_co1) <- sample_metadata_co1$Sample
sample_metadata_co1 <- sample_metadata_co1 %>% select(-Sample)
ASV_co1_2024 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_co1_2024 = tax_table(asv_taxonomy_mat)
samples_co1_2024 = sample_data(sample_metadata_co1)


# set NAs to zeroes
ASV_co1_2024[is.na(ASV_co1_2024)] <- 0

phylo_co1_2024 <- phyloseq(ASV_co1_2024, TAX_co1_2024, samples_co1_2024)

```

### 2025 data
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-CO1/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-CO1/ASV2Taxonomy/results-revamp-2025-CO1_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2025-CO1/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_co1_2025 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_co1_2025 = tax_table(asv_taxonomy_mat)
samples_co1_2025 = sample_data(sample_metadata)

phylo_co1_2025 <- phyloseq(ASV_co1_2025, TAX_co1_2025, samples_co1_2025)

```


# Filtering & QC
## How many taxa before filtering
First change names of phyloseq objects. Retain `phylo_XX_XX`, which is the unfiltered object. Make new objects, `ps_XX_XX`, which will be what remains after the following filtering steps:
```{r}
ps_mifish_2020 <- phylo_mifish_2020
ps_mifish_2021 <- phylo_mifish_2021
ps_mifish_2022 <- phylo_mifish_2022
ps_mifish_2023 <- phylo_mifish_2023
ps_mifish_2024 <- phylo_mifish_2024
ps_mifish_2025 <- phylo_mifish_2025


ps_elas_2021 <- phylo_elas_2021
ps_elas_2023 <- phylo_elas_2023
ps_elas_2024 <- phylo_elas_2024
ps_elas_2025 <- phylo_elas_2025


ps_co1_2024 <- phylo_co1_2024
ps_co1_2025 <- phylo_co1_2025

# How many taxa before filtering?
nrow(tax_table(phylo_mifish_2020)) 
nrow(tax_table(phylo_mifish_2021)) 
nrow(tax_table(phylo_mifish_2022)) # this number is misleading because I had to do the dada2 step in two subsets, then append the tables, so some of these ASVs very likely should overlap
nrow(tax_table(phylo_mifish_2023)) 
nrow(tax_table(phylo_mifish_2024)) 
nrow(tax_table(phylo_mifish_2025)) 


nrow(tax_table(phylo_elas_2021)) 
nrow(tax_table(phylo_elas_2023)) 
nrow(tax_table(phylo_elas_2024)) 
nrow(tax_table(phylo_elas_2025)) 

nrow(tax_table(phylo_co1_2024)) # same comment for CO1- it was done in 3 batches 
nrow(tax_table(phylo_co1_2025)) 
```



## Replace NA in taxa list with next highest annotated taxon
```{r}

tax_table(ps_mifish_2020) <- tax_table(fantaxtic::name_na_taxa(ps_mifish_2020, include_rank = T))
tax_table(ps_mifish_2021) <- tax_table(fantaxtic::name_na_taxa(ps_mifish_2021, include_rank = T))
tax_table(ps_mifish_2022) <- tax_table(fantaxtic::name_na_taxa(ps_mifish_2022, include_rank = T))
tax_table(ps_mifish_2023) <- tax_table(fantaxtic::name_na_taxa(ps_mifish_2023, include_rank = T))
tax_table(ps_mifish_2024) <- tax_table(fantaxtic::name_na_taxa(ps_mifish_2024, include_rank = T))
tax_table(ps_mifish_2025) <- tax_table(fantaxtic::name_na_taxa(ps_mifish_2025, include_rank = T))


tax_table(ps_elas_2021) <- tax_table(fantaxtic::name_na_taxa(ps_elas_2021, include_rank = T))
tax_table(ps_elas_2023) <- tax_table(fantaxtic::name_na_taxa(ps_elas_2023, include_rank = T))
tax_table(ps_elas_2024) <- tax_table(fantaxtic::name_na_taxa(ps_elas_2024, include_rank = T))
tax_table(ps_elas_2025) <- tax_table(fantaxtic::name_na_taxa(ps_elas_2025, include_rank = T))

tax_table(ps_co1_2024) <- tax_table(fantaxtic::name_na_taxa(ps_co1_2024, include_rank = T))
tax_table(ps_co1_2025) <- tax_table(fantaxtic::name_na_taxa(ps_co1_2025, include_rank = T))


```


## Remove prokaryotes
check
```{r}
unique(tax_table(ps_mifish_2020)[, "Kingdom"])
unique(tax_table(ps_mifish_2021)[, "Kingdom"])
unique(tax_table(ps_mifish_2022)[, "Kingdom"])
unique(tax_table(ps_mifish_2023)[, "Kingdom"])
unique(tax_table(ps_mifish_2024)[, "Kingdom"])
unique(tax_table(ps_mifish_2025)[, "Kingdom"])


unique(tax_table(ps_elas_2021)[, "Kingdom"])
unique(tax_table(ps_elas_2023)[, "Kingdom"])
unique(tax_table(ps_elas_2024)[, "Kingdom"])
unique(tax_table(ps_elas_2025)[, "Kingdom"])


unique(tax_table(ps_co1_2024)[, "Kingdom"])
unique(tax_table(ps_co1_2025)[, "Kingdom"])

```

remove
```{r}
#ps_mifish_2020 <- subset_taxa(ps_mifish_2020, !Kingdom == 'Bacteria')
ps_mifish_2021 <- subset_taxa(ps_mifish_2021, !Kingdom == 'Bacteria')
ps_mifish_2022 <- subset_taxa(ps_mifish_2022, !(Kingdom %in% c("Bacteria", "Archaea"))) # 2022 also had archaea
ps_mifish_2023 <- subset_taxa(ps_mifish_2023, !Kingdom == 'Bacteria')
ps_mifish_2024 <- subset_taxa(ps_mifish_2024, !Kingdom == 'Bacteria')
ps_mifish_2025 <- subset_taxa(ps_mifish_2025, !Kingdom == 'Bacteria')



# no prokaryotes are in elas nor co1 objects

```


check
```{r}
unique(tax_table(ps_mifish_2020)[, "Kingdom"])
unique(tax_table(ps_mifish_2021)[, "Kingdom"])
unique(tax_table(ps_mifish_2022)[, "Kingdom"])
unique(tax_table(ps_mifish_2023)[, "Kingdom"])
unique(tax_table(ps_mifish_2024)[, "Kingdom"])
unique(tax_table(ps_mifish_2025)[, "Kingdom"])

unique(tax_table(ps_elas_2021)[, "Kingdom"])
unique(tax_table(ps_elas_2023)[, "Kingdom"])
unique(tax_table(ps_elas_2024)[, "Kingdom"])
unique(tax_table(ps_elas_2025)[, "Kingdom"])

unique(tax_table(ps_co1_2024)[, "Kingdom"])
unique(tax_table(ps_co1_2025)[, "Kingdom"])
```


## Remove non-marine mammals
check unique classes
```{r}
unique(tax_table(ps_mifish_2020)[, "Class"])
unique(tax_table(ps_mifish_2021)[, "Class"])
unique(tax_table(ps_mifish_2022)[, "Class"])
unique(tax_table(ps_mifish_2023)[, "Class"])
unique(tax_table(ps_mifish_2024)[, "Class"])
unique(tax_table(ps_mifish_2025)[, "Class"])


unique(tax_table(ps_elas_2021)[, "Class"])
unique(tax_table(ps_elas_2023)[, "Class"])
unique(tax_table(ps_elas_2024)[, "Class"])
unique(tax_table(ps_elas_2025)[, "Class"])


unique(tax_table(ps_co1_2024)[, "Class"])
unique(tax_table(ps_co1_2025)[, "Class"])

```

all ps objects have Mammalia


Check whether these are contamination (eg. human, dog, etc) or something we want to keep (whales, dolphins, etc)
```{r}
unique(tax_table(ps_mifish_2020)[tax_table(ps_mifish_2020)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_mifish_2021)[tax_table(ps_mifish_2021)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_mifish_2022)[tax_table(ps_mifish_2022)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_mifish_2023)[tax_table(ps_mifish_2023)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_mifish_2024)[tax_table(ps_mifish_2024)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_mifish_2025)[tax_table(ps_mifish_2025)[, "Class"] == "Mammalia", "Species"])


unique(tax_table(ps_elas_2021)[tax_table(ps_elas_2021)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_elas_2023)[tax_table(ps_elas_2023)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_elas_2024)[tax_table(ps_elas_2024)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_elas_2025)[tax_table(ps_elas_2025)[, "Class"] == "Mammalia", "Species"])

unique(tax_table(ps_co1_2024)[tax_table(ps_co1_2024)[, "Class"] == "Mammalia", "Species"])
unique(tax_table(ps_co1_2025)[tax_table(ps_co1_2025)[, "Class"] == "Mammalia", "Species"])

```

Summary:
MiFish:
2020 - remove all mammals
2021 - remove all mammals (although Puma yagouaroundi is very interesting...)
2022 - retain Tursiops truncatus and Unknown Delphinidae (Family), remove rest of mammals
2023 - remove all mammals
2024 - remove all mammals
2025 - remove all mammals except guinea pig, positive control

Elas02:
remove all mammals from all datasets

CO1: 
2024: Remove all for now. There are mostly contamination (primates, generic mammals, cat, dog/wolf) but there are some interesting taxa here (opposums and bats) but they are not defined to species and not target of study.
2025: Lots of contaminants (human, cat) as well as bats, shrews, which are probably real but remove. Keep dolphins (Tursiops truncatus and Unknown Didelphimorphia (Order)) and guinea pig, positive control


```{r}
ps_mifish_2020 <- subset_taxa(ps_mifish_2020, !Class == 'Mammalia')
ps_mifish_2021 <- subset_taxa(ps_mifish_2021, !Class == 'Mammalia')
ps_mifish_2022 <- subset_taxa(ps_mifish_2022, !Species == 'Homo sapiens' & !Species == 'Felis catus' & !Species == 'Bos taurus' & !Species == 'Canis lupus' & !Species == 'Sus scrofa' & !Species == 'Unknown Bovidae (Family)' & !Species == 'Unknown Bos (Genus)' & !Species == 'Unknown Hominidae (Family)' & !Species == 'Unknown Felidae (Family)' & !Species == 'Unknown Primates (Order)')
ps_mifish_2023 <- subset_taxa(ps_mifish_2023, !Class == 'Mammalia')
ps_mifish_2024 <- subset_taxa(ps_mifish_2024, !Class == 'Mammalia')
ps_mifish_2025 <- subset_taxa(ps_mifish_2025, !Species == 'Homo sapiens' & !Species == 'Bos taurus' & !Species == 'Mus musculus' & !Species == 'Canis lupus' & !Species == 'Felis catus' & !Species == 'Unknown Bovidae (Family)' & !Species == 'Unknown Hominidae (Family)' & !Species == 'Sus scrofa' & !Species == 'Unknown Primates (Order)' & !Species == 'Unknown Artiodactyla (Order)' & !Species == 'Unknown Muridae (Family)' & !Species == 'Unknown Carnivora (Order)' & !Species == 'Unknown Canidae (Family)' & !Species == 'Unknown Homo (Genus)' & !Species == 'Unknown Macroscelidea (Order)' & !Species == 'Unknown Mammalia (Class)' & !Species == 'Unknown Rodentia (Order)' & !Species == 'Unknown Caviidae (Family)')

ps_elas_2021 <- subset_taxa(ps_elas_2021, !Class == 'Mammalia')
ps_elas_2023 <- subset_taxa(ps_elas_2023, !Class == 'Mammalia')
ps_elas_2024 <- subset_taxa(ps_elas_2024, !Class == 'Mammalia')
ps_elas_2025 <- subset_taxa(ps_elas_2025, !Class == 'Mammalia')


ps_co1_2024 <- subset_taxa(ps_co1_2024, !Class == 'Mammalia')
ps_co1_2025 <- subset_taxa(ps_co1_2025, !Species == 'Felis catus' & !Species == 'Canis lupaster' & !Species == 'Unknown Mammalia (Class)' & !Species == 'Unknown Primates (Order)' & !Species == 'Unknown Chiroptera (Order)' & !Species == 'Unknown Cavia (Genus)' & !Species == 'Unknown Caviidae (Family)' & !Species == 'Unknown Canis (Genus)' & !Species == 'Unknown Carnivora (Order)' & !Species == 'Unknown Rodentia (Order)' & !Species == 'Unknown Capra (Genus)' & !Species == 'Unknown Sciurus (Genus)' & !Species == 'Unknown Canidae (Family)' & !Species == 'Unknown Lagomorpha (Order)' & !Species == 'Unknown Felidae (Family)' & !Species == 'Unknown Eulipotyphla (Order)' & !Species == 'Unknown Didelphimorphia (Order)')


```

## Remove the non target taxa: Birds, Algae, Insects, etc


Check which Aves were detected
```{r}

#unique(tax_table(ps_mifish_2020)[tax_table(ps_mifish_2020)[, "Class"] == "Aves", "Species"]) # none detected in 2020
unique(tax_table(ps_mifish_2021)[tax_table(ps_mifish_2021)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_mifish_2022)[tax_table(ps_mifish_2022)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_mifish_2023)[tax_table(ps_mifish_2023)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_mifish_2024)[tax_table(ps_mifish_2024)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_mifish_2025)[tax_table(ps_mifish_2025)[, "Class"] == "Aves", "Species"])


unique(tax_table(ps_elas_2021)[tax_table(ps_elas_2021)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_elas_2023)[tax_table(ps_elas_2023)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_elas_2024)[tax_table(ps_elas_2024)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_elas_2025)[tax_table(ps_elas_2025)[, "Class"] == "Aves", "Species"])

unique(tax_table(ps_co1_2024)[tax_table(ps_co1_2024)[, "Class"] == "Aves", "Species"])
unique(tax_table(ps_co1_2025)[tax_table(ps_co1_2025)[, "Class"] == "Aves", "Species"])

```

Turkey (Meleagris gallopavo), broad landfowl taxa, cormorant (Phalacrocorax auritus), gulls (Gallus gallus), Candadian geese, common parakeet (Melopsittacus undulatus), common myna (Acridotheres tristis - [an invasive in southern US](https://www.audubon.org/field-guide/bird/common-myna) but can't find much about it near here), song sparrow (Melospiza melodia)... other parakeets from Asia, African greys

Some of these are very interesting but not the target of the trawl, so remove for the purpose of trawl comparison.


Zygnemophyceae are green algae, can remove from MiFish

```{r}
ps_mifish_2020 <- subset_taxa(ps_mifish_2020, !Class == 'Aves')
ps_mifish_2021 <- subset_taxa(ps_mifish_2021, !Class == 'Aves')
ps_mifish_2022 <- subset_taxa(ps_mifish_2022, !Class == 'Aves' & !Class == 'Zygnemophyceae')
ps_mifish_2023 <- subset_taxa(ps_mifish_2023, !Class == 'Aves')
ps_mifish_2024 <- subset_taxa(ps_mifish_2024, !Class == 'Aves')
ps_mifish_2025 <- subset_taxa(ps_mifish_2025, !Class == 'Aves' & !Class == 'Zygnemophyceae')


ps_elas_2021 <- subset_taxa(ps_elas_2021, !Class == 'Aves') # & !Class == 'Actinopteri')
ps_elas_2023 <- subset_taxa(ps_elas_2023, !Class == 'Aves') # & !Class == 'Actinopteri')
ps_elas_2024 <- subset_taxa(ps_elas_2024, !Class == 'Aves') # & !Class == 'Actinopteri')
ps_elas_2025 <- subset_taxa(ps_elas_2025, !Class == 'Aves') # & !Class == 'Actinopteri')


ps_co1_2024 <- subset_taxa(ps_co1_2024, !Class == 'Aves')
ps_co1_2025 <- subset_taxa(ps_co1_2025, !Class == 'Aves')

```

check
```{r}
unique(tax_table(ps_mifish_2020)[, "Class"])
unique(tax_table(ps_mifish_2021)[, "Class"])
unique(tax_table(ps_mifish_2022)[, "Class"])
unique(tax_table(ps_mifish_2023)[, "Class"])
unique(tax_table(ps_mifish_2024)[, "Class"])

unique(tax_table(ps_elas_2021)[, "Class"])
unique(tax_table(ps_elas_2023)[, "Class"])
unique(tax_table(ps_elas_2024)[, "Class"])

unique(tax_table(ps_co1_2024)[, "Class"])
```


Further filter CO1 to just taxa of interest: those that can be compared to trawl 
Bc there are insects, sponges, phytoplankton etc. that are targets of the primer but not targets of the trawl 

```{r}
ps_co1_2024 <- subset_taxa(ps_co1_2024, Phylum %in% c("Arthropoda","Mollusca","Chordata"))
ps_co1_2025 <- subset_taxa(ps_co1_2025, Phylum %in% c("Arthropoda","Mollusca","Chordata"))


unique(tax_table(ps_co1_2024)[, "Phylum"])
unique(tax_table(ps_co1_2025)[, "Phylum"])

```

But also want to focus on marine animals, so remove insect and other arthopods. First check what classes are there
```{r}
unique(tax_table(subset_taxa(ps_co1_2024, Phylum %in% c("Arthropoda")))[,"Class"])
unique(tax_table(subset_taxa(ps_co1_2025, Phylum %in% c("Arthropoda")))[,"Class"])


```

Keep only Malacostraca, Merostomata

Hexanauplia, Thecostraca are marine but small copepods, barnacles- not a target of the trawl
Ostracoda also marine - seed "shrimp" - but even that largest are v small and planktonic

```{r}
ps_co1_2024 <- subset_taxa(ps_co1_2024, !Class == 'Hexanauplia' & !Class == 'Insecta' & !Class == 'Arachnida' & !Class == 'Unknown Arthropoda (Phylum)' & !Class == 'Collembola' & !Class == 'Thecostraca' & !Class == 'Pycnogonida' & !Class == 'Branchiopoda' & !Class == 'Diplopoda' & !Class == 'Symphyla' & !Class == 'Chilopoda' & !Class == 'Protura' & !Class == 'Ostracoda' )

ps_co1_2025 <- subset_taxa(ps_co1_2025, !Class == 'Hexanauplia' & !Class == 'Insecta' & !Class == 'Arachnida' & !Class == 'Unknown Arthropoda (Phylum)' & !Class == 'Collembola' & !Class == 'Thecostraca' & !Class == 'Pycnogonida' & !Class == 'Branchiopoda' & !Class == 'Diplopoda' & !Class == 'Symphyla' & !Class == 'Chilopoda' & !Class == 'Protura' & !Class == 'Ostracoda' & !Class == 'Diplura__c')
```

check
```{r}
unique(tax_table(ps_co1_2024)[, "Class"])
unique(tax_table(ps_co1_2025)[, "Class"])
```

Remove Polyplacophora (chitons), Ascidiacea (sea squirts), Unknown Mollusca (Phylum), Thaliacea (tunicates), Unknown Chordata (Phylum) - wouldn't be targeted by the trawl and also to clean up the high level unknowns a bit
```{r}
ps_co1_2024 <- subset_taxa(ps_co1_2024, !Class == 'Polyplacophora' & !Class == 'Ascidiacea' & !Class == 'Unknown Mollusca (Phylum)')

ps_co1_2025 <- subset_taxa(ps_co1_2025, !Class == 'Polyplacophora' & !Class == 'Ascidiacea' & !Class == 'Unknown Mollusca (Phylum)' & !Class == 'Thaliacea' & !Class == 'Unknown Chordata (Phylum)')

unique(tax_table(ps_co1_2024)[, "Class"])
unique(tax_table(ps_co1_2025)[, "Class"])

```
Check Gastropods
```{r}
unique(tax_table(subset_taxa(ps_co1_2024, Class == 'Gastropoda'))[,'Order'])
unique(tax_table(subset_taxa(ps_co1_2025, Class == 'Gastropoda'))[,'Order'])

```
Remove all. There are no swimming animals here that would be target of the trawl. But come back to these at some points because they are interesting!

Stylommatophora (land snails/ slugs), Unknown Gastropoda (Class), Cephalaspidea (sea snails and slugs- non swimming), Ancylidae (limpets), Planorbidae (freshwater snail), Neogastropoda (Aquatic snails), Littorinimorpha (large snails), Lepetellida (small sea snails), Protancylidae (limpets), Valvatidae (valve snails), Architaenioglossa (snails with gills), Nudibranchia, Hermaeidae (sea slugs), Cerithiidae, Acteonidae (bubble snail), Limapontiidae (sea slugs), Runcinacea (sea slugs), Pyramidellidae (parasitic snails), Potamididae (horn snail), Cycloneritida (land and freshwater snail), Acroloxidae (river limpets), Ringiculidae (deep water sea snail), Pteropoda, Aplysiida (sea hares)


```{r}
ps_co1_2024 <- subset_taxa(ps_co1_2024, !Class == 'Gastropoda')
unique(tax_table(ps_co1_2024)[, "Class"])

ps_co1_2025 <- subset_taxa(ps_co1_2025, !Class == 'Gastropoda')
unique(tax_table(ps_co1_2025)[, "Class"])
```

within the Malacostraca are crabs, shrimp, etc but also isopods and amphipods which are not target of trawl:
```{r}
unique(tax_table(subset_taxa(ps_co1_2024, Class == 'Malacostraca'))[,'Order'])
unique(tax_table(subset_taxa(ps_co1_2025, Class == 'Malacostraca'))[,'Order'])

```
Remove Unknown Malacostraca, Amphipoda, Isopoda, Bathynellacea (worms)

```{r}
ps_co1_2024 <- subset_taxa(ps_co1_2024, !Order == 'Unknown Malacostraca (Class)' & !Order == 'Amphipoda' & !Order == 'Isopoda')
ps_co1_2025 <- subset_taxa(ps_co1_2025, !Order == 'Unknown Malacostraca (Class)' & !Order == 'Amphipoda' & !Order == 'Isopoda' & !Order == 'Bathynellacea')

unique(tax_table(ps_co1_2024)[, "Order"])
unique(tax_table(ps_co1_2025)[, "Order"])
```




## Add in Common Names
Working with common names from lab group's trawl database,
https://docs.google.com/spreadsheets/d/172eRb2JrIJvZHab2wK-DBpMfa1fjE4lp2AownH9Y6AY/edit?gid=0#gid=0
and my own database of common names (matched to scientific names and with speficified color keys)

Import
```{r}
commonnames <- read_csv(file = "../eDNA-databases/commonnames.csv")
trawl_taxa_db <- read_csv(file = "../eDNA-databases/ShiRP Database Species Names 20251014.csv")
```

Are all my common names in eDNA common name database in the trawl database?
```{r}
# Which names are in trawl_taxa_db but NOT in commonnames?
missing_in_common  <- setdiff(unique(trawl_taxa_db$Names), unique(commonnames$Taxon))

# Which taxa are in commonnames but NOT in trawl_taxa_db?
missing_in_trawl   <- setdiff(unique(commonnames$Taxon), unique(trawl_taxa_db$Names))

missing_in_common
# missing_in_trawl
```


Append common names
```{r}
# Extract, update, tax_table- MiFish 2020
taxonomy_mifish_2020 <- as.data.frame(tax_table(ps_mifish_2020)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_mifish_2020) <- taxonomy_mifish_2020$ASV
taxonomy_mifish_2020 <- taxonomy_mifish_2020 %>% select(-ASV)
# Update the phyloseq object
tax_table(ps_mifish_2020) <- as.matrix(taxonomy_mifish_2020)

# 2021
taxonomy_mifish_2021 <- as.data.frame(tax_table(ps_mifish_2021)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_mifish_2021) <- taxonomy_mifish_2021$ASV
taxonomy_mifish_2021 <- taxonomy_mifish_2021 %>% select(-ASV)
tax_table(ps_mifish_2021) <- as.matrix(taxonomy_mifish_2021)

# 2022
taxonomy_mifish_2022 <- as.data.frame(tax_table(ps_mifish_2022)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_mifish_2022) <- taxonomy_mifish_2022$ASV
taxonomy_mifish_2022 <- taxonomy_mifish_2022 %>% select(-ASV)
tax_table(ps_mifish_2022) <- as.matrix(taxonomy_mifish_2022)

# 2023
taxonomy_mifish_2023 <- as.data.frame(tax_table(ps_mifish_2023)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_mifish_2023) <- taxonomy_mifish_2023$ASV
taxonomy_mifish_2023 <- taxonomy_mifish_2023 %>% select(-ASV)
tax_table(ps_mifish_2023) <- as.matrix(taxonomy_mifish_2023)

# 2024
taxonomy_mifish_2024 <- as.data.frame(tax_table(ps_mifish_2024)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_mifish_2024) <- taxonomy_mifish_2024$ASV
taxonomy_mifish_2024 <- taxonomy_mifish_2024 %>% select(-ASV)
tax_table(ps_mifish_2024) <- as.matrix(taxonomy_mifish_2024)

# 2025
taxonomy_mifish_2025 <- as.data.frame(tax_table(ps_mifish_2025)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_mifish_2025) <- taxonomy_mifish_2025$ASV
taxonomy_mifish_2025 <- taxonomy_mifish_2025 %>% select(-ASV)
tax_table(ps_mifish_2025) <- as.matrix(taxonomy_mifish_2025)



# Elas
#2021
taxonomy_elas_2021 <- as.data.frame(tax_table(ps_elas_2021)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_elas_2021) <- taxonomy_elas_2021$ASV
taxonomy_elas_2021 <- taxonomy_elas_2021 %>% select(-ASV)
tax_table(ps_elas_2021) <- as.matrix(taxonomy_elas_2021)

#2023
taxonomy_elas_2023 <- as.data.frame(tax_table(ps_elas_2023)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_elas_2023) <- taxonomy_elas_2023$ASV
taxonomy_elas_2023 <- taxonomy_elas_2023 %>% select(-ASV)
tax_table(ps_elas_2023) <- as.matrix(taxonomy_elas_2023)

#2024
taxonomy_elas_2024 <- as.data.frame(tax_table(ps_elas_2024)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_elas_2024) <- taxonomy_elas_2024$ASV
taxonomy_elas_2024 <- taxonomy_elas_2024 %>% select(-ASV)
tax_table(ps_elas_2024) <- as.matrix(taxonomy_elas_2024)

#2025
taxonomy_elas_2025 <- as.data.frame(tax_table(ps_elas_2025)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_elas_2025) <- taxonomy_elas_2025$ASV
taxonomy_elas_2025 <- taxonomy_elas_2025 %>% select(-ASV)
tax_table(ps_elas_2025) <- as.matrix(taxonomy_elas_2025)



# CO1
#2024
taxonomy_co1_2024 <- as.data.frame(tax_table(ps_co1_2024)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_co1_2024) <- taxonomy_co1_2024$ASV
taxonomy_co1_2024 <- taxonomy_co1_2024 %>% select(-ASV)
tax_table(ps_co1_2024) <- as.matrix(taxonomy_co1_2024)

#2025
taxonomy_co1_2025 <- as.data.frame(tax_table(ps_co1_2025)) %>%
  rownames_to_column("ASV") %>%
  left_join(commonnames, by = c("Species" = "Taxon"))
row.names(taxonomy_co1_2025) <- taxonomy_co1_2025$ASV
taxonomy_co1_2025 <- taxonomy_co1_2025 %>% select(-ASV)
tax_table(ps_co1_2025) <- as.matrix(taxonomy_co1_2025)


```



Check for missing (NA) common names
(If there are, update commonnames database)
```{r}
as.data.frame(tax_table(ps_mifish_2020)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_mifish_2021)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_mifish_2022)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_mifish_2023)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_mifish_2024)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_mifish_2025)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))


as.data.frame(tax_table(ps_elas_2021)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_elas_2023)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_elas_2024)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_elas_2025)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))

as.data.frame(tax_table(ps_co1_2024)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
as.data.frame(tax_table(ps_co1_2025)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName))
```


`missingtaxaincommonnamedb` is a list of species that are missing from the common names database- update db as needed
```{r}
missingtaxaincommonnamedb <- as.data.frame(tax_table(ps_co1_2024)) %>% rownames_to_column("ASV") %>% filter(is.na(CommonName)) %>% select(Species) %>% unique()

missingtaxaincommonnamedb
```




## Remove Low Abundant ASVs
For MiFish, Natalia reviewed each sequencing run to determine unique cutoff based on crossover from positive controls. Each year has a different threshold for ASV removal

For Elas02, this was determined in 2021 with positive control tanks of non native species. Use 500 read threshold for all runs.

For CO1, use 150 based on guinea pig bleedover in samples (max 132 reads). There is more in blanks but that is due to low abundance of everything else in blanks

Based on some analysis in the `Expedition-eDNA-plots.Rmd` notebook, remove ASVs that have <500 reads in a sample
```{r}
data.frame(otu_table(ps_mifish_2020)) # compare before
ps_mifish_2020 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_mifish_2020,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_mifish_2020)) # and after

data.frame(otu_table(ps_mifish_2021)) # compare before
ps_mifish_2021 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_mifish_2021,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_mifish_2021)) # and after

data.frame(otu_table(ps_mifish_2022)) # compare before
ps_mifish_2022 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_mifish_2022,
  minabund = 1350,
  relabund = FALSE)
data.frame(otu_table(ps_mifish_2022)) # and after

data.frame(otu_table(ps_mifish_2023)) # compare before
ps_mifish_2023 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_mifish_2023,
  minabund = 1300,
  relabund = FALSE)
data.frame(otu_table(ps_mifish_2023)) # and after

data.frame(otu_table(ps_mifish_2024)) # compare before
ps_mifish_2024 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_mifish_2024,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_mifish_2024)) # and after

data.frame(otu_table(ps_mifish_2025)) # compare before
ps_mifish_2025 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_mifish_2025,
  minabund = 3000,
  relabund = FALSE)
data.frame(otu_table(ps_mifish_2025)) # and after

#Elas02
data.frame(otu_table(ps_elas_2021)) # compare before
ps_elas_2021 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_elas_2021,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_elas_2021)) # and after

data.frame(otu_table(ps_elas_2023)) # compare before
ps_elas_2023 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_elas_2023,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_elas_2023)) # and after

data.frame(otu_table(ps_elas_2024)) # compare before
ps_elas_2024 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_elas_2024,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_elas_2024)) # and after

data.frame(otu_table(ps_elas_2025)) # compare before
ps_elas_2025 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_elas_2025,
  minabund = 500,
  relabund = FALSE)
data.frame(otu_table(ps_elas_2025)) # and after


# CO1
data.frame(otu_table(ps_co1_2024)) # compare before
ps_co1_2024 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_co1_2024,
  minabund = 150,
  relabund = FALSE)
data.frame(otu_table(ps_co1_2024)) # and after

data.frame(otu_table(ps_co1_2025)) # compare before
ps_co1_2025 <- metagMisc::phyloseq_filter_sample_wise_abund_trim(
  ps_co1_2025,
  minabund = 150,
  relabund = FALSE)
data.frame(otu_table(ps_co1_2025)) # and after
```


## Remove Low Effort Samples

Plot bar plots of total sequence reads across all samples
```{r}
plot_bar(ps_mifish_2020)
plot_bar(ps_mifish_2021)
plot_bar(ps_mifish_2022)
plot_bar(ps_mifish_2023)
plot_bar(ps_mifish_2024)
plot_bar(ps_mifish_2025)


plot_bar(ps_elas_2021)
plot_bar(ps_elas_2023)
plot_bar(ps_elas_2024)
plot_bar(ps_elas_2025)

plot_bar(ps_co1_2024)
plot_bar(ps_co1_2025)

```

Cut out any sample with <25,000 reads from all libraries


```{r}
ps_mifish_2020 <- prune_samples(sample_sums(ps_mifish_2020)>=10000,ps_mifish_2020)
ps_mifish_2021 <- prune_samples(sample_sums(ps_mifish_2021)>=10000,ps_mifish_2021)
ps_mifish_2022 <- prune_samples(sample_sums(ps_mifish_2022)>=10000,ps_mifish_2022)
ps_mifish_2023 <- prune_samples(sample_sums(ps_mifish_2023)>=10000,ps_mifish_2023)
ps_mifish_2024 <- prune_samples(sample_sums(ps_mifish_2024)>=10000,ps_mifish_2024)
ps_mifish_2025 <- prune_samples(sample_sums(ps_mifish_2025)>=10000,ps_mifish_2025)



ps_elas_2021 <- prune_samples(sample_sums(ps_elas_2021)>=10000,ps_elas_2021)
ps_elas_2023 <- prune_samples(sample_sums(ps_elas_2023)>=10000,ps_elas_2023)
ps_elas_2024 <- prune_samples(sample_sums(ps_elas_2024)>=10000,ps_elas_2024)
ps_elas_2025 <- prune_samples(sample_sums(ps_elas_2025)>=10000,ps_elas_2025)


# dif threshold for CO1, for same reasons as above- sequencing depth is lower bc so many species. Set lower threshold
ps_co1_2024 <- prune_samples(sample_sums(ps_co1_2024)>=100,ps_co1_2024)
ps_co1_2025 <- prune_samples(sample_sums(ps_co1_2025)>=100,ps_co1_2025)



```



Plot again to check
```{r}
plot_bar(ps_mifish_2020)
plot_bar(ps_mifish_2021)
plot_bar(ps_mifish_2022)
plot_bar(ps_mifish_2023)
plot_bar(ps_mifish_2024)
plot_bar(ps_mifish_2025)


plot_bar(ps_elas_2021)
plot_bar(ps_elas_2023)
plot_bar(ps_elas_2024)
plot_bar(ps_elas_2025)


plot_bar(ps_co1_2024)
plot_bar(ps_co1_2025)


```


## Agglomerate 
Function `tax_glom` will group by Species or other rank. But the function expects full taxonomy, Kingdonm --> Species and I am glomming by "Taxon.CommonName" which is not even an actual rank. This ends up keeping "Anchovies" separate, for example, even though then have the same Taxon.CommonName field bc they have dif genus/ species. Removing all of the rest of the ranks won't work because the dimensions of the tax_table are too small and it gets treated as a vector instead of matrix. Retain only one rank, Kingdom, so the matrix has enough dimensions

```{r}
ps_mifish_2020_glommed <- ps_mifish_2020
ps_mifish_2021_glommed <- ps_mifish_2021
ps_mifish_2022_glommed <- ps_mifish_2022
ps_mifish_2023_glommed <- ps_mifish_2023
ps_mifish_2024_glommed <- ps_mifish_2024
ps_mifish_2025_glommed <- ps_mifish_2025

ps_elas_2021_glommed <- ps_elas_2021
ps_elas_2023_glommed <- ps_elas_2023
ps_elas_2024_glommed <- ps_elas_2024
ps_elas_2025_glommed <- ps_elas_2025


ps_co1_2024_glommed <- ps_co1_2024
ps_co1_2025_glommed <- ps_co1_2025





tax_table(ps_mifish_2020_glommed) <- tax_table(ps_mifish_2020_glommed)[,c(1, 9)]
tax_table(ps_mifish_2021_glommed) <- tax_table(ps_mifish_2021_glommed)[,c(1, 9)]
tax_table(ps_mifish_2022_glommed) <- tax_table(ps_mifish_2022_glommed)[,c(1, 9)]
tax_table(ps_mifish_2023_glommed) <- tax_table(ps_mifish_2023_glommed)[,c(1, 9)]
tax_table(ps_mifish_2024_glommed) <- tax_table(ps_mifish_2024_glommed)[,c(1, 9)]
tax_table(ps_mifish_2025_glommed) <- tax_table(ps_mifish_2025_glommed)[,c(1, 9)]


tax_table(ps_elas_2021_glommed) <- tax_table(ps_elas_2021_glommed)[,c(1, 9)]
tax_table(ps_elas_2023_glommed) <- tax_table(ps_elas_2023_glommed)[,c(1, 9)]
tax_table(ps_elas_2024_glommed) <- tax_table(ps_elas_2024_glommed)[,c(1, 9)]
tax_table(ps_elas_2025_glommed) <- tax_table(ps_elas_2025_glommed)[,c(1, 9)]


tax_table(ps_co1_2024_glommed) <- tax_table(ps_co1_2024_glommed)[,c(1, 9)]
tax_table(ps_co1_2025_glommed) <- tax_table(ps_co1_2025_glommed)[,c(1, 9)]


```


Then glom:
```{r}
ps_mifish_2020_glommed <- tax_glom(ps_mifish_2020_glommed, "Taxon.CommonName")
ps_mifish_2021_glommed <- tax_glom(ps_mifish_2021_glommed, "Taxon.CommonName")
ps_mifish_2022_glommed <- tax_glom(ps_mifish_2022_glommed, "Taxon.CommonName")
ps_mifish_2023_glommed <- tax_glom(ps_mifish_2023_glommed, "Taxon.CommonName")
ps_mifish_2024_glommed <- tax_glom(ps_mifish_2024_glommed, "Taxon.CommonName")
ps_mifish_2025_glommed <- tax_glom(ps_mifish_2025_glommed, "Taxon.CommonName")


ps_elas_2021_glommed <- tax_glom(ps_elas_2021_glommed, "Taxon.CommonName")
ps_elas_2023_glommed <- tax_glom(ps_elas_2023_glommed, "Taxon.CommonName")
ps_elas_2024_glommed <- tax_glom(ps_elas_2024_glommed, "Taxon.CommonName")
ps_elas_2025_glommed <- tax_glom(ps_elas_2025_glommed, "Taxon.CommonName")


ps_co1_2024_glommed <- tax_glom(ps_co1_2024_glommed, "Taxon.CommonName")
ps_co1_2025_glommed <- tax_glom(ps_co1_2025_glommed, "Taxon.CommonName")

```


check glommed data tables
```{r}
data.frame(tax_table(ps_mifish_2020)) # to compare
data.frame(tax_table(ps_mifish_2020_glommed))
data.frame(otu_table(ps_mifish_2020_glommed))

data.frame(tax_table(ps_mifish_2021)) # to compare
data.frame(tax_table(ps_mifish_2021_glommed))
data.frame(otu_table(ps_mifish_2021_glommed))

data.frame(tax_table(ps_mifish_2022)) # to compare
data.frame(tax_table(ps_mifish_2022_glommed))
data.frame(otu_table(ps_mifish_2022_glommed))

data.frame(tax_table(ps_mifish_2023)) # to compare
data.frame(tax_table(ps_mifish_2023_glommed))
data.frame(otu_table(ps_mifish_2023_glommed))

data.frame(tax_table(ps_mifish_2024)) # to compare
data.frame(tax_table(ps_mifish_2024_glommed))
data.frame(otu_table(ps_mifish_2024_glommed))

data.frame(tax_table(ps_mifish_2025)) # to compare
data.frame(tax_table(ps_mifish_2025_glommed))
data.frame(otu_table(ps_mifish_2025_glommed))



data.frame(tax_table(ps_elas_2021)) # to compare
data.frame(tax_table(ps_elas_2021_glommed))
data.frame(otu_table(ps_elas_2021_glommed))

data.frame(tax_table(ps_elas_2023)) # to compare
data.frame(tax_table(ps_elas_2023_glommed))
data.frame(otu_table(ps_elas_2023_glommed))

data.frame(tax_table(ps_elas_2024)) # to compare
data.frame(tax_table(ps_elas_2024_glommed))
data.frame(otu_table(ps_elas_2024_glommed))

data.frame(tax_table(ps_elas_2025)) # to compare
data.frame(tax_table(ps_elas_2025_glommed))
data.frame(otu_table(ps_elas_2025_glommed))



data.frame(tax_table(ps_co1_2024)) # to compare
data.frame(tax_table(ps_co1_2024_glommed))
data.frame(otu_table(ps_co1_2024_glommed))

data.frame(tax_table(ps_co1_2025)) # to compare
data.frame(tax_table(ps_co1_2025_glommed))
data.frame(otu_table(ps_co1_2025_glommed))

```


## krona plots

First need to remove weird taxonomy columns- make secondary, non-glommed ps object
```{r}
# mifish 
ps_mifish_2020_2 <- ps_mifish_2020
taxonomy_mifish_2020 <- as.data.frame(tax_table(ps_mifish_2020)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_mifish_2020) <- taxonomy_mifish_2020$ASV
taxonomy_mifish_2020 <- taxonomy_mifish_2020 %>% select(-ASV)
# new phyloseq object
tax_table(ps_mifish_2020_2) <- as.matrix(taxonomy_mifish_2020)

ps_mifish_2021_2 <- ps_mifish_2021
taxonomy_mifish_2021 <- as.data.frame(tax_table(ps_mifish_2021)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_mifish_2021) <- taxonomy_mifish_2021$ASV
taxonomy_mifish_2021 <- taxonomy_mifish_2021 %>% select(-ASV)
# new phyloseq object
tax_table(ps_mifish_2021_2) <- as.matrix(taxonomy_mifish_2021)

ps_mifish_2022_2 <- ps_mifish_2022
taxonomy_mifish_2022 <- as.data.frame(tax_table(ps_mifish_2022)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_mifish_2022) <- taxonomy_mifish_2022$ASV
taxonomy_mifish_2022 <- taxonomy_mifish_2022 %>% select(-ASV)
# new phyloseq object
tax_table(ps_mifish_2022_2) <- as.matrix(taxonomy_mifish_2022)

ps_mifish_2023_2 <- ps_mifish_2023
taxonomy_mifish_2023 <- as.data.frame(tax_table(ps_mifish_2023)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_mifish_2023) <- taxonomy_mifish_2023$ASV
taxonomy_mifish_2023 <- taxonomy_mifish_2023 %>% select(-ASV)
# new phyloseq object
tax_table(ps_mifish_2023_2) <- as.matrix(taxonomy_mifish_2023)

ps_mifish_2024_2 <- ps_mifish_2024
taxonomy_mifish_2024 <- as.data.frame(tax_table(ps_mifish_2024)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_mifish_2024) <- taxonomy_mifish_2024$ASV
taxonomy_mifish_2024 <- taxonomy_mifish_2024 %>% select(-ASV)
# new phyloseq object
tax_table(ps_mifish_2024_2) <- as.matrix(taxonomy_mifish_2024)

ps_mifish_2025_2 <- ps_mifish_2025
taxonomy_mifish_2025 <- as.data.frame(tax_table(ps_mifish_2025)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_mifish_2025) <- taxonomy_mifish_2025$ASV
taxonomy_mifish_2025 <- taxonomy_mifish_2025 %>% select(-ASV)
# new phyloseq object
tax_table(ps_mifish_2025_2) <- as.matrix(taxonomy_mifish_2025)



# elas
ps_elas_2021_2 <- ps_elas_2021
taxonomy_elas_2021 <- as.data.frame(tax_table(ps_elas_2021)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_elas_2021) <- taxonomy_elas_2021$ASV
taxonomy_elas_2021 <- taxonomy_elas_2021 %>% select(-ASV)
# new phyloseq object
tax_table(ps_elas_2021_2) <- as.matrix(taxonomy_elas_2021)

ps_elas_2023_2 <- ps_elas_2023
taxonomy_elas_2023 <- as.data.frame(tax_table(ps_elas_2023)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_elas_2023) <- taxonomy_elas_2023$ASV
taxonomy_elas_2023 <- taxonomy_elas_2023 %>% select(-ASV)
# new phyloseq object
tax_table(ps_elas_2023_2) <- as.matrix(taxonomy_elas_2023)

ps_elas_2024_2 <- ps_elas_2024
taxonomy_elas_2024 <- as.data.frame(tax_table(ps_elas_2024)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_elas_2024) <- taxonomy_elas_2024$ASV
taxonomy_elas_2024 <- taxonomy_elas_2024 %>% select(-ASV)
# new phyloseq object
tax_table(ps_elas_2024_2) <- as.matrix(taxonomy_elas_2024)

ps_elas_2025_2 <- ps_elas_2025
taxonomy_elas_2025 <- as.data.frame(tax_table(ps_elas_2025)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_elas_2025) <- taxonomy_elas_2025$ASV
taxonomy_elas_2025 <- taxonomy_elas_2025 %>% select(-ASV)
# new phyloseq object
tax_table(ps_elas_2025_2) <- as.matrix(taxonomy_elas_2025)



# co1
ps_co1_2024_2 <- ps_co1_2024
taxonomy_co1_2024 <- as.data.frame(tax_table(ps_co1_2024)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_co1_2024) <- taxonomy_co1_2024$ASV
taxonomy_co1_2024 <- taxonomy_co1_2024 %>% select(-ASV)
# new phyloseq object
tax_table(ps_co1_2024_2) <- as.matrix(taxonomy_co1_2024)

ps_co1_2025_2 <- ps_co1_2025
taxonomy_co1_2025 <- as.data.frame(tax_table(ps_co1_2025)) %>%
  rownames_to_column("ASV") %>% 
  select(!c("CommonName","Color"))
row.names(taxonomy_co1_2025) <- taxonomy_co1_2025$ASV
taxonomy_co1_2025 <- taxonomy_co1_2025 %>% select(-ASV)
# new phyloseq object
tax_table(ps_co1_2025_2) <- as.matrix(taxonomy_co1_2025)

```

MUST PASTE into the console (so removing from notebook chunk)

NOTE- this function doesn't allow you to name Krona files. I went in manually in `krona_files` and named them myself to share with collaborators

source('https://raw.githubusercontent.com/markschl/embed_krona/master/embed_krona.R')

mifish_2020_krona <- plot_krona(ps_mifish_2020_2) # gets written as krona_files/krona_1.html, etc
mifish_2021_krona <- plot_krona(ps_mifish_2021_2) 
mifish_2022_krona <- plot_krona(ps_mifish_2022_2) 
mifish_2023_krona <- plot_krona(ps_mifish_2023_2) 
mifish_2024_krona <- plot_krona(ps_mifish_2024_2) 
mifish_2025_krona <- plot_krona(ps_mifish_2025_2) 


elas_2021_krona <- plot_krona(ps_elas_2021_2) 
elas_2023_krona <- plot_krona(ps_elas_2023_2) 
elas_2024_krona <- plot_krona(ps_elas_2024_2) 
elas_2025_krona <- plot_krona(ps_elas_2025_2) 

co1_2024_krona <- plot_krona(ps_co1_2024_2)
co1_2025_krona <- plot_krona(ps_co1_2025_2) 


## Export tables
Every year has 5 tables:
1) ASV table
2) Taxonomy table (corresponds to ASVs)
3) Sample data
4) Glommed ASV table (aggregated by Taxon.CommonName)
5) Glommed taxonomy table (corresponds to glommed taxa from #4)

### Mifish
```{r}
#2020
mifish_2020_asv_table <- data.frame(otu_table(ps_mifish_2020))
write.csv(mifish_2020_asv_table,"filtered_data/mifish_2020_asv_table.csv", row.names = TRUE)

mifish_2020_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2020)), var = "ASV")
write.csv(mifish_2020_tax_table,"filtered_data/mifish_2020_tax_table.csv", row.names = FALSE)

mifish_2020_sample_data <- data.frame(sample_data(ps_mifish_2020))
write.csv(mifish_2020_sample_data,"filtered_data/mifish_2020_sample_data.csv", row.names = TRUE)

mifish_2020_glommed_asv_table <- data.frame(otu_table(ps_mifish_2020_glommed))
write.csv(mifish_2020_glommed_asv_table,"filtered_data/mifish_2020_glommed_asv_table.csv", row.names = TRUE)

mifish_2020_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2020_glommed)), var = "1stRepresentative_ASV")
write.csv(mifish_2020_glommed_tax_table,"filtered_data/mifish_2020_glommed_tax_table.csv", row.names = FALSE)

#2021
mifish_2021_asv_table <- data.frame(otu_table(ps_mifish_2021))
write.csv(mifish_2021_asv_table,"filtered_data/mifish_2021_asv_table.csv", row.names = TRUE)

mifish_2021_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2021)), var = "ASV")
write.csv(mifish_2021_tax_table,"filtered_data/mifish_2021_tax_table.csv", row.names = FALSE)

mifish_2021_sample_data <- data.frame(sample_data(ps_mifish_2021))
write.csv(mifish_2021_sample_data,"filtered_data/mifish_2021_sample_data.csv", row.names = TRUE)

mifish_2021_glommed_asv_table <- data.frame(otu_table(ps_mifish_2021_glommed))
write.csv(mifish_2021_glommed_asv_table,"filtered_data/mifish_2021_glommed_asv_table.csv", row.names = TRUE)

mifish_2021_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2021_glommed)), var = "1stRepresentative_ASV")
write.csv(mifish_2021_glommed_tax_table,"filtered_data/mifish_2021_glommed_tax_table.csv", row.names = FALSE)

#2022
mifish_2022_asv_table <- data.frame(otu_table(ps_mifish_2022))
write.csv(mifish_2022_asv_table,"filtered_data/mifish_2022_asv_table.csv", row.names = TRUE)

mifish_2022_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2022)), var = "ASV")
write.csv(mifish_2022_tax_table,"filtered_data/mifish_2022_tax_table.csv", row.names = FALSE)

mifish_2022_sample_data <- data.frame(sample_data(ps_mifish_2022))
write.csv(mifish_2022_sample_data,"filtered_data/mifish_2022_sample_data.csv", row.names = TRUE)

mifish_2022_glommed_asv_table <- data.frame(otu_table(ps_mifish_2022_glommed))
write.csv(mifish_2022_glommed_asv_table,"filtered_data/mifish_2022_glommed_asv_table.csv", row.names = TRUE)

mifish_2022_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2022_glommed)), var = "1stRepresentative_ASV")
write.csv(mifish_2022_glommed_tax_table,"filtered_data/mifish_2022_glommed_tax_table.csv", row.names = FALSE)

#2023
mifish_2023_asv_table <- data.frame(otu_table(ps_mifish_2023))
write.csv(mifish_2023_asv_table,"filtered_data/mifish_2023_asv_table.csv", row.names = TRUE)

mifish_2023_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2023)), var = "ASV")
write.csv(mifish_2023_tax_table,"filtered_data/mifish_2023_tax_table.csv", row.names = FALSE)

mifish_2023_sample_data <- data.frame(sample_data(ps_mifish_2023))
write.csv(mifish_2023_sample_data,"filtered_data/mifish_2023_sample_data.csv", row.names = TRUE)

mifish_2023_glommed_asv_table <- data.frame(otu_table(ps_mifish_2023_glommed))
write.csv(mifish_2023_glommed_asv_table,"filtered_data/mifish_2023_glommed_asv_table.csv", row.names = TRUE)

mifish_2023_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2023_glommed)), var = "1stRepresentative_ASV")
write.csv(mifish_2023_glommed_tax_table,"filtered_data/mifish_2023_glommed_tax_table.csv", row.names = FALSE)

#2024
mifish_2024_asv_table <- data.frame(otu_table(ps_mifish_2024))
write.csv(mifish_2024_asv_table,"filtered_data/mifish_2024_asv_table.csv", row.names = TRUE)

mifish_2024_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2024)), var = "ASV")
write.csv(mifish_2024_tax_table,"filtered_data/mifish_2024_tax_table.csv", row.names = FALSE)

mifish_2024_sample_data <- data.frame(sample_data(ps_mifish_2024))
write.csv(mifish_2024_sample_data,"filtered_data/mifish_2024_sample_data.csv", row.names = TRUE)

mifish_2024_glommed_asv_table <- data.frame(otu_table(ps_mifish_2024_glommed))
write.csv(mifish_2024_glommed_asv_table,"filtered_data/mifish_2024_glommed_asv_table.csv", row.names = TRUE)

mifish_2024_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2024_glommed)), var = "1stRepresentative_ASV")
write.csv(mifish_2024_glommed_tax_table,"filtered_data/mifish_2024_glommed_tax_table.csv", row.names = FALSE)


#2025
mifish_2025_asv_table <- data.frame(otu_table(ps_mifish_2025))
write.csv(mifish_2025_asv_table,"filtered_data/mifish_2025_asv_table.csv", row.names = TRUE)

mifish_2025_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2025)), var = "ASV")
write.csv(mifish_2025_tax_table,"filtered_data/mifish_2025_tax_table.csv", row.names = FALSE)

mifish_2025_sample_data <- data.frame(sample_data(ps_mifish_2025))
write.csv(mifish_2025_sample_data,"filtered_data/mifish_2025_sample_data.csv", row.names = TRUE)

mifish_2025_glommed_asv_table <- data.frame(otu_table(ps_mifish_2025_glommed))
write.csv(mifish_2025_glommed_asv_table,"filtered_data/mifish_2025_glommed_asv_table.csv", row.names = TRUE)

mifish_2025_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_mifish_2025_glommed)), var = "1stRepresentative_ASV")
write.csv(mifish_2025_glommed_tax_table,"filtered_data/mifish_2025_glommed_tax_table.csv", row.names = FALSE)

```



### Elas02
```{r}
#2021
elas_2021_asv_table <- data.frame(otu_table(ps_elas_2021))
write.csv(elas_2021_asv_table,"filtered_data/elas_2021_asv_table.csv", row.names = TRUE)

elas_2021_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2021)), var = "ASV")
write.csv(elas_2021_tax_table,"filtered_data/elas_2021_tax_table.csv", row.names = FALSE)

elas_2021_sample_data <- data.frame(sample_data(ps_elas_2021))
write.csv(elas_2021_sample_data,"filtered_data/elas_2021_sample_data.csv", row.names = TRUE)

elas_2021_glommed_asv_table <- data.frame(otu_table(ps_elas_2021_glommed))
write.csv(elas_2021_glommed_asv_table,"filtered_data/elas_2021_glommed_asv_table.csv", row.names = TRUE)

elas_2021_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2021_glommed)), var = "1stRepresentative_ASV")
write.csv(elas_2021_glommed_tax_table,"filtered_data/elas_2021_glommed_tax_table.csv", row.names = FALSE)

#2023
elas_2023_asv_table <- data.frame(otu_table(ps_elas_2023))
write.csv(elas_2023_asv_table,"filtered_data/elas_2023_asv_table.csv", row.names = TRUE)

elas_2023_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2023)), var = "ASV")
write.csv(elas_2023_tax_table,"filtered_data/elas_2023_tax_table.csv", row.names = FALSE)

elas_2023_sample_data <- data.frame(sample_data(ps_elas_2023))
write.csv(elas_2023_sample_data,"filtered_data/elas_2023_sample_data.csv", row.names = TRUE)

elas_2023_glommed_asv_table <- data.frame(otu_table(ps_elas_2023_glommed))
write.csv(elas_2023_glommed_asv_table,"filtered_data/elas_2023_glommed_asv_table.csv", row.names = TRUE)

elas_2023_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2023_glommed)), var = "1stRepresentative_ASV")
write.csv(elas_2023_glommed_tax_table,"filtered_data/elas_2023_glommed_tax_table.csv", row.names = FALSE)

#2024
elas_2024_asv_table <- data.frame(otu_table(ps_elas_2024))
write.csv(elas_2024_asv_table,"filtered_data/elas_2024_asv_table.csv", row.names = TRUE)

elas_2024_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2024)), var = "ASV")
write.csv(elas_2024_tax_table,"filtered_data/elas_2024_tax_table.csv", row.names = FALSE)

elas_2024_sample_data <- data.frame(sample_data(ps_elas_2024))
write.csv(elas_2024_sample_data,"filtered_data/elas_2024_sample_data.csv", row.names = TRUE)

elas_2024_glommed_asv_table <- data.frame(otu_table(ps_elas_2024_glommed))
write.csv(elas_2024_glommed_asv_table,"filtered_data/elas_2024_glommed_asv_table.csv", row.names = TRUE)

elas_2024_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2024_glommed)), var = "1stRepresentative_ASV")
write.csv(elas_2024_glommed_tax_table,"filtered_data/elas_2024_glommed_tax_table.csv", row.names = FALSE)


#2025
elas_2025_asv_table <- data.frame(otu_table(ps_elas_2025))
write.csv(elas_2025_asv_table,"filtered_data/elas_2025_asv_table.csv", row.names = TRUE)

elas_2025_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2025)), var = "ASV")
write.csv(elas_2025_tax_table,"filtered_data/elas_2025_tax_table.csv", row.names = FALSE)

elas_2025_sample_data <- data.frame(sample_data(ps_elas_2025))
write.csv(elas_2025_sample_data,"filtered_data/elas_2025_sample_data.csv", row.names = TRUE)

elas_2025_glommed_asv_table <- data.frame(otu_table(ps_elas_2025_glommed))
write.csv(elas_2025_glommed_asv_table,"filtered_data/elas_2025_glommed_asv_table.csv", row.names = TRUE)

elas_2025_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_elas_2025_glommed)), var = "1stRepresentative_ASV")
write.csv(elas_2025_glommed_tax_table,"filtered_data/elas_2025_glommed_tax_table.csv", row.names = FALSE)

```



### CO1
put "fortrawlcomparisononly" in file name because removed a lot of the organisms that will be interesting for other purposes
```{r}

#2024
co1_2024_asv_table <- data.frame(otu_table(ps_co1_2024))
write.csv(co1_2024_asv_table,"filtered_data/co1_2024_asv_table_fortrawlcomparisononly.csv", row.names = TRUE)

co1_2024_tax_table <- rownames_to_column(data.frame(tax_table(ps_co1_2024)), var = "ASV")
write.csv(co1_2024_tax_table,"filtered_data/co1_2024_tax_table_fortrawlcomparisononly.csv", row.names = FALSE)

co1_2024_sample_data <- data.frame(sample_data(ps_co1_2024))
write.csv(co1_2024_sample_data,"filtered_data/co1_2024_sample_data_fortrawlcomparisononly.csv", row.names = TRUE)

co1_2024_glommed_asv_table <- data.frame(otu_table(ps_co1_2024_glommed))
write.csv(co1_2024_glommed_asv_table,"filtered_data/co1_2024_glommed_asv_table_fortrawlcomparisononly.csv", row.names = TRUE)

co1_2024_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_co1_2024_glommed)), var = "1stRepresentative_ASV")
write.csv(co1_2024_glommed_tax_table,"filtered_data/co1_2024_glommed_tax_table_fortrawlcomparisononly.csv", row.names = FALSE)


#2025
co1_2025_asv_table <- data.frame(otu_table(ps_co1_2025))
write.csv(co1_2025_asv_table,"filtered_data/co1_2025_asv_table_fortrawlcomparisononly.csv", row.names = TRUE)

co1_2025_tax_table <- rownames_to_column(data.frame(tax_table(ps_co1_2025)), var = "ASV")
write.csv(co1_2025_tax_table,"filtered_data/co1_2025_tax_table_fortrawlcomparisononly.csv", row.names = FALSE)

co1_2025_sample_data <- data.frame(sample_data(ps_co1_2025))
write.csv(co1_2025_sample_data,"filtered_data/co1_2025_sample_data_fortrawlcomparisononly.csv", row.names = TRUE)

co1_2025_glommed_asv_table <- data.frame(otu_table(ps_co1_2025_glommed))
write.csv(co1_2025_glommed_asv_table,"filtered_data/co1_2025_glommed_asv_table_fortrawlcomparisononly.csv", row.names = TRUE)

co1_2025_glommed_tax_table <- rownames_to_column(data.frame(tax_table(ps_co1_2025_glommed)), var = "1stRepresentative_ASV")
write.csv(co1_2025_glommed_tax_table,"filtered_data/co1_2025_glommed_tax_table_fortrawlcomparisononly.csv", row.names = FALSE)

```

