---
title: "Plots"
output: html_notebook
---

My own script for making bubble plots, etc. Also partially borrowed/ modifed from REVAMP's [phyloseq.R](https://github.com/McAllister-NOAA/REVAMP/blob/main/assets/phyloseq.R)


<br/>

# Table of Contents
- [Load packages](#load-packages)
- [Import and prepare data](#import-and-prepare-data)
- [QC and filtering](#qc-and-filtering)
- [Visualizations](#visualizations)

      
<br/>



# Load packages

```{r}
library(tidyverse)
library(phyloseq)
library(vegan)
library(RColorBrewer)
library(microbiome)
library(metagMisc)
library(fantaxtic)

# library(readxl)
# library(Biostrings)
# #library(phangorn)
# library(readr)
# library(seqinr)
# #library(decontam)
# library(ape)
# #library(philr)
# #library(DESeq2)
# library(compositions);
# #library(cowplot)
# library(plotly)
# library(htmlwidgets)
# library(withr)
# library(lubridate)
```

#  Import and prepare data

```{r}
theme_set(theme_bw())

chem_headers <- read.delim(as.character("results-revamp-2021-Elas02/chem_headers.txt", header=FALSE, stringsAsFactors=FALSE))
chem_headers <- as.vector(chem_headers)

```

Import-
I am creating the equivalent of "phylo1" from REVAMP, the dataset is the ASV-based, raw reads, not qual filtered dataset; "qual filtered" means the unknown ASVs have been removed, & low effort samples removed. The only filtering that has been done is the unknown ASVs have been removed.

Side note- I checked some of the high abundance unknowns that were thrown out (ASV_4 from 2021 and ASV_2 and _3 from 2023) and they are diatoms. They are coming up as unknown bc they are not in my Elas02 database. 

**2021 data**
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-Elas02/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-Elas02/ASV2Taxonomy/results-revamp-2021-Elas02_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2021-Elas02/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_2021 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_2021 = tax_table(asv_taxonomy_mat)
samples_2021 = sample_data(sample_metadata)

phylo_2021 <- phyloseq(ASV_2021, TAX_2021, samples_2021)

```

```{r}
sample_order2021 <- read.delim("results-revamp-2021-Elas02/sample_order.txt", header=FALSE, stringsAsFactors=FALSE)
sample_order2021 <- as.vector(sample_order2021)

siteorder2021 <- as.data.frame(sample_metadata$sites)
siteorder2021 <- distinct(siteorder2021)
siteorder2021 <- as.vector(siteorder2021)


```

**2023 data**
```{r}
asv_count <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-Elas02/ASV2Taxonomy/ASVs_counts_NOUNKNOWNS.tsv", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_count) <- asv_count$x
asv_count <- asv_count %>% select(-x)
asv_count_mat <- as.matrix(asv_count)

asv_taxonomy <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-Elas02/ASV2Taxonomy/results-revamp-2023-Elas02_asvTaxonomyTable_NOUNKNOWNS.txt", header=TRUE, stringsAsFactors=FALSE)
row.names(asv_taxonomy) <- asv_taxonomy$ASV
asv_taxonomy <- asv_taxonomy %>% select(-ASV)
asv_taxonomy_mat <- as.matrix(asv_taxonomy)

sample_metadata <- read.delim("/Volumes/easystore/eDNA/shirp-edna/results-revamp-2023-Elas02/sample_metadata_forR.txt", header=TRUE, stringsAsFactors=TRUE)
row.names(sample_metadata) <- sample_metadata$Sample
sample_metadata <- sample_metadata %>% select(-Sample)

ASV_2023 = otu_table(asv_count_mat, taxa_are_rows = TRUE)
TAX_2023 = tax_table(asv_taxonomy_mat)
samples_2023 = sample_data(sample_metadata)

phylo_2023 <- phyloseq(ASV_2023, TAX_2023, samples_2023)

```

```{r}
sample_order2023 <- read.delim("results-revamp-2023-Elas02/sample_order.txt", header=FALSE, stringsAsFactors=FALSE)
sample_order2023 <- as.vector(sample_order2023)

siteorder2023 <- as.data.frame(sample_metadata$sites)
siteorder2023 <- distinct(siteorder2023)
siteorder2023 <- as.vector(siteorder2023)

```






# QC and filtering


## Rarefaction curves

```{r}
# output from phyloseq needs to be converted to matrix and transposed
table1 <- otu_table(phylo_2021)
class(table1) <- "matrix" 
table1 <- t(table1) 

rarecurve(table1, step=50, cex=0.5)

```
A few don't look great (T9S7, T5S6, T3Wessuck, T1Tiana, T7S7) but most are good!


```{r}
# output from phyloseq needs to be converted to matrix and transposed
table1 <- otu_table(phylo_2023)
class(table1) <- "matrix" 
table1 <- t(table1) 

rarecurve(table1, step=50, cex=0.5)

```
Keep an eye on SC5, SS9, SS3, T3S2, T5S1, T3S7, T3WC, and a few others

## Curating

For quick reference, scan through Krona plots:
- 2021 dataset [here](results-revamp-2021-Elas02/ASV2Taxonomy/KRONA_plots/results-revamp-2021-Elas02_samplesSummedKRONA.html) and 
- 2023 dataset [here](results-revamp-2023-Elas02/ASV2Taxonomy/KRONA_plots/results-revamp-2023-Elas02_samplesSummedKRONA.html))
- Note, unknowns in Krona plots were already removed from phylo objects

**2021**
Check some features of the phyloseq object
```{r}
ranks <- rank_names(phylo_2021)
ranks

unique(tax_table(phylo_2021)[, ranks[1]])
unique(tax_table(phylo_2021)[, ranks[2]])
unique(tax_table(phylo_2021)[, ranks[3]])
```

Remove the birds, mammals, and Actinopteri- even if some of these have exact match to primers, they are not really subject of study.
```{r}
nrow(tax_table(phylo_2021)) # number of taxa before filtering
ps2021 <- subset_taxa(phylo_2021, !Class == 'Aves' & !Class == 'Mammalia' & !Class == 'Actinopteri')
nrow(tax_table(ps2021)) # number of taxa left
unique(tax_table(ps2021)[, ranks[3]])
```
**2023**
Check some features of the phyloseq object
```{r}
ranks <- rank_names(phylo_2023)
ranks

unique(tax_table(phylo_2023)[, ranks[1]])
unique(tax_table(phylo_2023)[, ranks[2]])
unique(tax_table(phylo_2023)[, ranks[3]])
```

Remove the birds, mammals, and Actinopteri- even if some of these have exact match to primers, they are not really subject of study.
```{r}
nrow(tax_table(phylo_2023)) # number of taxa before filtering
ps2023 <- subset_taxa(phylo_2023, !Class == 'Aves' & !Class == 'Mammalia' & !Class == 'Actinopteri')
nrow(tax_table(ps2023)) # number of taxa left
unique(tax_table(ps2023)[, ranks[3]])
```



Remove low abundant reads
- Scan REVAMP output:`results-revamp-2021-Elas02_asvTaxonomyTable.txt` and `results-revamp-2023-Elas02_asvTaxonomyTable.txt` have taxonomy of all ASVs. 
  - In 2021, there are a few that don't make sense. There were a lot of pos controls that year and it seems that some crossover occurred and pos control reads were detected in other samples:
    - leopard shark (Triakis semifasciata: ASV_123, ASV_1406, ASV_22, ASV_293, ASV_308, ASV_556, ASV_659, ASV_801, ASV_940]
    - chain catshark [Scyliorhinus retifer: ASV_1030, ASV_1254, ASV_1495, ASV_1496, ASV_220, ASV_280, ASV_397, ASV_584, ASV_631, ASV_7, ASV_715, ASV_797]
    - Arabian smooth-hound [Mustelus mosis: ASV_78]
    - brown smooth-hound [Mustelus henlei: ASV_616, ASV_619]
    - Japanese wobbegong [Orectolobus japonicus, ASV_234]
    - and nurse shark [Ginglymostoma cirratum: ASV129, ASV_588]) 

From Libby's MS:
"Subcutaneous muscle tissue samples were collected for two local shark species: spiny dogfish (Squalus acanthius) and smooth dogfish (Musculus canis). A tissue sample was taken from each species. Each specimen was caught in Long Island waters and stored at -20Â°C as part of an alternate study. Water samples were taken from various tanks containing sharks at the Long Island Aquarium in October 2021. These tanks contained known species of elasmobranchs, including sand tiger sharks (Charcharias taurus), an ornate wobbegong (Orectolobus japonicus), and nurse sharks (Ginglymostoma cirratum). Additional water samples were taken from a shark tank at Cornell Cooperative Extension in Babylon, NY in October of 2021. This tank contained only chain catshark (Scyliorhinus retifer) but had a shared sump system with other elasmobranchs and used bay water. All positive controls were processed following the same protocol as the experimental samples."

**Might be interesting to make a plot of 2021 positive control detections too!**

Check some of these weird annotations:
```{r}
unique(tax_table(ps2021)[, ranks[7]])
unique(tax_table(ps2023)[, ranks[7]])


weird_taxa <- c("Triakis semifasciata", "Scyliorhinus retifer", "Mustelus mosis", "Mustelus henlei", "Orectolobus japonicus", "Ginglymostoma cirratum", "Mustelus griseus")
```

Check their read abundance from 2021:
```{r}
weird_taxa_ps <- subset_taxa(phylo_2021, Species %in% weird_taxa)
data.frame(otu_table(weird_taxa_ps)[,11:92]) # skip first 10 columns which are pos controls. These are expected to have some of these strange taxa. 
```

- Most are zeroes. Where detected, most of these are less than 1000 reads except for ASV_78 (Mustelus mosis), in T5S11, which has over 8000 reads. I blasted this ASV and its second blast hit is Mustelus canis so I think this is a misannotation.
- All others seem like bleed over from positive controls


Check their read abundance from 2023:
```{r}
weird_taxa_ps <- subset_taxa(phylo_2023, Species %in% weird_taxa)
data.frame(otu_table(weird_taxa_ps)) 
```
Again mostly zeroes. Highest read abundance is ~650 reads.

Remove ASVs with less than 1000 reads in a sample. 
```{r}
ps2021 <- phyloseq_filter_sample_wise_abund_trim(
  ps2021,
  minabund = 1000,
  relabund = FALSE)

data.frame(otu_table(ps2021))
```
```{r}
ps2023 <- phyloseq_filter_sample_wise_abund_trim(
  ps2023,
  minabund = 1000,
  relabund = FALSE)

data.frame(otu_table(ps2023))
```


### Manual curation

There are some abundant taxa resolved only to the genus level. For example:
- In 2021, Squalus makes up 8% of the Chondricthyes
- In 2023, Bathytosia makes up 19% of the Chondricthyes

I don't want to lose these  when I group by species. Can use NArm = FALSE option in `tax_glom` but then they will plot as NA and I will lose all tax information in the species plot... 

  - Found a cool package for replacing NAs with next lowest known rank, [fantaxtic](https://github.com/gmteunisse/fantaxtic)

- There are only [3 species of Bathytosia](https://www.fishbase.us/Nomenclature/ScientificNameSearchList.php?crit1_value=Bathytoshia) and only one is from the North Atlantic, *Bathytoshia centroura*, the roughtail stingray. So these are likely those. Replace the Species rank name, NA, with "Bathytoshia sp."

- There are [many more Squalus species](https://www.fishbase.us/Nomenclature/ScientificNameSearchList.php?) so harder to say but these are likely Squalus acanthias, spiny dogfish, which are common regionally. Still the blast hits are not really conclusive so can replace with *Squalus sp.*

- There are some Rhinoptera only identified to the genus level. Their blast hits are kind of all over the genus so replace with *Rhinoptera sp.*

- The genus Pteroplatytrygon was also identified at low abundance. There is only one species in this genus, *Pteroplatytrygon violacea*, so it is very likely this species... I don't know why REVAMP didn't pull out the spp name in this case. Replace with *Pteroplatytrygon sp.*

- **Remember to discuss all this in MS.**
  
```{r}
# Check what they look lik before replacement
data.frame(tax_table(subset_taxa(ps2021, is.na(Species))))
data.frame(tax_table(subset_taxa(ps2023, is.na(Species))))
```

```{r}
# Replace NA in Species with "Name-of-genus sp."
tax_table(ps2021) <- tax_table(name_na_taxa(ps2021, include_rank = F, na_label = "<tax> sp."))
tax_table(ps2023) <- tax_table(name_na_taxa(ps2023, include_rank = F, na_label = "<tax> sp."))

# Check after replacement
data.frame(tax_table(ps2021))
data.frame(tax_table(ps2023))

```




## Agglomerate by taxonomy
Generate new otu tables after grouping ASVs by annotated species name
```{r}
ps2021glommed <- tax_glom(ps2021, "Species")

# plot total reads
plot_bar(ps2021glommed, x = "Sample")

# view tax_table
data.frame(tax_table(ps2021glommed))
```



```{r}
ps2023glommed <- tax_glom(ps2023, "Species", NArm = FALSE)

# plot total reads
plot_bar(ps2023glommed, x = "Sample")

# view tax_table
data.frame(tax_table(ps2023glommed))
```

## Export taxonomy-based tables for collaborator

**2021**
```{r}
elas02_2021_asv_table_taxbased <- data.frame(otu_table(ps2021glommed))
elas02_2021_tax_table_taxbased <- data.frame(tax_table(ps2021glommed))
elas02_2021_sample_data_taxbased <- data.frame(sample_data(ps2021glommed))


# add on common names to tax_table
commonnames <- read_csv(file = "../eDNA-databases/commonnames.csv")
elas02_2021_tax_table_taxbased <- left_join(elas02_2021_tax_table_taxbased, commonnames, by = "Species")

# view
elas02_2021_asv_table_taxbased
elas02_2021_tax_table_taxbased
elas02_2021_sample_data_taxbased

# export
write.csv(elas02_2021_asv_table_taxbased,"figures/elas02_2021_asv_table_taxbased.csv", row.names = TRUE)
write.csv(elas02_2021_tax_table_taxbased,"figures/elas02_2021_tax_table_taxbased.csv", row.names = TRUE)
write.csv(elas02_2021_sample_data_taxbased,"figures/elas02_2021_sample_data_taxbased.csv", row.names = TRUE)
```


**2023**
```{r}
elas02_2023_asv_table_taxbased <- data.frame(otu_table(ps2023glommed))
elas02_2023_tax_table_taxbased <- data.frame(tax_table(ps2023glommed))
elas02_2023_sample_data_taxbased <- data.frame(sample_data(ps2023glommed))


# add on common names to tax_table
commonnames <- read_csv(file = "../eDNA-databases/commonnames.csv")
elas02_2023_tax_table_taxbased <- left_join(elas02_2023_tax_table_taxbased, commonnames, by = "Species")

# view
elas02_2023_asv_table_taxbased
elas02_2023_tax_table_taxbased
elas02_2023_sample_data_taxbased

# export
write.csv(elas02_2023_asv_table_taxbased,"figures/elas02_2023_asv_table_taxbased.csv", row.names = TRUE)
write.csv(elas02_2023_tax_table_taxbased,"figures/elas02_2023_tax_table_taxbased.csv", row.names = TRUE)
write.csv(elas02_2023_sample_data_taxbased,"figures/elas02_2023_sample_data_taxbased.csv", row.names = TRUE)
```


# Visualizations

### CONTINUE HERE 7/23

For plotting, use *relative abundances* (# of ASV sequences/sum total sequences in sample), calculated easily using microbiome::transform

```{r}
ps_ra <- microbiome::transform(ps, transform = "compositional")
```

Export the relative abundance matrix so Sara can have it:
```{r}
# Extract abundance matrix from the phyloseq object
RelAbun_matrix = as(otu_table(ps_ra), "matrix")

# Coerce to data.frame
RelAbun_dataframe = as.data.frame(RelAbun_matrix)

# Export
write.csv(RelAbun_dataframe,"results-elasmos/otutable_relabun_qualfilt.csv", row.names = TRUE)

## Also export the quality-filtered absolute abundance data
AbsAbun_matrix = as(otu_table(ps), "matrix")
AbsAbun_dataframe = as.data.frame(AbsAbun_matrix)
write.csv(AbsAbun_dataframe,"results-elasmos/otutable_absabun_qualfilt.csv", row.names = TRUE)


```



## Abundance at family level
Then aglomerate the ASVs at the family level using the phyloseq function, tax_glom
```{r}
familyGlommed_RA = tax_glom(ps_ra, "family")
family_barplot <- plot_bar(familyGlommed_RA, x = "Sample", fill = "family")
family_barplot


ggsave(filename = "Figures-elasmos/familybarplot.eps", plot = family_barplot, units = c("in"), width = 9, height = 6, dpi = 300, )# and save

```

Notice the diff between the clam/ oyster samples (AC/As JC/JS SC/SS) vs the T open water samples!


Remove Pos and Neg  Controls 
```{r}
ps <- subset_samples(ps, !SampleID == "A.Blank" & !SampleID == "A.Positive" & !SampleID == "J.Blank" & !SampleID == "J.Positive" & !SampleID == "S.Blank" & !SampleID == "S.Positive" & !SampleID == "T2Blank" & !SampleID == "T2Positive" & !SampleID == "T3Blank" & !SampleID == "T3Positive" & !SampleID == "T5Blank" & !SampleID == "T5Positive")


ps_ra <- subset_samples(ps_ra, !SampleID == "A.Blank" & !SampleID == "A.Positive" & !SampleID == "J.Blank" & !SampleID == "J.Positive" & !SampleID == "S.Blank" & !SampleID == "S.Positive" & !SampleID == "T2Blank" & !SampleID == "T2Positive" & !SampleID == "T3Blank" & !SampleID == "T3Positive" & !SampleID == "T5Blank" & !SampleID == "T5Positive")

ps
ps_ra
```
62 samples remain

Agglomerate by species to take a look at the unique species
```{r}
speciesGlommed_RA = tax_glom(ps_ra, "CommonName")
speciesGlommed_RA
tax_table(speciesGlommed_RA)
```



## Bubble plots
Based on my previous [scripts](https://github.com/lizsuter/Cariaco_Euk) with Cariaco Eukaryotic data
```{r}
# convert ps object to dataframe using phyloseq's psmelt
species_df <- psmelt(speciesGlommed_RA)

# replace zeroes in the table with NA
species_df[species_df == 0] <- NA

# and remove rows with NAs in abundance  (this is so they don't appear as small dots in plot)
species_df <-  filter(species_df, !is.na(Abundance))
```



Plot by species, scientific name
```{r}
speciesbubbleplot_eDNA_sciname <- ggplot(species_df, aes(x = Station_ShortName, y = fct_rev(species), color = Station)) + # the fancy stuff around y (species) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size = Abundance, fill = Station), color = "black", pch = 21)+
  scale_size(range = c(1,15)) +
  scale_size_area(breaks = c(0,.25,.5,.75,1), max_size = 6)+
  xlab("")+
  ylab("")+
  labs(size="Relative Abundance")+
  theme_bw() +
#  scale_fill_brewer(palette="Paired") +
  theme(axis.title.x=element_blank()) +
  facet_grid(Month~Bayside, scales = "free", space = "free", drop= TRUE)

speciesbubbleplot_eDNA_sciname
```

* Note on above that sampling dates were not same for open water stations and bivalve stations (C* and S* stations) although they were a few days apart

Plot by species common name
```{r}
speciesbubbleplot_eDNA_comname <- ggplot(species_df, aes(x = Station_ShortName, y = fct_rev(CommonName), color = Station)) + # the fancy stuff around y (CommonName) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = Station), color = "black", pch = 21)+
  scale_size(range = c(.25,15)) +
  scale_size_area(breaks = c(.25,.5,.75,1), max_size = 6)+
  xlab("")+
  ylab("")+
  labs(size="Relative Abundance")+
  theme_bw() +
#  scale_fill_brewer(palette="Paired") +
  theme(axis.title.x=element_blank()) +
  facet_grid(Month~Bayside, scales = "free", space = "free", drop= TRUE)

speciesbubbleplot_eDNA_comname
```


Exportfigures
```{r}
ggsave(filename = "Figures-elasmos/speciesbubbleplot_eDNA_sciname.eps", plot = speciesbubbleplot_eDNA_sciname, units = c("in"), width = 8, height = 7, dpi = 300)

ggsave(filename = "Figures-elasmos/speciesbubbleplot_eDNA_comname.eps", plot = speciesbubbleplot_eDNA_comname, units = c("in"), width = 8, height = 7, dpi = 300)
```


Plot by common name- # of reads (rather than rel abund)
```{r}
speciesGlommed = tax_glom(ps, "CommonName")
species_df_reads <- psmelt(speciesGlommed)

# replace zeroes in the table with NA
species_df_reads[species_df_reads == 0] <- NA

# and remove rows with NAs in abundance  (this is so they don't appear as small dots in plot)
species_df_reads <-  filter(species_df_reads, !is.na(Abundance))

speciesbubbleplot_eDNA_comname_reads <- ggplot(species_df_reads, aes(x = Station_ShortName, y = fct_rev(CommonName), color = Station)) + # the fancy stuff around y (CommonName) helps to present it in reverse order in the plot (from top to btm alphabetically)
  geom_point(aes(size=ifelse(Abundance==0, NA, Abundance), fill = Station), color = "black", pch = 21) +
#  scale_size(range = c(.25,15)) +
  scale_size_area(breaks = c(1000,10000,100000,1000000,10000000))+
  xlab("")+
  ylab("")+
  labs(size="Read Abundance")+
  theme_bw() +
#  scale_fill_brewer(palette="Paired") +
  theme(axis.title.x=element_blank()) +
  facet_grid(Month~Bayside, scales = "free", space = "free", drop= TRUE)

speciesbubbleplot_eDNA_comname_reads
```

Export
```{r}
ggsave(filename = "Figures-elasmos/speciesbubbleplot_eDNA_comname_reads.eps", plot = speciesbubbleplot_eDNA_comname_reads, units = c("in"), width = 8, height = 7, dpi = 300)
```


## Time Series Plots
Look at total read abundance and rel abundance for different spp per month. Aggregate sites by bayside (E/W) and type (open water/ bivalve habitat)

```{r}
# First group total reads by Species, Habitat_type and Month so there is one point per time point on the plot
TimeSeries_df <- species_df_reads %>%
  group_by(Month, Habitat_type, CommonName) %>%
  summarise(TotalReads = sum(Abundance))
TimeSeries_df
```


```{r}
#First force the order of Habitat type so it plots in desired order
TimeSeries_df$Habitat_type_f = factor(TimeSeries_df$Habitat_type, levels=c("OPEN WATER", "CLAM SANCTUARY", "OYSTER REEF"))


speciestimeseries_comname <- ggplot(TimeSeries_df, aes(x = Month, y = log10(TotalReads), color = CommonName)) +
  geom_line(aes(color = CommonName), show.legend = FALSE) +
  ylab("Log 10 (Reads)")+
  theme_bw() +
  theme(axis.title.x=element_blank(), legend.title=element_blank()) +
  scale_x_continuous(breaks=c(7,8,9), labels = c("J","A","S")) +
  facet_grid(Habitat_type_f~CommonName)

speciestimeseries_comname
```


## stopped here 5/28/24. Keep working on Figures above. Check if zeroes are plotting as small dots (may want to delete)
- change 7, 8, 9 to JAS in bubble plots. See this: https://r-graphics.org/recipe-facet-label-text
- export time series figure
- create dezcriptive figure captions

Next
- check for seasonality of cownose ray. Does it come in in August and outcompete other things? Is it more abundant at the bivalve sites?
- double check Pelagic sting ray is at the Inlet. Discuss in discussion
- think of a way of showing low abundance of reads in oyster/ clam shallow samples (except for occasional hit to cownose ray- possible predator of bivalves) - present total reads?
- emphasize diversity differences between bivalve vs open water sites















## Species Richness
Count unique species across all stations, grouped by date, for each method, trawl& eDNA (use filtered trawl data so only comparing MiFISh spp to MiFISh spp).

Count total number of species from each method for each date
```{r}
eDNA_richness <- tally(eDNA_uniques, name = "eDNA")
trawl_richness <- tally(trawl_uniques, name = "trawl")

speciesrichness <- full_join(eDNA_richness, trawl_richness, c("Datecode" = "DATECODE"))
speciesrichness <- pivot_longer(speciesrichness, !Datecode, names_to = "Method", values_to = "Richness")

speciesrichness$Datecode <- ymd(speciesrichness$Datecode) # convert to date format (better for plotting)

speciesrichness
```


Plot side-by-side
```{r}
species_richness_plot <- ggplot(speciesrichness, aes(x =Datecode, y = Richness)) +
  geom_line(aes(color = Method), size = 3) +
  theme_bw() +
  xlab("") +
  ylab("Species Richness")

species_richness_plot

# export plot
ggsave(filename = "Figures-elasmos/species_richness_plot.eps", plot = species_richness_plot, units = c("in"), width = 4, height = 3, dpi = 300)
```



Sum total number of species across all dates/ stations for entire study
```{r}
species_sums_abun_table <- trawl_eDNA_abun_table %>%
  group_by(CommonName) %>%
  summarise(CPUE = sum(Trawl_CPUE, na.rm = TRUE), 
            "Total Length (TL) PUE" = sum(Trawl_TLPUE, na.rm = TRUE), 
            "TL * Shedding Factor PUE" = sum(Trawl_Allometric_Shedding, na.rm = TRUE), 
            eDNA = sum(eDNA_RelAbun, na.rm=TRUE)) %>%
  pivot_longer(!CommonName, names_to = "Method", values_to = "Abundance")
  
# turn zeroes to NA so they don't plot 
species_sums_abun_table <- na_if(species_sums_abun_table,0)

species_sums_abun_table
```

## Species tally across whole study

For each species, plot side-by-side comparison of abundance (summed over whole study) using each method

```{r}
# First create a custom color scale to make this pretty
myColors <- colorRampPalette(brewer.pal(11,"Spectral"))(40)
names(myColors) <- levels(unique(species_sums_abun_table$CommonName))
colScale <- scale_colour_manual(name = "CommonName",values = myColors)

species_abun_sum_plot <- ggplot(species_sums_abun_table, aes(x = Abundance, y = reorder(CommonName, Abundance, function(x){sum(x,na.rm = TRUE)}), color = CommonName)) +
  geom_point(size = 5) +
  facet_wrap(~factor(Method, levels = c('CPUE','Total Length (TL) PUE','TL * Shedding Factor PUE','eDNA')), scales = "free_x", ncol = 4) +
  theme_bw() +
  xlab("Abundance") +
  ylab("") + 
  colScale +
  theme(legend.position = "none")

species_abun_sum_plot
```

Export plot
```{r}
ggsave(filename = "Figures-elasmos/species_abun_sum_plot.eps", plot = species_abun_sum_plot, units = c("in"), width = 10, height = 6, dpi = 300)
```




